{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-18T14:48:43.770427Z",
     "start_time": "2018-03-18T14:48:43.743052Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/Dropbox/wd/18-03-18\n"
     ]
    }
   ],
   "source": [
    "cd /Users/sr320/Dropbox/wd/18-03-18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-18T14:56:38.120534Z",
     "start_time": "2018-03-18T14:48:54.849764Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t138298\n",
      "Mapping efficiency:\t24.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t378173\n",
      "Sequence pairs did not map uniquely:\t57875\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t22372\t((converted) top strand)\n",
      "GA/CT/CT:\t46562\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t47173\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t22191\t((converted) bottom strand)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Did not find Samtools on the system. Alignments will be compressed with GZIP instead (.sam.gz)\n",
      "Reference genome folder provided is /Users/sr320/Dropbox/wd/18-03-15/genome/\t(absolute path is '/Users/sr320/Dropbox/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Users/sr320/Dropbox/wd/18-03-18'):\n",
      "/Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq\n",
      "/Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq\n",
      "Library was specified to be not strand-specific (non-directional), therefore alignments to all four possible bisulfite strands (OT, CTOT, OB and CTOB) will be reported\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Users/sr320/Dropbox/wd/18-03-18\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Users/sr320/Dropbox/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq and /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file 10_32_S32_L001_R1_001.fastq to 10_32_S32_L001_R1_001.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file 10_32_S32_L001_R1_001.fastq to 10_32_S32_L001_R1_001.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file 10_32_S32_L001_R1_001.fastq (574346 sequences in total)\n",
      "\n",
      "Writing a C -> T converted version of the input file 10_32_S32_L001_R2_001.fastq to 10_32_S32_L001_R2_001.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file 10_32_S32_L001_R2_001.fastq to 10_32_S32_L001_R2_001.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file 10_32_S32_L001_R2_001.fastq (574346 sequences in total)\n",
      "\n",
      "Input files are 10_32_S32_L001_R1_001.fastq_C_to_T.fastq and 10_32_S32_L001_R1_001.fastq_G_to_A.fastq and 10_32_S32_L001_R2_001.fastq_C_to_T.fastq and 10_32_S32_L001_R2_001.fastq_G_to_A.fastq (FastQ)\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Users/sr320/Dropbox/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.2 --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from 10_32_S32_L001_R1_001.fastq_C_to_T.fastq and 10_32_S32_L001_R2_001.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.2 --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_1:N:0:32/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTAGTTTTAGTTTGTATATTTTTTTTTAGTTGTATTTTTTTTTATAATTTTATATTAAGATGTGTGTGTAAGAAT\tCCCCCGGGGGGGGGGGFFCFGGGGGGGG8,CE,C,CEEFFEFEG,,,,<CEF,C,CF,,,,<,,,,,,,:,,,,,,\tYT:Z:UP\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_2:N:0:32/2\t141\t*\t0\t0\t*\t*\t0\t0\tCTTACCAATAAAATAAAAAACAAAAACTATACAAAAATCCTAAACAAACAAAAACTAAACACTTTCTTACCCTCTC\t-8A-8@--;,-,-<,,,,,+8,,,,,6C,C,<,,,,,;;@C,,,;,,,;,,,,,8C,,,;,8@BDEEE,99ACA@@\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2GAgenome (reading in sequences from 10_32_S32_L001_R1_001.fastq_G_to_A.fastq and 10_32_S32_L001_R2_001.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.2 --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_1:N:0:32/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTAATTTTAATTCATATATTTTTTTTTAATTATATTTTTTTTTATAATTTTATATTAAAATATATATATAAAAAT\tCCCCCGGGGGGGGGGGFFCFGGGGGGGG8,CE,C,CEEFFEFEG,,,,<CEF,C,CF,,,,<,,,,,,,:,,,,,,\tYT:Z:UP\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_2:N:0:32/2\t141\t*\t0\t0\t*\t*\t0\t0\tTTTATTAATAAAATAAAAAATAAAAATTATATAAAAATTTTAAATAAATAAAAATTAAATGTTTTTTTATTTTTTT\t-8A-8@--;,-,-<,,,,,+8,,,,,6C,C,<,,,,,;;@C,,,;,,,;,,,,,8C,,,;,8@BDEEE,99ACA@@\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2CTgenome (reading in sequences from 10_32_S32_L001_R1_001.fastq_G_to_A.fastq and 10_32_S32_L001_R2_001.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.2 --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_1:N:0:32/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTAATTTTAATTCATATATTTTTTTTTAATTATATTTTTTTTTATAATTTTATATTAAAATATATATATAAAAAT\tCCCCCGGGGGGGGGGGFFCFGGGGGGGG8,CE,C,CEEFFEFEG,,,,<CEF,C,CF,,,,<,,,,,,,:,,,,,,\tYT:Z:UP\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_2:N:0:32/2\t141\t*\t0\t0\t*\t*\t0\t0\tTTTATTAATAAAATAAAAAATAAAAATTATATAAAAATTTTAAATAAATAAAAATTAAATGTTTTTTTATTTTTTT\t-8A-8@--;,-,-<,,,,,+8,,,,,6C,C,<,,,,,;;@C,,,;,,,;,,,,,8C,,,;,8@BDEEE,99ACA@@\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from 10_32_S32_L001_R1_001.fastq_C_to_T.fastq and 10_32_S32_L001_R2_001.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.2 --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_1:N:0:32/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTAGTTTTAGTTTGTATATTTTTTTTTAGTTGTATTTTTTTTTATAATTTTATATTAAGATGTGTGTGTAAGAAT\tCCCCCGGGGGGGGGGGFFCFGGGGGGGG8,CE,C,CEEFFEFEG,,,,<CEF,C,CF,,,,<,,,,,,,:,,,,,,\tYT:Z:UP\n",
      "M03631:341:000000000-BLY9V:1:1101:13318:1785_2:N:0:32/2\t141\t*\t0\t0\t*\t*\t0\t0\tCTTACCAATAAAATAAAAAACAAAAACTATACAAAAATCCTAAACAAACAAAAACTAAACACTTTCTTACCCTCTC\t-8A-8@--;,-,-<,,,,,+8,,,,,6C,C,<,,,,,;;@C,,,;,,,;,,,,,8C,,,;,8@BDEEE,99ACA@@\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to 10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq and /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq\n",
      "574346 reads; of these:\n",
      "  574346 (100.00%) were paired; of these:\n",
      "    533029 (92.81%) aligned concordantly 0 times\n",
      "    22880 (3.98%) aligned concordantly exactly 1 time\n",
      "    18437 (3.21%) aligned concordantly >1 times\n",
      "7.19% overall alignment rate\n",
      "574346 reads; of these:\n",
      "  574346 (100.00%) were paired; of these:\n",
      "    533250 (92.84%) aligned concordantly 0 times\n",
      "    22623 (3.94%) aligned concordantly exactly 1 time\n",
      "    18473 (3.22%) aligned concordantly >1 times\n",
      "7.16% overall alignment rate\n",
      "574346 reads; of these:\n",
      "  574346 (100.00%) were paired; of these:\n",
      "    487235 (84.83%) aligned concordantly 0 times\n",
      "    48283 (8.41%) aligned concordantly exactly 1 time\n",
      "    38828 (6.76%) aligned concordantly >1 times\n",
      "15.17% overall alignment rate\n",
      "574346 reads; of these:\n",
      "  574346 (100.00%) were paired; of these:\n",
      "    487944 (84.96%) aligned concordantly 0 times\n",
      "    47750 (8.31%) aligned concordantly exactly 1 time\n",
      "    38652 (6.73%) aligned concordantly >1 times\n",
      "15.04% overall alignment rate\n",
      "Processed 574346 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files 10_32_S32_L001_R1_001.fastq_C_to_T.fastq, 10_32_S32_L001_R1_001.fastq_G_to_A.fastq, 10_32_S32_L001_R2_001.fastq_C_to_T.fastq and 10_32_S32_L001_R2_001.fastq_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t574346\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t4098290\n",
      "\n",
      "Total methylated C's in CpG context:\t449826\n",
      "Total methylated C's in CHG context:\t5957\n",
      "Total methylated C's in CHH context:\t15565\n",
      "Total methylated C's in Unknown context:\t215\n",
      "\n",
      "Total unmethylated C's in CpG context:\t136534\n",
      "Total unmethylated C's in CHG context:\t899729\n",
      "Total unmethylated C's in CHH context:\t2590679\n",
      "Total unmethylated C's in Unknown context:\t2589\n",
      "\n",
      "C methylated in CpG context:\t76.7%\n",
      "C methylated in CHG context:\t0.7%\n",
      "C methylated in CHH context:\t0.6%\n",
      "C methylated in unknown context (CN or CHN):\t7.7%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 7m 42s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "--non_directional \\\n",
    "-1 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq \\\n",
    "-2 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-18T15:16:25.242042Z",
     "start_time": "2018-03-18T15:16:25.063888Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "DESCRIPTION\n",
      "\n",
      "The following is a brief description of all options to control the Bismark\n",
      "methylation extractor. The script reads in a bisulfite read alignment results file \n",
      "produced by the Bismark bisulfite mapper (in BAM/CRAM/SAM format) and extracts the\n",
      "methylation information for individual cytosines. This information is found in the\n",
      "methylation call field which can contain the following characters:\n",
      "\n",
      "       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
      "       ~~~   X   for methylated C in CHG context                      ~~~\n",
      "       ~~~   x   for not methylated C CHG                             ~~~\n",
      "       ~~~   H   for methylated C in CHH context                      ~~~\n",
      "       ~~~   h   for not methylated C in CHH context                  ~~~\n",
      "       ~~~   Z   for methylated C in CpG context                      ~~~\n",
      "       ~~~   z   for not methylated C in CpG context                  ~~~\n",
      "       ~~~   U   for methylated C in Unknown context (CN or CHN       ~~~\n",
      "       ~~~   u   for not methylated C in Unknown context (CN or CHN)  ~~~\n",
      "       ~~~   .   for any bases not involving cytosines                ~~~\n",
      "       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\n",
      "\n",
      "The methylation extractor outputs result files for cytosines in CpG, CHG and CHH\n",
      "context (this distinction is actually already made in Bismark itself). As the methylation\n",
      "information for every C analysed can produce files which easily have tens or even hundreds of\n",
      "millions of lines, file sizes can become very large and more difficult to handle. The C\n",
      "methylation info additionally splits cytosine methylation calls up into one of the four possible\n",
      "strands a given bisulfite read aligned against:\n",
      "\n",
      "             OT      original top strand\n",
      "             CTOT    complementary to original top strand\n",
      "\n",
      "             OB      original bottom strand\n",
      "             CTOB    complementary to original bottom strand\n",
      "\n",
      "Thus, by default twelve individual output files are being generated per input file (unless\n",
      "--comprehensive is specified, see below). The output files can be imported into a genome\n",
      "viewer, such as SeqMonk, and re-combined into a single data group if desired (in fact\n",
      "unless the bisulfite reads were generated preserving directionality it doesn't make any\n",
      "sense to look at the data in a strand-specific manner). Strand-specific output files can\n",
      "optionally be skipped, in which case only three output files for CpG, CHG or CHH context\n",
      "will be generated. For both the strand-specific and comprehensive outputs there is also\n",
      "the option to merge both non-CpG contexts (CHG and CHH) into one single non-CpG context.\n",
      "\n",
      "\n",
      "The output files are in the following format (tab delimited):\n",
      "\n",
      "<sequence_id>     <strand>      <chromosome>     <position>     <methylation call>\n",
      "\n",
      "\n",
      "USAGE: bismark_methylation_extractor [options] <filenames>\n",
      "\n",
      "\n",
      "ARGUMENTS:\n",
      "==========\n",
      "\n",
      "<filenames>              A space-separated list of Bismark result files in SAM format from\n",
      "                         which methylation information is extracted for every cytosine in\n",
      "                         the reads. For alignment files in the older custom Bismark output\n",
      "                         see option '--vanilla'.\n",
      "\n",
      "OPTIONS:\n",
      "\n",
      "-s/--single-end          Input file(s) are Bismark result file(s) generated from single-end\n",
      "                         read data. If neither -s nor -p is set the type of experiment will\n",
      "                         be determined automatically.\n",
      "\n",
      "-p/--paired-end          Input file(s) are Bismark result file(s) generated from paired-end\n",
      "                         read data. If neither -s nor -p is set the type of experiment will\n",
      "                         be determined automatically.\n",
      "\n",
      "--vanilla                The Bismark result input file(s) are in the old custom Bismark format\n",
      "                         (up to version 0.5.x) and not in SAM format which is the default as\n",
      "                         of Bismark version 0.6.x or higher. Default: OFF.\n",
      "\n",
      "--no_overlap             For paired-end reads it is theoretically possible that read_1 and\n",
      "                         read_2 overlap. This option avoids scoring overlapping methylation\n",
      "                         calls twice (only methylation calls of read 1 are used for in the process\n",
      "                         since read 1 has historically higher quality basecalls than read 2).\n",
      "                         Whilst this option removes a bias towards more methylation calls\n",
      "                         in the center of sequenced fragments it may de facto remove a sizable\n",
      "                         proportion of the data. This option is on by default for paired-end data\n",
      "                         but can be disabled using --include_overlap. Default: ON.\n",
      "\n",
      "--include_overlap        For paired-end data all methylation calls will be extracted irrespective of\n",
      "                         of whether they overlap or not. Default: OFF.\n",
      "\n",
      "--ignore <int>           Ignore the first <int> bp from the 5' end of Read 1 (or single-end alignment\n",
      "                         files) when processing the methylation call string. This can remove e.g. a\n",
      "                         restriction enzyme site at the start of each read or any other source of\n",
      "                         bias (such as PBAT-Seq data).\n",
      "\n",
      "--ignore_r2 <int>        Ignore the first <int> bp from the 5' end of Read 2 of paired-end sequencing\n",
      "                         results only. Since the first couple of bases in Read 2 of BS-Seq experiments\n",
      "                         show a severe bias towards non-methylation as a result of end-repairing\n",
      "                         sonicated fragments with unmethylated cytosines (see M-bias plot), it is\n",
      "                         recommended that the first couple of bp of Read 2 are removed before\n",
      "                         starting downstream analysis. Please see the section on M-bias plots in the\n",
      "                         Bismark User Guide for more details.\n",
      "\n",
      "--ignore_3prime <int>    Ignore the last <int> bp from the 3' end of Read 1 (or single-end alignment\n",
      "                         files) when processing the methylation call string. This can remove unwanted\n",
      "                         biases from the end of reads.\n",
      "\n",
      "--ignore_3prime_r2 <int> Ignore the last <int> bp from the 3' end of Read 2 of paired-end sequencing\n",
      "                         results only. This can remove unwanted biases from the end of reads.\n",
      "\n",
      "--comprehensive          Specifying this option will merge all four possible strand-specific\n",
      "                         methylation info into context-dependent output files. The default\n",
      "\n",
      "                         contexts are:\n",
      "                          - CpG context\n",
      "                          - CHG context\n",
      "                          - CHH context\n",
      "\n",
      "--merge_non_CpG          This will produce two output files (in --comprehensive mode) or eight\n",
      "                         strand-specific output files (default) for Cs in\n",
      "                          - CpG context\n",
      "                          - non-CpG context\n",
      "\n",
      "--report                 Prints out a short methylation summary as well as the paramaters used to run\n",
      "                         this script. Default: ON.\n",
      "\n",
      "--no_header              Suppresses the Bismark version header line in all output files for more convenient\n",
      "                         batch processing.\n",
      "\n",
      "-o/--output DIR          Allows specification of a different output directory (absolute or relative\n",
      "                         path). If not specified explicitly, the output will be written to the current directory.\n",
      "\n",
      "--samtools_path          The path to your Samtools installation, e.g. /home/user/samtools/. Does not need to be specified\n",
      "                         explicitly if Samtools is in the PATH already.\n",
      "\n",
      "--gzip                   The methylation extractor files (CpG_OT_..., CpG_OB_... etc) will be written out in\n",
      "                         a GZIP compressed form to save disk space. This option is also passed on to the genome-wide\n",
      "                         cytosine report. BedGraph and coverage files will be written out as .gz by default.\n",
      "\n",
      "--version                Displays version information.\n",
      "\n",
      "-h/--help                Displays this help file and exits.\n",
      "\n",
      "--mbias_only             The methylation extractor will read the entire file but only output the M-bias table and plots as \n",
      "                         well as a report (optional) and then quit. Default: OFF.\n",
      "\n",
      "--mbias_off              The methylation extractor will process the entire file as usual but doesn't write out any M-bias report.\n",
      "                         Only recommended for users who deliberately want to keep an earlier version of the M-bias report. \n",
      "                         Default: OFF.\n",
      "\n",
      "--parallel <int>         May also be --multicore <int>. Sets the number of cores to be used for the methylation extraction process.\n",
      "                         If system resources are plentiful this is a viable option to speed up the extraction process (we observed a\n",
      "                         near linear speed increase for up to 10 cores used). Please note that a typical process of extracting a BAM file\n",
      "                         and writing out '.gz' output streams will in fact use ~3 cores per value of --parallel <int>\n",
      "                         specified (1 for the methylation extractor itself, 1 for a Samtools stream, 1 for GZIP stream), so\n",
      "                         --parallel 10 is likely to use around 30 cores of system resources. This option has no bearing\n",
      "                         on the bismark2bedGraph or genome-wide cytosine report processes.\n",
      "\n",
      "--yacht                  This option (Yet Another Context Hunting Tool) writes out additional information about the read a methylation\n",
      "                         call belongs to, and its output is meant to be fed into the NOMe_filtering script. This option writes out\n",
      "                         a single 'any_C_context' file that contains all methylation calls for a read consecutively. Its intended use\n",
      "                         is single-cell NOMe-Seq data, and thus this option works only in single-end mode (paired-end reads often suffer\n",
      "                         from chimaera problems...)\n",
      "\n",
      "                         --yacht will add three additional columns to the standard methylation call files:\n",
      "                                    <read start>  <read end>  <read orientation>\n",
      "                         For forward reads (+ orientation) the start position is the left-most position wheras for reverse reads\n",
      "                         (- orientation) it is the rightmost position.\n",
      "\n",
      "\n",
      "bedGraph specific options:\n",
      "==========================\n",
      "\n",
      "--bedGraph               After finishing the methylation extraction, the methylation output is written into a\n",
      "                         sorted bedGraph file that reports the position of a given cytosine and its methylation \n",
      "                         state (in %, see details below). The methylation extractor output is temporarily split up into\n",
      "                         temporary files, one per chromosome (written into the current directory or folder\n",
      "                         specified with -o/--output); these temp files are then used for sorting and deleted\n",
      "                         afterwards. By default, only cytosines in CpG context will be sorted. The option\n",
      "                         '--CX_context' may be used to report all cytosines irrespective of sequence context\n",
      "                         (this will take MUCH longer!). The default folder for temporary files during the sorting\n",
      "                         process is the output directory. The bedGraph conversion step is performed by the external\n",
      "                         module 'bismark2bedGraph'; this script needs to reside in the same folder as the \n",
      "                         bismark_methylation_extractor itself.\n",
      "\n",
      "--zero_based             Write out an additional coverage file (ending in .zero.cov) that uses 0-based genomic start\n",
      "                         and 1-based genomic end coordinates (zero-based, half-open), like used in the bedGraph file,\n",
      "                         instead of using 1-based coordinates throughout. Default: OFF.\n",
      "\n",
      "\n",
      "--cutoff [threshold]     The minimum number of times any methylation state (methylated or unmethylated) has to be seen\n",
      "                         for a nucleotide before its methylation percentage is reported. Default: 1.\n",
      "\n",
      "--remove_spaces          Replaces whitespaces in the sequence ID field with underscores to allow sorting.\n",
      "\n",
      "--CX/--CX_context        The sorted bedGraph output file contains information on every single cytosine that was covered\n",
      "                         in the experiment irrespective of its sequence context. This applies to both forward and\n",
      "                         reverse strands. Please be aware that this option may generate large temporary and output files\n",
      "                         and may take a long time to sort (up to many hours). Default: OFF.\n",
      "                         (i.e. Default = CpG context only).\n",
      "\n",
      "--buffer_size <string>   This allows you to specify the main memory sort buffer when sorting the methylation information.\n",
      "                         Either specify a percentage of physical memory by appending % (e.g. --buffer_size 50%) or\n",
      "\t\t\t a multiple of 1024 bytes, e.g. 'K' multiplies by 1024, 'M' by 1048576 and so on for 'T' etc. \n",
      "                         (e.g. --buffer_size 20G). For more information on sort type 'info sort' on a command line.\n",
      "                         Defaults to 2G.\n",
      "\n",
      "--scaffolds/--gazillion  Users working with unfinished genomes sporting tens or even hundreds of thousands of\n",
      "                         scaffolds/contigs/chromosomes frequently encountered errors with pre-sorting reads to\n",
      "                         individual chromosome files. These errors were caused by the operating system's limit\n",
      "                         of the number of filehandle that can be written to at any one time (typically 1024; to\n",
      "                         find out this limit on Linux, type: ulimit -a).\n",
      "                         To bypass the limitation of open filehandles, the option --scaffolds does not pre-sort\n",
      "                         methylation calls into individual chromosome files. Instead, all input files are\n",
      "                         temporarily merged into a single file (unless there is only a single file), and this\n",
      "                         file will then be sorted by both chromosome AND position using the Unix sort command.\n",
      "                         Please be aware that this option might take a looooong time to complete, depending on\n",
      "                         the size of the input files, and the memory you allocate to this process (see --buffer_size).\n",
      "                         Nevertheless, it seems to be working.\n",
      "\n",
      "--ample_memory           Using this option will not sort chromosomal positions using the UNIX 'sort' command, but will\n",
      "                         instead use two arrays to sort methylated and unmethylated calls. This may result in a faster\n",
      "                         sorting process of very large files, but this comes at the cost of a larger memory footprint\n",
      "                         (two arrays of the length of the largest human chromosome 1 (~250M bp) consume around 16GB\n",
      "                         of RAM). Due to overheads in creating and looping through these arrays it seems that it will\n",
      "                         actually be *slower* for small files (few million alignments), and we are currently testing at\n",
      "                         which point it is advisable to use this option. Note that --ample_memory is not compatible\n",
      "                         with options '--scaffolds/--gazillion' (as it requires pre-sorted files to begin with).\n",
      "\n",
      "\n",
      "\n",
      "Genome-wide cytosine methylation report specific options:\n",
      "=========================================================\n",
      "\n",
      "--cytosine_report        After the conversion to bedGraph has completed, the option '--cytosine_report' produces a\n",
      "                         genome-wide methylation report for all cytosines in the genome. By default, the output uses 1-based\n",
      "                         chromosome coordinates (zero-based start coords are optional) and reports CpG context only (all\n",
      "                         cytosine context is optional). The output considers all Cs on both forward and reverse strands and\n",
      "                         reports their position, strand, trinucleotide content and methylation state (counts are 0 if not\n",
      "                         covered). The cytosine report conversion step is performed by the external module\n",
      "                         'coverage2cytosine'; this script needs to reside in the same folder as the bismark_methylation_extractor\n",
      "                         itself.\n",
      "\n",
      "--CX/--CX_context        The output file contains information on every single cytosine in the genome irrespective of\n",
      "                         its context. This applies to both forward and reverse strands. Please be aware that this will\n",
      "                         generate output files with > 1.1 billion lines for a mammalian genome such as human or mouse.\n",
      "                         Default: OFF (i.e. Default = CpG context only).\n",
      "\n",
      "--zero_based             Uses 0-based genomic coordinates instead of 1-based coordinates. Default: OFF.\n",
      "\n",
      "--genome_folder <path>   Enter the genome folder you wish to use to extract sequences from (full path only). Accepted\n",
      "                         formats are FastA files ending with '.fa' or '.fasta'. Specifying a genome folder path is mandatory.\n",
      "\n",
      "--split_by_chromosome    Writes the output into individual files for each chromosome instead of a single output file. Files\n",
      "                         will be named to include the input filename and the chromosome number.\n",
      "\n",
      "\n",
      "\n",
      "OUTPUT:\n",
      "\n",
      "The bismark_methylation_extractor output is in the form:\n",
      "========================================================\n",
      "<seq-ID>  <methylation state*>  <chromosome>  <start position (= end position)>  <methylation call>\n",
      "\n",
      "* Methylated cytosines receive a '+' orientation,\n",
      "* Unmethylated cytosines receive a '-' orientation.\n",
      "\n",
      "The bismark_methylation_extractor output with --yacht (optional) specified is in the form:\n",
      "==========================================================================================\n",
      "<seq-ID>  <methylation state*>  <chromosome>  <start position (= end position)>  <methylation call>  <read start>  <read end>  <read orientation>\n",
      "\n",
      "* Methylated cytosines receive a '+' orientation,\n",
      "* Unmethylated cytosines receive a '-' orientation.\n",
      "\n",
      "\n",
      "The bedGraph output (optional) looks like this (tab-delimited; 0-based start coords, 1-based end coords):\n",
      "=========================================================================================================\n",
      "\n",
      "track type=bedGraph (header line)\n",
      "\n",
      "<chromosome>  <start position>  <end position>  <methylation percentage>\n",
      "\n",
      "\n",
      "The coverage output looks like this (tab-delimited, 1-based genomic coords; zero-based half-open coordinates available with '--zero_based'):\n",
      "============================================================================================================================================\n",
      "\n",
      "<chromosome>  <start position>  <end position>  <methylation percentage>  <count methylated>  <count non-methylated>\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "The genome-wide cytosine methylation output file is tab-delimited in the following format:\n",
      "==========================================================================================\n",
      "<chromosome>  <position>  <strand>  <count methylated>  <count non-methylated>  <C-context>  <trinucleotide context>\n",
      "\n",
      "\n",
      "\n",
      "This script was last modified on 05 October 2017.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-03-18T15:22:36.563955Z",
     "start_time": "2018-03-18T15:21:40.530739Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file /Users/sr320/Dropbox/wd/18-03-18/10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz for positional sorting (which would be bad...)\t"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " *** Bismark methylation extractor version v0.19.0 ***\n",
      "\n",
      "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
      "\n",
      "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Bismark paired-end SAM format specified (default)\n",
      "Number of cores to be used: 1\n",
      "Output will be written to the current directory ('/Users/sr320/Dropbox/wd/18-03-18')\n",
      "\n",
      "...passed!\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz\n",
      "\n",
      "Now reading in Bismark result file /Users/sr320/Dropbox/wd/18-03-18/10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --non_directional -1 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq -2 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq\"\n",
      "\n",
      "Processed 138298 lines in total\n",
      "Total number of methylation call strings processed: 276596\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t3679310\n",
      "\n",
      "Total methylated C's in CpG context:\t396021\n",
      "Total methylated C's in CHG context:\t5322\n",
      "Total methylated C's in CHH context:\t13563\n",
      "\n",
      "Total C to T conversions in CpG context:\t122805\n",
      "Total C to T conversions in CHG context:\t805457\n",
      "Total C to T conversions in CHH context:\t2336142\n",
      "\n",
      "C methylated in CpG context:\t76.3%\n",
      "C methylated in CHG context:\t0.7%\n",
      "C methylated in CHH context:\t0.6%\n",
      "\n",
      "\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 76\n",
      "Maximum read length of Read 2: 76\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 76\n",
      "Maximum read length of Read 2: 76\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "CpG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CpG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CpG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CpG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHH_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHH_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHH_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "CHH_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz.txt.gz contains data ->\tkept\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --gzip \\\n",
    "--samtools_path /Applications/bioinfo/samtools-1.7 \\\n",
    "/Users/sr320/Dropbox/wd/18-03-18/10_32_S32_L001_R1_001_bismark_bt2_pe.sam.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-03-23T20:53:48.224Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "--non_directional \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-1 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R1_001.fastq \\\n",
    "-2 /Users/sr320/Dropbox/wd/18-03-15/10_32_S32_L001_R2_001.fastq \\\n",
    "-o /Users/sr320/Dropbox/wd/18-03-23/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
