{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-07\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-04-07/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "mkdir fqc\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of zr2096_10_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_10_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_1_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_1_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_10_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_10_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_10_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_10_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_2_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_2_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_1_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_1_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_1_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_1_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_3_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_3_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_2_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_2_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_2_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_2_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_4_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_4_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 100% complete for zr2096_3_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_3_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 100% complete for zr2096_3_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_3_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_5_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_5_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_4_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_4_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_4_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_4_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_6_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_6_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_5_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_5_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_5_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_5_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_7_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_7_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_6_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_6_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_6_s1_R1.fastq.gz\n",
      "Started analysis of zr2096_8_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_6_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_8_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 20% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_7_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_7_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_7_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_7_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_9_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Started analysis of zr2096_9_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 5% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 5% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 10% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 10% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_8_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 15% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 15% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_8_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_8_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 20% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 25% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 25% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 30% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 30% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 35% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 35% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 40% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 40% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 45% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 45% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 50% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 50% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 55% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 55% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 60% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 60% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 65% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 65% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 70% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 70% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 75% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 75% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 80% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 80% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 85% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 85% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 90% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 90% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Approx 95% complete for zr2096_9_s1_R2.fastq.gz\n",
      "Approx 95% complete for zr2096_9_s1_R1.fastq.gz\n",
      "Analysis complete for zr2096_9_s1_R2.fastq.gz\n",
      "Analysis complete for zr2096_9_s1_R1.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/FastQC/fastqc -t 4 \\\n",
    "-o /Volumes/Serine/wd/18-04-07/fqc \\\n",
    "/Volumes/Serine/wd/18-04-07/zr*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.0.dev0\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '/Volumes/Serine/wd/18-04-07/fqc'\n",
      "[INFO   ]          fastqc : Found 20 reports\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc /Volumes/Serine/wd/18-04-07/fqc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "source": [
    "# trimming"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      " USAGE:\r\n",
      "\r\n",
      "trim_galore [options] <filename(s)>\r\n",
      "\r\n",
      "\r\n",
      "-h/--help               Print this help message and exits.\r\n",
      "\r\n",
      "-v/--version            Print the version information and exits.\r\n",
      "\r\n",
      "-q/--quality <INT>      Trim low-quality ends from reads in addition to adapter removal. For\r\n",
      "                        RRBS samples, quality trimming will be performed first, and adapter\r\n",
      "                        trimming is carried in a second round. Other files are quality and adapter\r\n",
      "                        trimmed in a single pass. The algorithm is the same as the one used by BWA\r\n",
      "                        (Subtract INT from all qualities; compute partial sums from all indices\r\n",
      "                        to the end of the sequence; cut sequence at the index at which the sum is\r\n",
      "                        minimal). Default Phred score: 20.\r\n",
      "\r\n",
      "--phred33               Instructs Cutadapt to use ASCII+33 quality scores as Phred scores\r\n",
      "                        (Sanger/Illumina 1.9+ encoding) for quality trimming. Default: ON.\r\n",
      "\r\n",
      "--phred64               Instructs Cutadapt to use ASCII+64 quality scores as Phred scores\r\n",
      "                        (Illumina 1.5 encoding) for quality trimming.\r\n",
      "\r\n",
      "--fastqc                Run FastQC in the default mode on the FastQ file once trimming is complete.\r\n",
      "\r\n",
      "--fastqc_args \"<ARGS>\"  Passes extra arguments to FastQC. If more than one argument is to be passed\r\n",
      "                        to FastQC they must be in the form \"arg1 arg2 etc.\". An example would be:\r\n",
      "                        --fastqc_args \"--nogroup --outdir /home/\". Passing extra arguments will\r\n",
      "                        automatically invoke FastQC, so --fastqc does not have to be specified\r\n",
      "                        separately.\r\n",
      "\r\n",
      "-a/--adapter <STRING>   Adapter sequence to be trimmed. If not specified explicitly, Trim Galore will\r\n",
      "                        try to auto-detect whether the Illumina universal, Nextera transposase or Illumina\r\n",
      "                        small RNA adapter sequence was used. Also see '--illumina', '--nextera' and\r\n",
      "                        '--small_rna'. If no adapter can be detected within the first 1 million sequences\r\n",
      "                        of the first file specified Trim Galore defaults to '--illumina'.\r\n",
      "\r\n",
      "-a2/--adapter2 <STRING> Optional adapter sequence to be trimmed off read 2 of paired-end files. This\r\n",
      "                        option requires '--paired' to be specified as well.\r\n",
      "\r\n",
      "--illumina              Adapter sequence to be trimmed is the first 13bp of the Illumina universal adapter\r\n",
      "                        'AGATCGGAAGAGC' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--nextera               Adapter sequence to be trimmed is the first 12bp of the Nextera adapter\r\n",
      "                        'CTGTCTCTTATA' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--small_rna             Adapter sequence to be trimmed is the first 12bp of the Illumina Small RNA Adapter\r\n",
      "                        'ATGGAATTCTCG' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "\r\n",
      "--stringency <INT>      Overlap with adapter sequence required to trim a sequence. Defaults to a\r\n",
      "                        very stringent setting of 1, i.e. even a single bp of overlapping sequence\r\n",
      "                        will be trimmed off from the 3' end of any read.\r\n",
      "\r\n",
      "-e <ERROR RATE>         Maximum allowed error rate (no. of errors divided by the length of the matching\r\n",
      "                        region) (default: 0.1)\r\n",
      "\r\n",
      "--gzip                  Compress the output file with GZIP. If the input files are GZIP-compressed\r\n",
      "                        the output files will automatically be GZIP compressed as well. As of v0.2.8 the\r\n",
      "                        compression will take place on the fly.\r\n",
      "\r\n",
      "--dont_gzip             Output files won't be compressed with GZIP. This option overrides --gzip.\r\n",
      "\r\n",
      "--length <INT>          Discard reads that became shorter than length INT because of either\r\n",
      "                        quality or adapter trimming. A value of '0' effectively disables\r\n",
      "                        this behaviour. Default: 20 bp.\r\n",
      "\r\n",
      "                        For paired-end files, both reads of a read-pair need to be longer than\r\n",
      "                        <INT> bp to be printed out to validated paired-end files (see option --paired).\r\n",
      "                        If only one read became too short there is the possibility of keeping such\r\n",
      "                        unpaired single-end reads (see --retain_unpaired). Default pair-cutoff: 20 bp.\r\n",
      "\r\n",
      "-o/--output_dir <DIR>   If specified all output will be written to this directory instead of the current\r\n",
      "                        directory.\r\n",
      "\r\n",
      "--no_report_file        If specified no report file will be generated.\r\n",
      "\r\n",
      "--suppress_warn         If specified any output to STDOUT or STDERR will be suppressed.\r\n",
      "\r\n",
      "--clip_R1 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 1 (or single-end\r\n",
      "                        reads). This may be useful if the qualities were very poor, or if there is some\r\n",
      "                        sort of unwanted bias at the 5' end. Default: OFF.\r\n",
      "\r\n",
      "--clip_R2 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 2 (paired-end reads\r\n",
      "                        only). This may be useful if the qualities were very poor, or if there is some sort\r\n",
      "                        of unwanted bias at the 5' end. For paired-end BS-Seq, it is recommended to remove\r\n",
      "                        the first few bp because the end-repair reaction may introduce a bias towards low\r\n",
      "                        methylation. Please refer to the M-bias plot section in the Bismark User Guide for\r\n",
      "                        some examples. Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R1 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 1 (or single-end\r\n",
      "                        reads) AFTER adapter/quality trimming has been performed. This may remove some unwanted\r\n",
      "                        bias from the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R2 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 2 AFTER\r\n",
      "                        adapter/quality trimming has been performed. This may remove some unwanted bias from\r\n",
      "                        the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--path_to_cutadapt </path/to/cutadapt>     You may use this option to specify a path to the Cutadapt executable,\r\n",
      "                        e.g. /my/home/cutadapt-1.7.1/bin/cutadapt. Else it is assumed that Cutadapt is in\r\n",
      "                        the PATH.\r\n",
      "\r\n",
      "\r\n",
      "RRBS-specific options (MspI digested material):\r\n",
      "\r\n",
      "--rrbs                  Specifies that the input file was an MspI digested RRBS sample (recognition\r\n",
      "                        site: CCGG). Sequences which were adapter-trimmed will have a further 2 bp\r\n",
      "                        removed from their 3' end. This is to avoid that the filled-in C close to the\r\n",
      "                        second MspI site in a sequence is used for methylation calls. Sequences which\r\n",
      "                        were merely trimmed because of poor quality will not be shortened further.\r\n",
      "\r\n",
      "--non_directional       Selecting this option for non-directional RRBS libraries will screen\r\n",
      "                        quality-trimmed sequences for 'CAA' or 'CGA' at the start of the read\r\n",
      "                        and, if found, removes the first two basepairs. Like with the option\r\n",
      "                        '--rrbs' this avoids using cytosine positions that were filled-in\r\n",
      "                        during the end-repair step. '--non_directional' requires '--rrbs' to\r\n",
      "                        be specified as well.\r\n",
      "\r\n",
      "--keep                  Keep the quality trimmed intermediate file. Default: off, which means\r\n",
      "                        the temporary file is being deleted after adapter trimming. Only has\r\n",
      "                        an effect for RRBS samples since other FastQ files are not trimmed\r\n",
      "                        for poor qualities separately.\r\n",
      "\r\n",
      "\r\n",
      "Note for RRBS using MseI:\r\n",
      "\r\n",
      "If your DNA material was digested with MseI (recognition motif: TTAA) instead of MspI it is NOT necessary\r\n",
      "to specify --rrbs or --non_directional since virtually all reads should start with the sequence\r\n",
      "'TAA', and this holds true for both directional and non-directional libraries. As the end-repair of 'TAA'\r\n",
      "restricted sites does not involve any cytosines it does not need to be treated especially. Instead, simply\r\n",
      "run Trim Galore! in the standard (i.e. non-RRBS) mode.\r\n",
      "\r\n",
      "\r\n",
      "Paired-end specific options:\r\n",
      "\r\n",
      "--paired                This option performs length trimming of quality/adapter/RRBS trimmed reads for\r\n",
      "                        paired-end files. To pass the validation test, both sequences of a sequence pair\r\n",
      "                        are required to have a certain minimum length which is governed by the option\r\n",
      "                        --length (see above). If only one read passes this length threshold the\r\n",
      "                        other read can be rescued (see option --retain_unpaired). Using this option lets\r\n",
      "                        you discard too short read pairs without disturbing the sequence-by-sequence order\r\n",
      "                        of FastQ files which is required by many aligners.\r\n",
      "\r\n",
      "                        Trim Galore! expects paired-end files to be supplied in a pairwise fashion, e.g.\r\n",
      "                        file1_1.fq file1_2.fq SRR2_1.fq.gz SRR2_2.fq.gz ... .\r\n",
      "\r\n",
      "-t/--trim1              Trims 1 bp off every read from its 3' end. This may be needed for FastQ files that\r\n",
      "                        are to be aligned as paired-end data with Bowtie. This is because Bowtie (1) regards\r\n",
      "                        alignments like this:\r\n",
      "\r\n",
      "                          R1 --------------------------->     or this:    ----------------------->  R1\r\n",
      "                          R2 <---------------------------                       <-----------------  R2\r\n",
      "\r\n",
      "                        as invalid (whenever a start/end coordinate is contained within the other read).\r\n",
      "\r\n",
      "--retain_unpaired       If only one of the two paired-end reads became too short, the longer\r\n",
      "                        read will be written to either '.unpaired_1.fq' or '.unpaired_2.fq'\r\n",
      "                        output files. The length cutoff for unpaired single-end reads is\r\n",
      "                        governed by the parameters -r1/--length_1 and -r2/--length_2. Default: OFF.\r\n",
      "\r\n",
      "-r1/--length_1 <INT>    Unpaired single-end read length cutoff needed for read 1 to be written to\r\n",
      "                        '.unpaired_1.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "-r2/--length_2 <INT>    Unpaired single-end read length cutoff needed for read 2 to be written to\r\n",
      "                        '.unpaired_2.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "\r\n",
      "Last modified on 06 May 2015.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/trim_galore_zip/trim_galore -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.16\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "smallRNA\t0\tATGGAATTCTCG\t1000000\t0.00\n",
      "Illumina\t0\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "Use of uninitialized value in concatenation (.) or string at /Applications/bioinfo/trim_galore_zip/trim_galore line 145, <AUTODETECT> line 4000000.\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Nextera: 0, count smallRNA: )\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to '/Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.0\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "\n",
      "Writing final adapter and quality trimmed output to zr2096_5_s1_R1_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "This is cutadapt 1.16 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 817.93 s (26 us/read; 2.33 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:              31,778,715\n",
      "Reads with adapters:                11,626,318 (36.6%)\n",
      "Reads written (passing filters):    31,778,715 (100.0%)\n",
      "\n",
      "Total basepairs processed: 2,881,909,346 bp\n",
      "Quality-trimmed:               9,171,799 bp (0.3%)\n",
      "Total written (filtered):  2,858,698,169 bp (99.2%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 11626318 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 42.8%\n",
      "  C: 13.2%\n",
      "  G: 10.5%\n",
      "  T: 33.5%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t10070072\t7944678.8\t0\t10070072\n",
      "2\t1099130\t1986169.7\t0\t1099130\n",
      "3\t285046\t496542.4\t0\t285046\n",
      "4\t142993\t124135.6\t0\t142993\n",
      "5\t5542\t31033.9\t0\t5542\n",
      "6\t4163\t7758.5\t0\t4163\n",
      "7\t2228\t1939.6\t0\t2228\n",
      "8\t1001\t484.9\t0\t1001\n",
      "9\t563\t121.2\t0\t225 338\n",
      "10\t927\t30.3\t1\t238 689\n",
      "11\t698\t7.6\t1\t53 645\n",
      "12\t533\t1.9\t1\t82 451\n",
      "13\t615\t0.5\t1\t0 615\n",
      "14\t799\t0.5\t1\t0 799\n",
      "15\t1534\t0.5\t1\t0 1534\n",
      "16\t4395\t0.5\t1\t0 4395\n",
      "17\t4610\t0.5\t1\t0 4610\n",
      "18\t328\t0.5\t1\t0 328\n",
      "19\t75\t0.5\t1\t0 75\n",
      "20\t30\t0.5\t1\t0 30\n",
      "21\t26\t0.5\t1\t0 26\n",
      "22\t33\t0.5\t1\t0 33\n",
      "23\t35\t0.5\t1\t0 35\n",
      "24\t31\t0.5\t1\t0 31\n",
      "25\t32\t0.5\t1\t0 32\n",
      "26\t37\t0.5\t1\t0 37\n",
      "27\t35\t0.5\t1\t0 35\n",
      "28\t33\t0.5\t1\t0 33\n",
      "29\t35\t0.5\t1\t0 35\n",
      "30\t28\t0.5\t1\t0 28\n",
      "31\t28\t0.5\t1\t0 28\n",
      "32\t37\t0.5\t1\t0 37\n",
      "33\t33\t0.5\t1\t0 33\n",
      "34\t24\t0.5\t1\t0 24\n",
      "35\t29\t0.5\t1\t0 29\n",
      "36\t29\t0.5\t1\t0 29\n",
      "37\t29\t0.5\t1\t0 29\n",
      "38\t17\t0.5\t1\t0 17\n",
      "39\t38\t0.5\t1\t0 38\n",
      "40\t17\t0.5\t1\t0 17\n",
      "41\t33\t0.5\t1\t0 33\n",
      "42\t24\t0.5\t1\t0 24\n",
      "43\t31\t0.5\t1\t0 31\n",
      "44\t17\t0.5\t1\t0 17\n",
      "45\t26\t0.5\t1\t0 26\n",
      "46\t19\t0.5\t1\t0 19\n",
      "47\t27\t0.5\t1\t0 27\n",
      "48\t17\t0.5\t1\t0 17\n",
      "49\t24\t0.5\t1\t0 24\n",
      "50\t15\t0.5\t1\t0 15\n",
      "51\t19\t0.5\t1\t0 19\n",
      "52\t13\t0.5\t1\t0 13\n",
      "53\t16\t0.5\t1\t0 16\n",
      "54\t14\t0.5\t1\t0 14\n",
      "55\t16\t0.5\t1\t0 16\n",
      "56\t13\t0.5\t1\t0 13\n",
      "57\t9\t0.5\t1\t0 9\n",
      "58\t4\t0.5\t1\t0 4\n",
      "59\t11\t0.5\t1\t0 11\n",
      "60\t11\t0.5\t1\t0 11\n",
      "61\t11\t0.5\t1\t0 11\n",
      "62\t3\t0.5\t1\t0 3\n",
      "63\t4\t0.5\t1\t0 4\n",
      "64\t4\t0.5\t1\t0 4\n",
      "65\t8\t0.5\t1\t0 8\n",
      "66\t8\t0.5\t1\t0 8\n",
      "67\t4\t0.5\t1\t0 4\n",
      "68\t3\t0.5\t1\t0 3\n",
      "69\t3\t0.5\t1\t0 3\n",
      "70\t3\t0.5\t1\t0 3\n",
      "71\t4\t0.5\t1\t0 4\n",
      "72\t3\t0.5\t1\t0 3\n",
      "73\t4\t0.5\t1\t0 4\n",
      "74\t1\t0.5\t1\t0 1\n",
      "77\t1\t0.5\t1\t0 1\n",
      "78\t1\t0.5\t1\t0 1\n",
      "79\t1\t0.5\t1\t0 1\n",
      "81\t2\t0.5\t1\t0 2\n",
      "82\t2\t0.5\t1\t0 2\n",
      "83\t1\t0.5\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq\n",
      "=============================================\n",
      "31778715 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/Volumes/Serine/wd/18-04-10/zr2096_5_s1_R2.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R2.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.0\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "\n",
      "Writing final adapter and quality trimmed output to zr2096_5_s1_R2_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/Serine/wd/18-04-10/zr2096_5_s1_R2.fastq <<< \n",
      "cutadapt version 1.16\n",
      "Copyright (C) 2010-2017 Marcel Martin <marcel.martin@scilifelab.se>\n",
      "\n",
      "cutadapt removes adapter sequences from high-throughput sequencing reads.\n",
      "\n",
      "Usage:\n",
      "    cutadapt -a ADAPTER [options] [-o output.fastq] input.fastq\n",
      "\n",
      "For paired-end reads:\n",
      "    cutadapt -a ADAPT1 -A ADAPT2 [options] -o out1.fastq -p out2.fastq in1.fastq in2.fastq\n",
      "\n",
      "Replace \"ADAPTER\" with the actual sequence of your 3' adapter. IUPAC wildcard\n",
      "characters are supported. The reverse complement is *not* automatically\n",
      "searched. All reads from input.fastq will be written to output.fastq with the\n",
      "adapter sequence removed. Adapter matching is error-tolerant. Multiple adapter\n",
      "sequences can be given (use further -a options), but only the best-matching\n",
      "adapter will be removed.\n",
      "\n",
      "Input may also be in FASTA format. Compressed input and output is supported and\n",
      "auto-detected from the file name (.gz, .xz, .bz2). Use the file name '-' for\n",
      "standard input/output. Without the -o option, output is sent to standard output.\n",
      "\n",
      "Citation:\n",
      "\n",
      "Marcel Martin. Cutadapt removes adapter sequences from high-throughput\n",
      "sequencing reads. EMBnet.Journal, 17(1):10-12, May 2011.\n",
      "http://dx.doi.org/10.14806/ej.17.1.200\n",
      "\n",
      "Use \"cutadapt --help\" to see all command-line options.\n",
      "See http://cutadapt.readthedocs.io/ for full documentation.\n",
      "\n",
      "cutadapt: error: [Errno 2] No such file or directory: '/Volumes/Serine/wd/18-04-10/zr2096_5_s1_R2.fastq'\n",
      "\n",
      "\n",
      "Cutadapt terminated with exit signal: '512'.\n",
      "Terminating Trim Galore run, please check error message(s) to get an idea what went wrong...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/trim_galore_zip/trim_galore \\\n",
    "--clip_R1 15 \\\n",
    "--clip_R2 15 \\\n",
    "--paired \\\n",
    "--output_dir /Volumes/Serine/wd/18-04-10/ \\\n",
    "/Volumes/Serine/wd/18-04-10/zr2096_5_s1_R1.fastq \\\n",
    "/Volumes/Serine/wd/18-04-10/zr2096_5_s1_R2.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# new version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "/Applications/bioinfo/TrimGalore-0.4.5/trim_galore"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
