{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--non_directional \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-1 /Volumes/Serine/wd/18-03-16/10_32_S32_L001_R1_001.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-03-16/10_32_S32_L001_R2_001.fastq.gz \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10\n",
      "zr2096_1\n",
      "zr2096_2\n",
      "zr2096_3\n",
      "zr2096_4\n",
      "zr2096_5\n",
      "zr2096_6\n",
      "zr2096_7\n",
      "zr2096_8\n",
      "zr2096_9\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-04-07/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!mkdir /Volumes/Serine/wd/18-04-07/bm01\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-07/bm01\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-04-07/bm01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3229397\n",
      "Mapping efficiency:\t18.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t13321198\n",
      "Sequence pairs did not map uniquely:\t1166532\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1611649\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1617748\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2033691\n",
      "Mapping efficiency:\t7.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t26209079\n",
      "Sequence pairs did not map uniquely:\t739996\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1014935\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1018754\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5003424\n",
      "Mapping efficiency:\t16.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t23887292\n",
      "Sequence pairs did not map uniquely:\t1907866\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2496390\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2507034\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5310557\n",
      "Mapping efficiency:\t17.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t22495650\n",
      "Sequence pairs did not map uniquely:\t2085795\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2649044\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2661513\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4516691\n",
      "Mapping efficiency:\t18.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t18136564\n",
      "Sequence pairs did not map uniquely:\t1688713\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2251307\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2265382\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5499890\n",
      "Mapping efficiency:\t17.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t24300069\n",
      "Sequence pairs did not map uniquely:\t1978756\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2749961\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2749928\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4541444\n",
      "Mapping efficiency:\t18.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t17961401\n",
      "Sequence pairs did not map uniquely:\t1734445\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2266667\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2274777\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5038566\n",
      "Mapping efficiency:\t17.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t22726177\n",
      "Sequence pairs did not map uniquely:\t1770003\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2514606\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2523958\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4646143\n",
      "Mapping efficiency:\t15.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t23487537\n",
      "Sequence pairs did not map uniquely:\t1628157\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2325979\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2320162\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5644532\n",
      "Mapping efficiency:\t17.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t24857593\n",
      "Sequence pairs did not map uniquely:\t2134106\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2816231\t((converted) top strand)\n",
      "GA/CT/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2828300\t((converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_10_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_10_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_10_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_10_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_10_s1_R1.fastq.gz to zr2096_10_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_10_s1_R1.fastq.gz (17717127 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_10_s1_R2.fastq.gz to zr2096_10_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_10_s1_R2.fastq.gz (17717127 sequences in total)\n",
      "\n",
      "Input files are zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG/1\t77\t*\t0\t0\t*\t*\t0\t0\tGTTATATATAATTGTATTAATTATAAAAAATATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATTTATTATTAAAAATATTAAATATATTATTTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNATTTTATATTTTTAATAATAAATATAATTATATATTAAAATAAAATAATATATTTAATATTTTTAATAATAAATACAATTAAATATCATAATAA\t#####BB<FFFFFFFFFFFFFBFFFFFFFFFFFFFFFFBFFBFFFFFFFFFBFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1232:2155_1:N:0:GATCAG/1\t77\t*\t0\t0\t*\t*\t0\t0\tGTTATATATAATTGTATTAATTATAAAAAATATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNATTTATTATTAAAAATATTAAATATATTATTTTA\tBBBBBFFFFFFFFFFFFFFBFFFFFFFFFFFFF#################################<</<FFFFFFFFFFFFFFFFFFFFFFFFFF<FFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1232:2155_2:N:0:GATCAG/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNATTTTATATTTTTAATAATAAATATAATTATATATTAAAATAAAATAATATATTTAATATTTTTAATAATAAATACAATTAAATATCATAATAA\t#####BB<FFFFFFFFFFFFFBFFFFFFFFFFFFFFFFBFFBFFFFFFFFFBFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_10_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_10_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_10_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "17717127 reads; of these:\n",
      "  17717127 (100.00%) were paired; of these:\n",
      "    14747848 (83.24%) aligned concordantly 0 times\n",
      "    1470661 (8.30%) aligned concordantly exactly 1 time\n",
      "    1498618 (8.46%) aligned concordantly >1 times\n",
      "16.76% overall alignment rate\n",
      "17717127 reads; of these:\n",
      "  17717127 (100.00%) were paired; of these:\n",
      "    14757625 (83.30%) aligned concordantly 0 times\n",
      "    1460774 (8.24%) aligned concordantly exactly 1 time\n",
      "    1498728 (8.46%) aligned concordantly >1 times\n",
      "16.70% overall alignment rate\n",
      "Processed 17717127 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_10_s1_R1.fastq.gz_C_to_T.fastq and zr2096_10_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t17717127\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t115390413\n",
      "\n",
      "Total methylated C's in CpG context:\t12245061\n",
      "Total methylated C's in CHG context:\t893329\n",
      "Total methylated C's in CHH context:\t3246422\n",
      "Total methylated C's in Unknown context:\t143378\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4475568\n",
      "Total unmethylated C's in CHG context:\t26065245\n",
      "Total unmethylated C's in CHH context:\t68464788\n",
      "Total unmethylated C's in Unknown context:\t445618\n",
      "\n",
      "C methylated in CpG context:\t73.2%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.5%\n",
      "C methylated in unknown context (CN or CHN):\t24.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 1h 20m 30s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_1_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_1_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_1_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_1_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_1_s1_R1.fastq.gz to zr2096_1_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_1_s1_R1.fastq.gz (28982766 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_1_s1_R2.fastq.gz to zr2096_1_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_1_s1_R2.fastq.gz (28982766 sequences in total)\n",
      "\n",
      "Input files are zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT/1\t99\tNC_035789.1_CT_converted\t4531039\t2\t37M4I59M\t=\t4531139\t200\tGAGTTTTTTTGATTATTTGTTGTTTGTTGTTTGTTTGNNNNTNNNTNNGTTTGTTTGTTTGTTTGTTTGTTTGTAAATTTTTTATATTTTTATTTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<##<<<FFFFFFFFFFFFFFFF<FFFFFF/</FFFFFF/B/FFFFF/FFFFFFFF\tAS:i:-22\tXS:i:-31\tXN:i:0\tXM:i:5\tXO:i:1\tXG:i:4\tNM:i:9\tMD:Z:38T0T0G1T0T52\tYS:i:-24\tYT:Z:CP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT/2\t147\tNC_035789.1_CT_converted\t4531139\t2\t100M\t=\t4531039\t-200\tGAATTATTGGGTTAATTTTAATTAAATTTGATATAAAGTATTTTTAGGTGAAGGGGATTTAAGTTTGTTTGAATGAAGGGTTTTGTTTTTTTTTAAGGGG\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-24\tXS:i:-12\tXN:i:0\tXM:i:4\tXO:i:0\tXG:i:0\tNM:i:4\tMD:Z:30G14G9A14A29\tYS:i:-22\tYT:Z:CP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1249:2156_1:N:0:CGATGT/1\t83\tNC_035780.1_GA_converted\t8090288\t6\t26M4D74M\t=\t8090028\t-364\tAAAAAAAATAAAAATATAAAAAATTTACAAACAAACAAACAAACAAACAAACNNANNNANNNNCAAACAAACAACAAACAACAAATAATCAAAAAAACTC\tFFFFFFFF/FFFFF/B/FFFFFF/</FFFFFF<FFFFFFFFFFFFFFFF<<<##<###<####FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBBBBB\tAS:i:-26\tXS:i:-9\tXN:i:0\tXM:i:9\tXO:i:1\tXG:i:4\tNM:i:13\tMD:Z:26^ACAA26A0A1C0A0A1C0A0A0A37\tYS:i:-6\tYT:Z:CP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1249:2156_2:N:0:CGATGT/2\t163\tNC_035780.1_GA_converted\t8090028\t6\t100M\t=\t8090288\t364\tCCCCTTAAAAAAAAACAAAACCCTTCATTCAAACAAACTTAAATCCCCTTCACCTAAAAATACTTTATATCAAATTTAATTAAAATTAACCCAATAATTC\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:-6\tXS:i:-18\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:29T70\tYS:i:-26\tYT:Z:CP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_1_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_1_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_1_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1109:8558:78841_1:N:0:CGATGT\tNC_035780.1\t2\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1309:7531:24389_1:N:0:CGATGT\tNC_035781.1\t2\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "28982766 reads; of these:\n",
      "  28982766 (100.00%) were paired; of these:\n",
      "    27108904 (93.53%) aligned concordantly 0 times\n",
      "    922592 (3.18%) aligned concordantly exactly 1 time\n",
      "    951270 (3.28%) aligned concordantly >1 times\n",
      "6.47% overall alignment rate\n",
      "28982766 reads; of these:\n",
      "  28982766 (100.00%) were paired; of these:\n",
      "    27112037 (93.55%) aligned concordantly 0 times\n",
      "    917899 (3.17%) aligned concordantly exactly 1 time\n",
      "    952830 (3.29%) aligned concordantly >1 times\n",
      "6.45% overall alignment rate\n",
      "Processed 28982766 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_1_s1_R1.fastq.gz_C_to_T.fastq and zr2096_1_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t28982766\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t71179750\n",
      "\n",
      "Total methylated C's in CpG context:\t7169367\n",
      "Total methylated C's in CHG context:\t509609\n",
      "Total methylated C's in CHH context:\t2063668\n",
      "Total methylated C's in Unknown context:\t92234\n",
      "\n",
      "Total unmethylated C's in CpG context:\t2455710\n",
      "Total unmethylated C's in CHG context:\t15623504\n",
      "Total unmethylated C's in CHH context:\t43357892\n",
      "Total unmethylated C's in Unknown context:\t305784\n",
      "\n",
      "C methylated in CpG context:\t74.5%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t4.5%\n",
      "C methylated in unknown context (CN or CHN):\t23.2%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 1h 20m 23s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_2_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_2_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_2_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_2_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_2_s1_R1.fastq.gz to zr2096_2_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_2_s1_R1.fastq.gz (30798582 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_2_s1_R2.fastq.gz to zr2096_2_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_2_s1_R2.fastq.gz (30798582 sequences in total)\n",
      "\n",
      "Input files are zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTTTTATTAAAATTTATTATTTTAAAAATAAAAATTNNNNTNNNTTTTATTATTTTTTTTAATATTTTTTAATATAATTAAATAATAAAAAAAATTAAT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA/2\t141\t*\t0\t0\t*\t*\t0\t0\tTCACTTCAAAAAAATAATAAATAAAAAAAATTAATACAACAATATTTTTAATAATACAACAATATTTAAAAAATTATTTATTAATTTTTTTTATTATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFBFF/BFF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1148:2157_1:N:0:TGACCA/1\t77\t*\t0\t0\t*\t*\t0\t0\tTTTTTTATTAAAATTTATTATTTTAAAAATAAAAATTNNNNTNNNTTTTATTATTTTTTTTAATATTTTTTAATATAATTAAATAATAAAAAAAATTAAT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF####<###<<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1148:2157_2:N:0:TGACCA/2\t141\t*\t0\t0\t*\t*\t0\t0\tTCACTTCAAAAAAATAATAAATAAAAAAAATTAATACAACAATATTTTTAATAATACAACAATATTTAAAAAATTATTTATTAATTTTTTTTATTATTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFFFFBFFFFFFFFBFF/BFF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_2_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_2_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_2_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "Processed 30000000 sequence pairs so far\n",
      "30798582 reads; of these:\n",
      "  30798582 (100.00%) were paired; of these:\n",
      "    26162568 (84.95%) aligned concordantly 0 times\n",
      "    2337766 (7.59%) aligned concordantly exactly 1 time\n",
      "    2298248 (7.46%) aligned concordantly >1 times\n",
      "15.05% overall alignment rate\n",
      "30798582 reads; of these:\n",
      "  30798582 (100.00%) were paired; of these:\n",
      "    26171347 (84.98%) aligned concordantly 0 times\n",
      "    2324622 (7.55%) aligned concordantly exactly 1 time\n",
      "    2302613 (7.48%) aligned concordantly >1 times\n",
      "15.02% overall alignment rate\n",
      "Processed 30798582 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_2_s1_R1.fastq.gz_C_to_T.fastq and zr2096_2_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t30798582\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t166030240\n",
      "\n",
      "Total methylated C's in CpG context:\t17625283\n",
      "Total methylated C's in CHG context:\t1346374\n",
      "Total methylated C's in CHH context:\t5725619\n",
      "Total methylated C's in Unknown context:\t225266\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5879632\n",
      "Total unmethylated C's in CHG context:\t38361849\n",
      "Total unmethylated C's in CHH context:\t97091483\n",
      "Total unmethylated C's in Unknown context:\t693976\n",
      "\n",
      "C methylated in CpG context:\t75.0%\n",
      "C methylated in CHG context:\t3.4%\n",
      "C methylated in CHH context:\t5.6%\n",
      "C methylated in unknown context (CN or CHN):\t24.5%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 2h 0m 2s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_3_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_3_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_3_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_3_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_3_s1_R1.fastq.gz to zr2096_3_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_3_s1_R1.fastq.gz (29892002 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_3_s1_R2.fastq.gz to zr2096_3_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_3_s1_R2.fastq.gz (29892002 sequences in total)\n",
      "\n",
      "Input files are zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG/1\t77\t*\t0\t0\t*\t*\t0\t0\tTNTTTTAAAATAATTTAAATAATTTTATATAAATNNNNNNNNNNNNNNNTATTTAAATTAAAAATTTAAATAAATTTTTTTAATTATTTATTGTTTAATG\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNAAATAAACATTAAACAATAAATAATTAAAAAAATTTATTTAAATTTTTAATTTAAATAAATTAAAAATATTAAATTTATATAAAATTATTTAAAT\t####<BBFFFFFFFFFFFFF<FFFFFF<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFB\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1059:2153_1:N:0:ACAGTG/1\t77\t*\t0\t0\t*\t*\t0\t0\tTNTTTTAAAATAATTTAAATAATTTTATATAAATNNNNNNNNNNNNNNNTATTTAAATTAAAAATTTAAATAAATTTTTTTAATTATTTATTGTTTAATG\tB#<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFF###############<<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1059:2153_2:N:0:ACAGTG/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNAAATAAACATTAAACAATAAATAATTAAAAAAATTTATTTAAATTTTTAATTTAAATAAATTAAAAATATTAAATTTATATAAAATTATTTAAAT\t####<BBFFFFFFFFFFFFF<FFFFFF<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFB\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_3_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_3_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_3_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "29892002 reads; of these:\n",
      "  29892002 (100.00%)2 9w8e9r2e0 0p2ai red; of these:\n",
      "    24805854 (r82.e9a8%) alignedd s; of thconcordantly 0ese:\n",
      "   tim29892002 (es\n",
      "1 0   2396064 (8.00.00%) were paired; of these:\n",
      "    24824535 (83.2%) al0i5%) agned concordantly exactly 1 time\n",
      "    ligned concordantly 0 times\n",
      "    2374860 (7.94%) aligned concordantly exactly 1 time\n",
      "    2692607 (9.01%) aligned concordantly >1 times\n",
      "16.95% overall alignment rate\n",
      "2690084 (9.00%) aligned concordantly >1 times\n",
      "17.02% overall alignment rate\n",
      "Processed 29892002 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_3_s1_R1.fastq.gz_C_to_T.fastq and zr2096_3_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t29892002\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t189484349\n",
      "\n",
      "Total methylated C's in CpG context:\t22776539\n",
      "Total methylated C's in CHG context:\t1476555\n",
      "Total methylated C's in CHH context:\t5479846\n",
      "Total methylated C's in Unknown context:\t243262\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5703037\n",
      "Total unmethylated C's in CHG context:\t42937976\n",
      "Total unmethylated C's in CHH context:\t111110396\n",
      "Total unmethylated C's in Unknown context:\t758691\n",
      "\n",
      "C methylated in CpG context:\t80.0%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.7%\n",
      "C methylated in unknown context (CN or CHN):\t24.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 2h 18m 46s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_4_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_4_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_4_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_4_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_4_s1_R1.fastq.gz to zr2096_4_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_4_s1_R1.fastq.gz (24341968 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_4_s1_R2.fastq.gz to zr2096_4_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_4_s1_R2.fastq.gz (24341968 sequences in total)\n",
      "\n",
      "Input files are zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT/1\t77\t*\t0\t0\t*\t*\t0\t0\tNGATTAGTGTATAGTGGTTGTTATGGAAATTAGNNNNNNNNNNNNNNNNNNNNNNNNNTGNANTNTTTTTATTAGAATTGTTAAGGTTTATTTGATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNTCCATCATTAATTTTTAAAAAAAAAAAACCATCCTTAATATCAATTTTATAAATCAAATAAACCTTAACAATTCTAATAAAAACACTTCATATAT\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFF/BBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1343:2133_1:N:0:GCCAAT/1\t77\t*\t0\t0\t*\t*\t0\t0\tNGATTAGTGTATAGTGGTTGTTATGGAAATTAGNNNNNNNNNNNNNNNNNNNNNNNNNTGNANTNTTTTTATTAGAATTGTTAAGGTTTATTTGATTTAT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF#########################<<#<#/#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1343:2133_2:N:0:GCCAAT/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNTCCATCATTAATTTTTAAAAAAAAAAAACCATCCTTAATATCAATTTTATAAATCAAATAAACCTTAACAATTCTAATAAAAACACTTCATATAT\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFFF/BBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_4_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_4_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_4_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:2213:19455:71351_1:N:0:GCCAAT\tNC_035780.1\t2\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:2310:6331:40757_1:N:0:GCCAAT\tNC_035780.1\t2\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "24341968 reads; of these:\n",
      "  24341968 (100.00%) were paired; of these:\n",
      "    20185330 (82.92%) aligned concordantly 0 times\n",
      "    2086737 (8.57%) aligned concordantly exactly 1 time\n",
      "    2069901 (8.50%) aligned concordantly >1 times\n",
      "17.08% overall alignment rate\n",
      "24341968 reads; of these:\n",
      "  24341968 (100.00%) were paired; of these:\n",
      "    20177991 (82.89%) aligned concordantly 0 times\n",
      "    2097974 (8.62%) aligned concordantly exactly 1 time\n",
      "    2066003 (8.49%) aligned concordantly >1 times\n",
      "17.11% overall alignment rate\n",
      "Processed 24341968 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_4_s1_R1.fastq.gz_C_to_T.fastq and zr2096_4_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t24341968\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t148447494\n",
      "\n",
      "Total methylated C's in CpG context:\t14022606\n",
      "Total methylated C's in CHG context:\t1147944\n",
      "Total methylated C's in CHH context:\t4897777\n",
      "Total methylated C's in Unknown context:\t205137\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5922512\n",
      "Total unmethylated C's in CHG context:\t33264805\n",
      "Total unmethylated C's in CHH context:\t89191850\n",
      "Total unmethylated C's in Unknown context:\t643585\n",
      "\n",
      "C methylated in CpG context:\t70.3%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t5.2%\n",
      "C methylated in unknown context (CN or CHN):\t24.2%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 1h 44m 39s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_5_s1_R1.fastq.gz to zr2096_5_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_5_s1_R1.fastq.gz (31778715 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_5_s1_R2.fastq.gz to zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_5_s1_R2.fastq.gz (31778715 sequences in total)\n",
      "\n",
      "Input files are zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_5_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:2103:13206:55080_1:N:0:CAGATC\tNC_035780.1\t2\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "Processed 30000000 sequence pairs so far\n",
      "Processed 31000000 sequence pairs so far\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26752546 (84.18%) aligned concordantly 0 times\n",
      "    2514975 (7.91%) aligned concordantly exactly 1 time\n",
      "    2511194 (7.90%) aligned concordantly >1 times\n",
      "15.82% overall alignment rate\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26745514 (84.16%) aligned concordantly 0 times\n",
      "    2511973 (7.90%) aligned concordantly exactly 1 time\n",
      "    2521228 (7.93%) aligned concordantly >1 times\n",
      "15.84% overall alignment rate\n",
      "Processed 31778715 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t31778715\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t189151053\n",
      "\n",
      "Total methylated C's in CpG context:\t17864954\n",
      "Total methylated C's in CHG context:\t1429272\n",
      "Total methylated C's in CHH context:\t5654793\n",
      "Total methylated C's in Unknown context:\t253526\n",
      "\n",
      "Total unmethylated C's in CpG context:\t7027555\n",
      "Total unmethylated C's in CHG context:\t41425253\n",
      "Total unmethylated C's in CHH context:\t115749226\n",
      "Total unmethylated C's in Unknown context:\t809323\n",
      "\n",
      "C methylated in CpG context:\t71.8%\n",
      "C methylated in CHG context:\t3.3%\n",
      "C methylated in CHH context:\t4.7%\n",
      "C methylated in unknown context (CN or CHN):\t23.9%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 2h 14m 53s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_6_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_6_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_6_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_6_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_6_s1_R1.fastq.gz to zr2096_6_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_6_s1_R1.fastq.gz (24237290 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_6_s1_R2.fastq.gz to zr2096_6_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_6_s1_R2.fastq.gz (24237290 sequences in total)\n",
      "\n",
      "Input files are zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA/1\t77\t*\t0\t0\t*\t*\t0\t0\tNTTATTAAATTTAATTTTATTATTAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNTANANAAATATTAATTAAATATTTTTAAAATAATAAATTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNTAAAAATTTTTTTAAATTTATTATTTTAAAAATATTTAATTAATATTTATTTATATTTTATAATTAACAATAATTATTTTTTTATTTAATAATAA\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFBFFFFBFFFFB/\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1093:2134_1:N:0:CTTGTA/1\t77\t*\t0\t0\t*\t*\t0\t0\tNTTATTAAATTTAATTTTATTATTAAATAAAAANNNNNNNNNNNNNNNNNNNNNNANNTANANAAATATTAATTAAATATTTTTAAAATAATAAATTTAA\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF######################<##<<#<#<<FFFFFFFFBFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1093:2134_2:N:0:CTTGTA/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNTAAAAATTTTTTTAAATTTATTATTTTAAAAATATTTAATTAATATTTATTTATATTTTATAATTAACAATAATTATTTTTTTATTTAATAATAA\t#####BBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFBFFFFFFFFFFFFFFBFFFFBFFFFB/\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_6_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_6_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_6_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "24237290 reads; of these:\n",
      "  24237290 (100.00%) were paired; of these:\n",
      "    20001750 (82.52%) aligned concordantly 0 times\n",
      "    2095964 (8.65%) aligned concordantly exactly 1 time\n",
      "    2139576 (8.83%) aligned concordantly >1 times\n",
      "17.48% overall alignment rate\n",
      "24237290 reads; of these:\n",
      "  24237290 (100.00%) were paired; of these:\n",
      "    20006524 (82.54%) aligned concordantly 0 times\n",
      "    2087321 (8.61%) aligned concordantly exactly 1 time\n",
      "    2143445 (8.84%) aligned concordantly >1 times\n",
      "17.46% overall alignment rate\n",
      "Processed 24237290 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_6_s1_R1.fastq.gz_C_to_T.fastq and zr2096_6_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t24237290\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t151279876\n",
      "\n",
      "Total methylated C's in CpG context:\t16106095\n",
      "Total methylated C's in CHG context:\t1186234\n",
      "Total methylated C's in CHH context:\t4868728\n",
      "Total methylated C's in Unknown context:\t221460\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4666662\n",
      "Total unmethylated C's in CHG context:\t33709092\n",
      "Total unmethylated C's in CHH context:\t90743065\n",
      "Total unmethylated C's in Unknown context:\t672803\n",
      "\n",
      "C methylated in CpG context:\t77.5%\n",
      "C methylated in CHG context:\t3.4%\n",
      "C methylated in CHH context:\t5.1%\n",
      "C methylated in unknown context (CN or CHN):\t24.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 1h 45m 38s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_7_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_7_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_7_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_7_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_7_s1_R1.fastq.gz to zr2096_7_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_7_s1_R1.fastq.gz (29534746 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_7_s1_R2.fastq.gz to zr2096_7_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_7_s1_R2.fastq.gz (29534746 sequences in total)\n",
      "\n",
      "Input files are zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG/1\t77\t*\t0\t0\t*\t*\t0\t0\tAAAAAAATGATTATTATGATGAATTTAAATATTAATAATAAAATTTAAAAAAAATATAAATGTTAAATATTAAATTTTTTAAAAAATGATTATGG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG/2\t141\t*\t0\t0\t*\t*\t0\t0\tCCATAATCATTTTTTAAAAAATTTAATATTTAACATTTATATTTTTTTTAAATTTTATTATTAATATTTAAATTCATCATAATAATCATTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/FFBBFFFFFFBBFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1132:2189_1:N:0:ATCACG/1\t77\t*\t0\t0\t*\t*\t0\t0\tAAAAAAATGATTATTATGATGAATTTAAATATTAATAATAAAATTTAAAAAAAATATAAATGTTAAATATTAAATTTTTTAAAAAATGATTATGG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1132:2189_2:N:0:ATCACG/2\t141\t*\t0\t0\t*\t*\t0\t0\tCCATAATCATTTTTTAAAAAATTTAATATTTAACATTTATATTTTTTTTAAATTTTATTATTAATATTTAAATTCATCATAATAATCATTTTTTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF/FFBBFFFFFFBBFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_7_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_7_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_7_s1_R2.fastq.gz\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1103:1105:80383_1:N:0:ATCACG\tNC_007175.2\t17150\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1302:9910:66096_1:N:0:ATCACG\tNC_007175.2\t17146\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "29534746 reads; of these:\n",
      "  29534746 (100.00%) were paired; of these:\n",
      "    24968913 (84.54%) aligned concordantly 0 times\n",
      "    2301909 (7.79%) aligned concordantly exactly 1 time\n",
      "    2263924 (7.67%) aligned concordantly >1 times\n",
      "15.46% overall alignment rate\n",
      "29534746 reads; of these:\n",
      "  29534746 (100.00%) were paired; of these:\n",
      "    24981202 (84.58%) aligned concordantly 0 times\n",
      "    2293473 (7.77%) aligned concordantly exactly 1 time\n",
      "    2260071 (7.65%) aligned concordantly >1 times\n",
      "15.42% overall alignment rate\n",
      "Processed 29534746 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_7_s1_R1.fastq.gz_C_to_T.fastq and zr2096_7_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t29534746\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t175381402\n",
      "\n",
      "Total methylated C's in CpG context:\t17526556\n",
      "Total methylated C's in CHG context:\t1367591\n",
      "Total methylated C's in CHH context:\t5269964\n",
      "Total methylated C's in Unknown context:\t227515\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5641309\n",
      "Total unmethylated C's in CHG context:\t38963896\n",
      "Total unmethylated C's in CHH context:\t106612086\n",
      "Total unmethylated C's in Unknown context:\t735121\n",
      "\n",
      "C methylated in CpG context:\t75.7%\n",
      "C methylated in CHG context:\t3.4%\n",
      "C methylated in CHH context:\t4.7%\n",
      "C methylated in unknown context (CN or CHN):\t23.6%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 2h 6m 34s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_8_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_8_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_8_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_8_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_8_s1_R1.fastq.gz to zr2096_8_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_8_s1_R1.fastq.gz (29761837 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_8_s1_R2.fastq.gz to zr2096_8_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_8_s1_R2.fastq.gz (29761837 sequences in total)\n",
      "\n",
      "Input files are zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC/1\t77\t*\t0\t0\t*\t*\t0\t0\tTAATTTAAAATTAAAAAATTTATTAAATTTATTATTATTTNTTATTTTTTATAAAAGAAAATAGAGAGAGAGAGAGAGAGAGAGAGAG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCTCTCTCTCTCTCTCTCTCTCTCTCTATTTTCTTTTATAAAAAATAATAAATAATAATAAATTTAATAAATTTTTTAATTTTAAATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1281:2170_1:N:0:TTAGGC/1\t77\t*\t0\t0\t*\t*\t0\t0\tTAATTTAAAATTAAAAAATTTATTAAATTTATTATTATTTNTTATTTTTTATAAAAGAAAATAGAGAGAGAGAGAGAGAGAGAGAGAG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<FFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1281:2170_2:N:0:TTAGGC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCTCTCTCTCTCTCTCTCTCTCTCTCTATTTTCTTTTATAAAAAATAATAAATAATAATAAATTTAATAAATTTTTTAATTTTAAATTA\tBBBBBFFFFFFFFFFFFFFFFFFFFF/FFFF<FFFFBF</F</<<<<FF/<FF/<<</F//<FF</BFF/FFFFFF//<FFFB//BF/\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_8_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_8_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_8_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1303:16780:7528_1:N:0:TTAGGC\tNC_007175.2\t17147\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1315:20840:40040_1:N:0:TTAGGC\tNC_007175.2\t17163\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "29761837 reads; of these:\n",
      "  29761837 (100.00%) were paired; of these:\n",
      "    25580112 (85.95%) aligned concordantly 0 times\n",
      "    2143425 (7.20%) aligned concordantly exactly 1 time\n",
      "    2038300 (6.85%) aligned concordantly >1 times\n",
      "14.05% overall alignment rate\n",
      "29761837 reads; of these:\n",
      "  29761837 (100.00%) were paired; of these:\n",
      "    25587746 (85.98%) aligned concordantly 0 times\n",
      "    2136784 (7.18%) aligned concordantly exactly 1 time\n",
      "    2037307 (6.85%) aligned concordantly >1 times\n",
      "14.02% overall alignment rate\n",
      "Processed 29761837 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_8_s1_R1.fastq.gz_C_to_T.fastq and zr2096_8_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t29761837\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t157822868\n",
      "\n",
      "Total methylated C's in CpG context:\t14123725\n",
      "Total methylated C's in CHG context:\t1232277\n",
      "Total methylated C's in CHH context:\t5100411\n",
      "Total methylated C's in Unknown context:\t204421\n",
      "\n",
      "Total unmethylated C's in CpG context:\t5895006\n",
      "Total unmethylated C's in CHG context:\t35229169\n",
      "Total unmethylated C's in CHH context:\t96242280\n",
      "Total unmethylated C's in Unknown context:\t663867\n",
      "\n",
      "C methylated in CpG context:\t70.6%\n",
      "C methylated in CHG context:\t3.4%\n",
      "C methylated in CHH context:\t5.0%\n",
      "C methylated in unknown context (CN or CHN):\t23.5%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 1h 58m 10s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n",
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-07/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_9_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_9_s1_R2.fastq.gz\n",
      "Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-07/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_9_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_9_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_9_s1_R1.fastq.gz to zr2096_9_s1_R1.fastq.gz_C_to_T.fastq\n",
      "\n",
      "Created C -> T converted version of the FastQ file zr2096_9_s1_R1.fastq.gz (32636231 sequences in total)\n",
      "\n",
      "Writing a G -> A converted version of the input file zr2096_9_s1_R2.fastq.gz to zr2096_9_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created G -> A converted version of the FastQ file zr2096_9_s1_R2.fastq.gz (32636231 sequences in total)\n",
      "\n",
      "Input files are zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 2 instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA/1\t77\t*\t0\t0\t*\t*\t0\t0\tNATTAAATAAAAAATAATAAATAATAAATTANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAGG\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAAAAAATAAAAAACACATTAAAAATCTATAATTTAATTTATTATTTATTATTTTTTATTTAATTAAATCNNNNAAACATCATATAAAAAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1135:2128_1:N:0:ACTTGA/1\t77\t*\t0\t0\t*\t*\t0\t0\tNATTAAATAAAAAATAATAAATAATAAATTANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTATAGG\t#/<<</<<B<<B<FF<BBFFBFFFF<B/FFF####################################<<<<FBF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:2:1101:1135:2128_2:N:0:ACTTGA/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNNNAANTAAAAAAATAAAAAACACATTAAAAATCTATAATTTAATTTATTATTTATTATTTTTTATTTAATTAAATCNNNNAAACATCATATAAAAAA\t#####<<#</BFFFB/FBFBF<FFFFF/FBFF//<FFF/</<FFF//</<<B/FB//FB/<FFFFF/B/F/FFFFB///####<</FFF<FBFF/F</7B\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_9_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_9_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_9_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:2:1214:4921:70468_1:N:0:ACTTGA\tNC_007175.2\t17151\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "Processed 30000000 sequence pairs so far\n",
      "Processed 31000000 sequence pairs so far\n",
      "Processed 32000000 sequence pairs so far\n",
      "32636231 reads; of these:\n",
      "  32636231 (100.00%) were paired; of these:\n",
      "    27375708 (83.88%) aligned concordantly 0 times\n",
      "    2603810 (7.98%) aligned concordantly exactly 1 time\n",
      "    2656713 (8.14%) aligned concordantly >1 times\n",
      "16.12% overall alignment rate\n",
      "32636231 reads; of these:\n",
      "  32636231 (100.00%) were paired; of these:\n",
      "    27380120 (83.89%) aligned concordantly 0 times\n",
      "    2587080 (7.93%) aligned concordantly exactly 1 time\n",
      "    2669031 (8.18%) aligned concordantly >1 times\n",
      "16.11% overall alignment rate\n",
      "Processed 32636231 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_9_s1_R1.fastq.gz_C_to_T.fastq and zr2096_9_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t32636231\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t203684656\n",
      "\n",
      "Total methylated C's in CpG context:\t23798172\n",
      "Total methylated C's in CHG context:\t1639976\n",
      "Total methylated C's in CHH context:\t5453380\n",
      "Total methylated C's in Unknown context:\t272146\n",
      "\n",
      "Total unmethylated C's in CpG context:\t6006825\n",
      "Total unmethylated C's in CHG context:\t45798164\n",
      "Total unmethylated C's in CHH context:\t120988139\n",
      "Total unmethylated C's in Unknown context:\t826171\n",
      "\n",
      "C methylated in CpG context:\t79.8%\n",
      "C methylated in CHG context:\t3.5%\n",
      "C methylated in CHH context:\t4.3%\n",
      "C methylated in unknown context (CN or CHN):\t24.8%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 2h 19m 33s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-04-07/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1.fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-p 4 \\\n",
    "-1 /Volumes/Serine/wd/18-04-07/{}_s1_R1.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-04-07/{}_s1_R2.fastq.gz  \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat /Volumes/Serine/wd/18-04-07/bm01/*PE_report.txt | grep \"Mapping Efficiency\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-10/bm01\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-04-10/bm01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t10617999\n",
      "Mapping efficiency:\t33.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t17366736\n",
      "Sequence pairs did not map uniquely:\t3793980\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t3\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2702790\t((converted) top strand)\n",
      "GA/CT/CT:\t2609389\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t2601979\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2703838\t((converted) bottom strand)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-10/bm01'):\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz\n",
      "/Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Library was specified to be not strand-specific (non-directional), therefore alignments to all four possible bisulfite strands (OT, CTOT, OB and CTOB) will be reported\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-10/bm01\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file zr2096_5_s1_R1.fastq.gz to zr2096_5_s1_R1.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_5_s1_R1.fastq.gz to zr2096_5_s1_R1.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_5_s1_R1.fastq.gz (31778715 sequences in total)\n",
      "\n",
      "Writing a C -> T converted version of the input file zr2096_5_s1_R2.fastq.gz to zr2096_5_s1_R2.fastq.gz_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file zr2096_5_s1_R2.fastq.gz to zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file zr2096_5_s1_R2.fastq.gz (31778715 sequences in total)\n",
      "\n",
      "Input files are zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R1.fastq.gz_G_to_A.fastq and zr2096_5_s1_R2.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq (FastQ)\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2GAgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_G_to_A.fastq and zr2096_5_s1_R2.fastq.gz_C_to_T.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAACAAAACACCACTCAAATCTTACTATCCACANNNNNNNNNNNNNNNNNNATTNCTTCTATCATATATATAAAATCATTCCAAAATTACTATATTTTTC\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTTGAGTATATTGAAGAGGAATTATTGATAGTTTTTGTTGAAAAATATAGTAATTTTGGAATGATTTTATATATATGATAGAAGAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2CTgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_G_to_A.fastq and zr2096_5_s1_R2.fastq.gz_C_to_T.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAACAAAACACCACTCAAATCTTACTATCCACANNNNNNNNNNNNNNNNNNATTNCTTCTATCATATATATAAAATCATTCCAAAATTACTATATTTTTC\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTTGAGTATATTGAAGAGGAATTATTGATAGTTTTTGTTGAAAAATATAGTAATTTTGGAATGATTTTATATATATGATAGAAGAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from zr2096_5_s1_R1.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tNAATAAAATATTATTTAAATTTTATTATTTATANNNNNNNNNNNNNNNNNNATTNTTTTTATTATATATATAAAATTATTTTAAAATTATTATATTTTTT\t#<<BBFFFFFFFFFFFFFFFFFFFFFFFFFFFF##################<<<#<<FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\tYF:Z:NS\n",
      "HWI-C00124:321:CC781ANXX:1:1101:1218:2142_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tNNNTCAAATATATTAAAAAAAAATTATTAATAATTTTCATTAAAAAATATAATAATTTTAAAATAATTTTATATATATAATAAAAAAAATAATAATAAT\t###<<BFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFB\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to zr2096_5_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz and /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz\n",
      "Processed 1000000 sequence pairs so far\n",
      "Processed 2000000 sequence pairs so far\n",
      "Processed 3000000 sequence pairs so far\n",
      "Processed 4000000 sequence pairs so far\n",
      "Processed 5000000 sequence pairs so far\n",
      "Processed 6000000 sequence pairs so far\n",
      "Processed 7000000 sequence pairs so far\n",
      "Processed 8000000 sequence pairs so far\n",
      "Processed 9000000 sequence pairs so far\n",
      "Processed 10000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1216:10289:18419_1:N:0:CAGATC\tNC_035780.1\t3\n",
      "Processed 11000000 sequence pairs so far\n",
      "Processed 12000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:1305:7699:42587_1:N:0:CAGATC\tNC_007175.2\t2\n",
      "Processed 13000000 sequence pairs so far\n",
      "Processed 14000000 sequence pairs so far\n",
      "Processed 15000000 sequence pairs so far\n",
      "Processed 16000000 sequence pairs so far\n",
      "Chromosomal sequence could not be extracted for\tHWI-C00124:321:CC781ANXX:1:2103:13206:55080_1:N:0:CAGATC\tNC_035780.1\t2\n",
      "Processed 17000000 sequence pairs so far\n",
      "Processed 18000000 sequence pairs so far\n",
      "Processed 19000000 sequence pairs so far\n",
      "Processed 20000000 sequence pairs so far\n",
      "Processed 21000000 sequence pairs so far\n",
      "Processed 22000000 sequence pairs so far\n",
      "Processed 23000000 sequence pairs so far\n",
      "Processed 24000000 sequence pairs so far\n",
      "Processed 25000000 sequence pairs so far\n",
      "Processed 26000000 sequence pairs so far\n",
      "Processed 27000000 sequence pairs so far\n",
      "Processed 28000000 sequence pairs so far\n",
      "Processed 29000000 sequence pairs so far\n",
      "Processed 30000000 sequence pairs so far\n",
      "Processed 31000000 sequence pairs so far\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26933937 (84.75%) aligned concordantly 0 times\n",
      "    2426724 (7.64%) aligned concordantly exactly 1 time\n",
      "    2418054 (7.61%) aligned concordantly >1 times\n",
      "15.25% overall alignment rate\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26948157 (84.80%) aligned concordantly 0 times\n",
      "    2421214 (7.62%) aligned concordantly exactly 1 time\n",
      "    2409344 (7.58%) aligned concordantly >1 times\n",
      "15.20% overall alignment rate\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26745514 (84.16%) aligned concordantly 0 times\n",
      "    2511973 (7.90%) aligned concordantly exactly 1 time\n",
      "    2521228 (7.93%) aligned concordantly >1 times\n",
      "15.84% overall alignment rate\n",
      "31778715 reads; of these:\n",
      "  31778715 (100.00%) were paired; of these:\n",
      "    26752546 (84.18%) aligned concordantly 0 times\n",
      "    2514975 (7.91%) aligned concordantly exactly 1 time\n",
      "    2511194 (7.90%) aligned concordantly >1 times\n",
      "15.82% overall alignment rate\n",
      "Processed 31778715 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files zr2096_5_s1_R1.fastq.gz_C_to_T.fastq, zr2096_5_s1_R1.fastq.gz_G_to_A.fastq, zr2096_5_s1_R2.fastq.gz_C_to_T.fastq and zr2096_5_s1_R2.fastq.gz_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t31778715\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t366537981\n",
      "\n",
      "Total methylated C's in CpG context:\t34486919\n",
      "Total methylated C's in CHG context:\t2210810\n",
      "Total methylated C's in CHH context:\t6230765\n",
      "Total methylated C's in Unknown context:\t438615\n",
      "\n",
      "Total unmethylated C's in CpG context:\t13926155\n",
      "Total unmethylated C's in CHG context:\t82047506\n",
      "Total unmethylated C's in CHH context:\t227635826\n",
      "Total unmethylated C's in Unknown context:\t1526433\n",
      "\n",
      "C methylated in CpG context:\t71.2%\n",
      "C methylated in CHG context:\t2.6%\n",
      "C methylated in CHH context:\t2.7%\n",
      "C methylated in unknown context (CN or CHN):\t22.3%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 4h 34m 27s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R1.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-04-07/zr2096_5_s1_R2.fastq.gz  \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir ../bm02"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-10/bm02\n"
     ]
    }
   ],
   "source": [
    "cd ../bm02/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3342\n",
      "Mapping efficiency:\t33.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5490\n",
      "Sequence pairs did not map uniquely:\t1168\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t856\t((converted) top strand)\n",
      "GA/CT/CT:\t824\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t787\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t875\t((converted) bottom strand)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-10/bm02'):\n",
      "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq\n",
      "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Library was specified to be not strand-specific (non-directional), therefore alignments to all four possible bisulfite strands (OT, CTOT, OB and CTOB) will be reported\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-10/bm02\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq and /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file sub-zr2096_5_s1_R1.fastq to sub-zr2096_5_s1_R1.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file sub-zr2096_5_s1_R1.fastq to sub-zr2096_5_s1_R1.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file sub-zr2096_5_s1_R1.fastq (10000 sequences in total)\n",
      "\n",
      "Writing a C -> T converted version of the input file sub-zr2096_5_s1_R2.fastq to sub-zr2096_5_s1_R2.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file sub-zr2096_5_s1_R2.fastq to sub-zr2096_5_s1_R2.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file sub-zr2096_5_s1_R2.fastq (10000 sequences in total)\n",
      "\n",
      "Input files are sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq (FastQ)\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAATTTTAATTTTTTTTTTTTATAAATAAAAAATAAAAATTATAAATTTATTTAAATTTAATATATAATTATATTATTTATAAAATG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCATTTTATAAATAATATAATTATATATTAAATTTAAATAAATTTATAATTTTTATTTTTTATTTATAAAAAAAAAAAATTAAAATT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2GAgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAACCTTAACCTTTCCCCCCTATAAATAAAAAACAAAAACTATAAATCTACCTAAATCCAACACATAACTACATTATTTATAAAATA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tTATTTTATAAATAATGTAGTTATGTGTTGGATTTAGGTAGATTTATAGTTTTTGTTTTTTATTTATAGGGGGGAAAGGTTAAGGTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2CTgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAACCTTAACCTTTCCCCCCTATAAATAAAAAACAAAAACTATAAATCTACCTAAATCCAACACATAACTACATTATTTATAAAATA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tTATTTTATAAATAATGTAGTTATGTGTTGGATTTAGGTAGATTTATAGTTTTTGTTTTTTATTTATAGGGGGGAAAGGTTAAGGTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.6 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAATTTTAATTTTTTTTTTTTATAAATAAAAAATAAAAATTATAAATTTATTTAAATTTAATATATAATTATATTATTTATAAAATG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCATTTTATAAATAATATAATTATATATTAAATTTAAATAAATTTATAATTTTTATTTTTTATTTATAAAAAAAAAAAATTAAAATT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to sub-zr2096_5_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq and /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    8481 (84.81%) aligned concordantly 0 times\n",
      "    785 (7.85%) aligned concordantly exactly 1 time\n",
      "    734 (7.34%) aligned concordantly >1 times\n",
      "15.19% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    8530 (85.30%) aligned concordantly 0 times\n",
      "    754 (7.54%) aligned concordantly exactly 1 time\n",
      "    716 (7.16%) aligned concordantly >1 times\n",
      "14.70% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    8444 (84.44%) aligned concordantly 0 times\n",
      "    784 (7.84%) aligned concordantly exactly 1 time\n",
      "    772 (7.72%) aligned concordantly >1 times\n",
      "15.56% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    8404 (84.04%) aligned concordantly 0 times\n",
      "    813 (8.13%) aligned concordantly exactly 1 time\n",
      "    783 (7.83%) aligned concordantly >1 times\n",
      "15.96% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files sub-zr2096_5_s1_R1.fastq_C_to_T.fastq, sub-zr2096_5_s1_R1.fastq_G_to_A.fastq, sub-zr2096_5_s1_R2.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t10000\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t115476\n",
      "\n",
      "Total methylated C's in CpG context:\t11099\n",
      "Total methylated C's in CHG context:\t743\n",
      "Total methylated C's in CHH context:\t1984\n",
      "Total methylated C's in Unknown context:\t155\n",
      "\n",
      "Total unmethylated C's in CpG context:\t4412\n",
      "Total unmethylated C's in CHG context:\t25992\n",
      "Total unmethylated C's in CHH context:\t71246\n",
      "Total unmethylated C's in Unknown context:\t500\n",
      "\n",
      "C methylated in CpG context:\t71.6%\n",
      "C methylated in CHG context:\t2.8%\n",
      "C methylated in CHH context:\t2.7%\n",
      "C methylated in unknown context (CN or CHN):\t23.7%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 1m 18s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq \\\n",
    "-2 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq  \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1473\n",
      "Mapping efficiency:\t14.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7887\n",
      "Sequence pairs did not map uniquely:\t640\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t367\t((converted) top strand)\n",
      "GA/CT/CT:\t388\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t347\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t371\t((converted) bottom strand)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bowtie seems to be working fine (tested command '/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2 --version' [2.3.4])\n",
      "Output format is BAM (default)\n",
      "Alignments will be written out in BAM format. Samtools found here: '/Applications/bioinfo/samtools-1.3.1/samtools'\n",
      "Reference genome folder provided is /Volumes/Serine/wd/18-03-15/genome/\t(absolute path is '/Volumes/Serine/wd/18-03-15/genome/)'\n",
      "\n",
      "Input files to be analysed (in current folder '/Volumes/Serine/wd/18-04-10/bm02'):\n",
      "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq\n",
      "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Library was specified to be not strand-specific (non-directional), therefore alignments to all four possible bisulfite strands (OT, CTOT, OB and CTOB) will be reported\n",
      "Setting parallelization to single-threaded (default)\n",
      "\n",
      "Current working directory is: /Volumes/Serine/wd/18-04-10/bm02\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/18-03-15/genome/\n",
      "\n",
      "Single-core mode: setting pid to 1\n",
      "\n",
      "Paired-end alignments will be performed\n",
      "=======================================\n",
      "\n",
      "The provided filenames for paired-end alignments are /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq and /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Input files are in FastQ format\n",
      "Writing a C -> T converted version of the input file sub-zr2096_5_s1_R1.fastq to sub-zr2096_5_s1_R1.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file sub-zr2096_5_s1_R1.fastq to sub-zr2096_5_s1_R1.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file sub-zr2096_5_s1_R1.fastq (10000 sequences in total)\n",
      "\n",
      "Writing a C -> T converted version of the input file sub-zr2096_5_s1_R2.fastq to sub-zr2096_5_s1_R2.fastq_C_to_T.fastq\n",
      "Writing a G -> A converted version of the input file sub-zr2096_5_s1_R2.fastq to sub-zr2096_5_s1_R2.fastq_G_to_A.fastq\n",
      "\n",
      "Created C -> T as well as G -> A converted versions of the FastQ file sub-zr2096_5_s1_R2.fastq (10000 sequences in total)\n",
      "\n",
      "Input files are sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq (FastQ)\n",
      "Now running 4 individual instances of Bowtie 2 against the bisulfite genome of /Volumes/Serine/wd/18-03-15/genome/ with the specified options: -q --score-min L,0,-0.2 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500\n",
      "\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2CTgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.2 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAATTTTAATTTTTTTTTTTTATAAATAAAAAATAAAAATTATAAATTTATTTAAATTTAATATATAATTATATTATTTATAAAATG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCATTTTATAAATAATATAATTATATATTAAATTTAAATAAATTTATAATTTTTATTTTTTATTTATAAAAAAAAAAAATTAAAATT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2GAgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.2 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --norc))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAACCTTAACCTTTCCCCCCTATAAATAAAAAACAAAAACTATAAATCTACCTAAATCCAACACATAACTACATTATTTATAAAATA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tTATTTTATAAATAATGTAGTTATGTGTTGGATTTAGGTAGATTTATAGTTTTTGTTTTTTATTTATAGGGGGGAAAGGTTAAGGTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for GAread1CTread2CTgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_G_to_A.fastq and sub-zr2096_5_s1_R2.fastq_C_to_T.fastq, with the options: -q --score-min L,0,-0.2 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAACCTTAACCTTTCCCCCCTATAAATAAAAAACAAAAACTATAAATCTACCTAAATCCAACACATAACTACATTATTTATAAAATA\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tTATTTTATAAATAATGTAGTTATGTGTTGGATTTAGGTAGATTTATAGTTTTTGTTTTTTATTTATAGGGGGGAAAGGTTAAGGTT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "Now starting a Bowtie 2 paired-end alignment for CTread1GAread2GAgenome (reading in sequences from sub-zr2096_5_s1_R1.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq, with the options: -q --score-min L,0,-0.2 -p 4 --reorder --ignore-quals --no-mixed --no-discordant --dovetail --maxins 500 --nofw))\n",
      "Found first alignment:\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_1:N:0:CAGATC/1\t77\t*\t0\t0\t*\t*\t0\t0\tAATTTTAATTTTTTTTTTTTATAAATAAAAAATAAAAATTATAAATTTATTTAAATTTAATATATAATTATATTATTTATAAAATG\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tYT:Z:UP\n",
      "HWI-C00124:321:CC781ANXX:1:2316:14348:89361_2:N:0:CAGATC/2\t141\t*\t0\t0\t*\t*\t0\t0\tCATTTTATAAATAATATAATTATATATTAAATTTAAATAAATTTATAATTTTTATTTTTTATTTATAAAAAAAAAAAATTAAAATT\tBBBBBFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF<BBFFB<<FFFFBF\tYT:Z:UP\n",
      "\n",
      ">>> Writing bisulfite mapping results to sub-zr2096_5_s1_R1_bismark_bt2_pe.bam <<<\n",
      "\n",
      "\n",
      "Reading in the sequence files /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq and /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    9305 (93.05%) aligned concordantly 0 times\n",
      "    417 (4.17%) aligned concordantly exactly 1 time\n",
      "    278 (2.78%) aligned concordantly >1 times\n",
      "6.95% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    9351 (93.51%) aligned concordantly 0 times\n",
      "    372 (3.72%) aligned concordantly exactly 1 time\n",
      "    277 (2.77%) aligned concordantly >1 times\n",
      "6.49% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    9312 (93.12%) aligned concordantly 0 times\n",
      "    393 (3.93%) aligned concordantly exactly 1 time\n",
      "    295 (2.95%) aligned concordantly >1 times\n",
      "6.88% overall alignment rate\n",
      "10000 reads; of these:\n",
      "  10000 (100.00%) were paired; of these:\n",
      "    9322 (93.22%) aligned concordantly 0 times\n",
      "    386 (3.86%) aligned concordantly exactly 1 time\n",
      "    292 (2.92%) aligned concordantly >1 times\n",
      "6.78% overall alignment rate\n",
      "Processed 10000 sequences in total\n",
      "\n",
      "\n",
      "Successfully deleted the temporary files sub-zr2096_5_s1_R1.fastq_C_to_T.fastq, sub-zr2096_5_s1_R1.fastq_G_to_A.fastq, sub-zr2096_5_s1_R2.fastq_C_to_T.fastq and sub-zr2096_5_s1_R2.fastq_G_to_A.fastq\n",
      "\n",
      "Final Alignment report\n",
      "======================\n",
      "Sequence pairs analysed in total:\t10000\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t52154\n",
      "\n",
      "Total methylated C's in CpG context:\t5400\n",
      "Total methylated C's in CHG context:\t286\n",
      "Total methylated C's in CHH context:\t525\n",
      "Total methylated C's in Unknown context:\t6\n",
      "\n",
      "Total unmethylated C's in CpG context:\t1726\n",
      "Total unmethylated C's in CHG context:\t12076\n",
      "Total unmethylated C's in CHH context:\t32141\n",
      "Total unmethylated C's in Unknown context:\t33\n",
      "\n",
      "C methylated in CpG context:\t75.8%\n",
      "C methylated in CHG context:\t2.3%\n",
      "C methylated in CHH context:\t1.6%\n",
      "C methylated in unknown context (CN or CHN):\t15.4%\n",
      "\n",
      "\n",
      "Bismark completed in 0d 0h 0m 18s\n",
      "\n",
      "====================\n",
      "Bismark run complete\n",
      "====================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq \\\n",
    "-2 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq  \\"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.16\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t10000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t10000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t10000\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count Nextera: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "Writing report to '/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.4_dev\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "\n",
      "Writing final adapter and quality trimmed output to sub-zr2096_5_s1_R1_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq <<< \n",
      "This is cutadapt 1.16 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.25 s (25 us/read; 2.40 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,000\n",
      "Reads with adapters:                     3,691 (36.9%)\n",
      "Reads written (passing filters):        10,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:       907,194 bp\n",
      "Quality-trimmed:                   2,310 bp (0.3%)\n",
      "Total written (filtered):        900,474 bp (99.3%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3691 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 42.6%\n",
      "  C: 13.5%\n",
      "  G: 10.5%\n",
      "  T: 33.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3216\t2500.0\t0\t3216\n",
      "2\t341\t625.0\t0\t341\n",
      "3\t83\t156.2\t0\t83\n",
      "4\t39\t39.1\t0\t39\n",
      "5\t3\t9.8\t0\t3\n",
      "6\t3\t2.4\t0\t3\n",
      "7\t2\t0.6\t0\t2\n",
      "10\t1\t0.0\t1\t0 1\n",
      "16\t1\t0.0\t1\t0 1\n",
      "17\t2\t0.0\t1\t0 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq\n",
      "=============================================\n",
      "10000 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.4_dev\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 15 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "\n",
      "Writing final adapter and quality trimmed output to sub-zr2096_5_s1_R2_trimmed.fq\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq <<< \n",
      "This is cutadapt 1.16 with Python 2.7.12\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 0.19 s (19 us/read; 3.24 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:                  10,000\n",
      "Reads with adapters:                     3,714 (37.1%)\n",
      "Reads written (passing filters):        10,000 (100.0%)\n",
      "\n",
      "Total basepairs processed:       905,579 bp\n",
      "Quality-trimmed:                   3,382 bp (0.4%)\n",
      "Total written (filtered):        897,751 bp (99.1%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 3714 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 43.2%\n",
      "  C: 13.2%\n",
      "  G: 9.8%\n",
      "  T: 33.9%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t3232\t2500.0\t0\t3232\n",
      "2\t342\t625.0\t0\t342\n",
      "3\t99\t156.2\t0\t99\n",
      "4\t37\t39.1\t0\t37\n",
      "15\t1\t0.0\t1\t0 1\n",
      "16\t2\t0.0\t1\t0 2\n",
      "38\t1\t0.0\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq\n",
      "=============================================\n",
      "10000 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files sub-zr2096_5_s1_R1_trimmed.fq and sub-zr2096_5_s1_R2_trimmed.fq\n",
      "file_1: sub-zr2096_5_s1_R1_trimmed.fq, file_2: sub-zr2096_5_s1_R2_trimmed.fq\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: sub-zr2096_5_s1_R1_trimmed.fq and sub-zr2096_5_s1_R2_trimmed.fq <<<<<\n",
      "Writing validated paired-end read 1 reads to sub-zr2096_5_s1_R1_val_1.fq\n",
      "Writing validated paired-end read 2 reads to sub-zr2096_5_s1_R2_val_2.fq\n",
      "\n",
      "Total number of sequences analysed: 10000\n",
      "\n",
      "Number of sequence pairs removed because at least one read was shorter than the length cutoff (20 bp): 61 (0.61%)\n",
      "\n",
      "Deleting both intermediate output files sub-zr2096_5_s1_R1_trimmed.fq and sub-zr2096_5_s1_R2_trimmed.fq\n",
      "\n",
      "====================================================================================================\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore \\\n",
    "--clip_R1 15 \\\n",
    "--clip_R2 15 \\\n",
    "--paired \\\n",
    "--output_dir /Volumes/Serine/wd/18-04-10/ \\\n",
    "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq \\\n",
    "/Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R1.fastq \\\n",
    "-2 /Volumes/Serine/wd/18-04-10/sub-zr2096_5_s1_R2.fastq  \\"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
