{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir /Volumes/Serine/wd/18-04-12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-12\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-04-12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file /Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R2_val_2.fq.gz\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing paired-end Bismark output file(s) (SAM format):\n",
      "/Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "\n",
      "\n",
      "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
      "\n",
      "Checking file >>/Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
      "\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n",
      "\n",
      "...passed!\n",
      "Output file is: zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "\n",
      "\n",
      "Total number of alignments analysed in /Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam:\t7939368\n",
      "Total number duplicated alignments removed:\t600951 (7.57%)\n",
      "Duplicated alignments were found at:\t253991 different position(s)\n",
      "\n",
      "Total count of deduplicated leftover sequences: 7338417 (92.43% of total)\n",
      "\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "/Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file /Volumes/Serine/wd/18-04-12/zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-12/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-12/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-12/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-12/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " *** Bismark methylation extractor version v0.19.0 ***\n",
      "\n",
      "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Bismark paired-end SAM format specified (default)\n",
      "Number of cores to be used: 2\n",
      "Output will be written to the current directory ('/Volumes/Serine/wd/18-04-12')\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/Serine/wd/18-04-12/zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam<< for signs of file truncation...\n",
      "\n",
      "...passed!\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "Now reading in Bismark result file /Volumes/Serine/wd/18-04-12/zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "\n",
      "Now reading in Bismark result file /Volumes/Serine/wd/18-04-12/zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n",
      "\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "Processed lines: 500000\n",
      "Processed lines: 500000\n",
      "Processed lines: 1000000\n",
      "Processed lines: 1000000\n",
      "Processed lines: 1500000\n",
      "Processed lines: 1500000\n",
      "Processed lines: 2000000\n",
      "Processed lines: 2000000\n",
      "Processed lines: 2500000\n",
      "Processed lines: 2500000\n",
      "Processed lines: 3000000\n",
      "Processed lines: 3000000\n",
      "Processed lines: 3500000\n",
      "Processed lines: 3500000\n",
      "Processed lines: 4000000\n",
      "Processed lines: 4000000\n",
      "Processed lines: 4500000\n",
      "Processed lines: 4500000\n",
      "Processed lines: 5000000\n",
      "Processed lines: 5000000\n",
      "Processed lines: 5500000\n",
      "Processed lines: 5500000\n",
      "Processed lines: 6000000\n",
      "Processed lines: 6000000\n",
      "Processed lines: 6500000\n",
      "Processed lines: 6500000\n",
      "Processed lines: 7000000\n",
      "Processed lines: 7000000\n",
      "Finished processing child process. Exiting..\n",
      "Now waiting for all child processes to complete\n",
      "\n",
      "Merging individual splitting reports into overall report: 'zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt'\n",
      "Merging from these individual files:\n",
      "\n",
      "Processed 7338417 lines in total\n",
      "Total number of methylation call strings processed: 14676834\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t128523903\n",
      "\n",
      "Total methylated C's in CpG context:\t11164357\n",
      "Total methylated C's in CHG context:\t770640\n",
      "Total methylated C's in CHH context:\t5087000\n",
      "\n",
      "Total C to T conversions in CpG context:\t4939282\n",
      "Total C to T conversions in CHG context:\t25656873\n",
      "Total C to T conversions in CHH context:\t80905751\n",
      "\n",
      "C methylated in CpG context:\t69.3%\n",
      "C methylated in CHG context:\t2.9%\n",
      "C methylated in CHH context:\t5.9%\n",
      "\n",
      "\n",
      "\n",
      "Merging individual M-bias reports into overall M-bias statistics from these 2 individual files:\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 86\n",
      "Maximum read length of Read 2: 86\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 86\n",
      "Maximum read length of Read 2: 86\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
      "output directory:\t\t><\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "Writing bedGraph to file: zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "\n",
      "The genome of interest was specified to contain gazillions of chromosomes or scaffolds. Merging all input files and sorting everything in memory instead of writing out individual chromosome files...\n",
      "Writing all merged methylation calls to temp file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged\n",
      "\n",
      "Finished writing methylation calls from CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt to merged temp file\n",
      "Sorting input file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph.gz.methylation_calls.merged by positions (using -S of 2G)\n",
      "sort: stray character in field spec: invalid field specification `3,3V'\n",
      "Died at /Applications/bioinfo/Bismark_v0.19.0/bismark2bedGraph line 486.\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "/Volumes/Serine/wd/18-04-12/zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file /Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t/Volumes/Serine/wd/18-04-12/02/zr2096_1_s1_R1_val_1_bismark_bt2_pe_splitting_report.txt.1\n",
      "/Volumes/Serine/wd/18-04-12/02/zr2096_1_s1_R1_val_1_bismark_bt2_pe_splitting_report.txt.2\n",
      "\n",
      "/Volumes/Serine/wd/18-04-12/02/zr2096_1_s1_R1_val_1_bismark_bt2_pe_splitting_report.txt.1.mbias\n",
      "/Volumes/Serine/wd/18-04-12/02/zr2096_1_s1_R1_val_1_bismark_bt2_pe_splitting_report.txt.2.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-12/02/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\t/Volumes/Serine/wd/18-04-12/02/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\t/Volumes/Serine/wd/18-04-12/02/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\t/Volumes/Serine/wd/18-04-12/02/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      " *** Bismark methylation extractor version v0.19.0 ***\n",
      "\n",
      "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
      "\n",
      "Summarising Bismark methylation extractor parameters:\n",
      "===============================================================\n",
      "Bismark paired-end SAM format specified (default)\n",
      "Number of cores to be used: 2\n",
      "Output path specified as: /Volumes/Serine/wd/18-04-12/02/\n",
      "\n",
      "\n",
      "Summarising bedGraph parameters:\n",
      "===============================================================\n",
      "Generating additional output in bedGraph and coverage format\n",
      "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
      "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
      "\n",
      "Using a cutoff of 1 read(s) to report cytosine positions\n",
      "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
      "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
      "\n",
      "Checking file >>/Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
      "\n",
      "...passed!\n",
      "Writing result file containing methylation information for C in CpG context from the original top strand to /Volumes/Serine/wd/18-04-12/02/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to /Volumes/Serine/wd/18-04-12/02/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to /Volumes/Serine/wd/18-04-12/02/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CpG context from the original bottom strand to /Volumes/Serine/wd/18-04-12/02/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "\n",
      "Writing result file containing methylation information for C in CHG context from the original top strand to /Volumes/Serine/wd/18-04-12/02/CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to /Volumes/Serine/wd/18-04-12/02/CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to /Volumes/Serine/wd/18-04-12/02/CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHG context from the original bottom strand to /Volumes/Serine/wd/18-04-12/02/CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "\n",
      "Writing result file containing methylation information for C in CHH context from the original top strand to /Volumes/Serine/wd/18-04-12/02/CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to /Volumes/Serine/wd/18-04-12/02/CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to /Volumes/Serine/wd/18-04-12/02/CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "Writing result file containing methylation information for C in CHH context from the original bottom strand to /Volumes/Serine/wd/18-04-12/02/CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "\n",
      "Now reading in Bismark result file /Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "\n",
      "Now reading in Bismark result file /Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\n",
      "\n",
      "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping SAM header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping SAM header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-10/trim14/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "Processed lines: 500000\n",
      "Processed lines: 500000\n",
      "Processed lines: 1000000\n",
      "Processed lines: 1000000\n",
      "Processed lines: 1500000\n",
      "Processed lines: 1500000\n",
      "Processed lines: 2000000\n",
      "Processed lines: 2000000\n",
      "Processed lines: 2500000\n",
      "Processed lines: 2500000\n",
      "Processed lines: 3000000\n",
      "Processed lines: 3000000\n",
      "Processed lines: 3500000\n",
      "Processed lines: 3500000\n",
      "Processed lines: 4000000\n",
      "Processed lines: 4000000\n",
      "Processed lines: 4500000\n",
      "Processed lines: 4500000\n",
      "Processed lines: 5000000\n",
      "Processed lines: 5000000\n",
      "Processed lines: 5500000\n",
      "Processed lines: 5500000\n",
      "Processed lines: 6000000\n",
      "Processed lines: 6000000\n",
      "Processed lines: 6500000\n",
      "Processed lines: 6500000\n",
      "Processed lines: 7000000\n",
      "Processed lines: 7000000\n",
      "Processed lines: 7500000\n",
      "Processed lines: 7500000\n",
      "Finished processing child process. Exiting..\n",
      "Now waiting for all child processes to complete\n",
      "\n",
      "Merging individual splitting reports into overall report: '/Volumes/Serine/wd/18-04-12/02/zr2096_1_s1_R1_val_1_bismark_bt2_pe_splitting_report.txt'\n",
      "Merging from these individual files:\n",
      "\n",
      "Processed 7939368 lines in total\n",
      "Total number of methylation call strings processed: 15878736\n",
      "\n",
      "Final Cytosine Methylation Report\n",
      "=================================\n",
      "Total number of C's analysed:\t134818960\n",
      "\n",
      "Total methylated C's in CpG context:\t11567126\n",
      "Total methylated C's in CHG context:\t884976\n",
      "Total methylated C's in CHH context:\t6648583\n",
      "\n",
      "Total C to T conversions in CpG context:\t5077028\n",
      "Total C to T conversions in CHG context:\t26418928\n",
      "Total C to T conversions in CHH context:\t84222319\n",
      "\n",
      "C methylated in CpG context:\t69.5%\n",
      "C methylated in CHG context:\t3.2%\n",
      "C methylated in CHH context:\t7.3%\n",
      "\n",
      "\n",
      "\n",
      "Merging individual M-bias reports into overall M-bias statistics from these 2 individual files:\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 86\n",
      "Maximum read length of Read 2: 86\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Determining maximum read lengths for M-Bias plots\n",
      "Maximum read length of Read 1: 86\n",
      "Maximum read length of Read 2: 86\n",
      "\n",
      "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
      "Deleting unused files ...\n",
      "\n",
      "/Volumes/Serine/wd/18-04-12/02/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "/Volumes/Serine/wd/18-04-12/02/CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt contains data ->\tkept\n",
      "\n",
      "\n",
      "Using these input files: /Volumes/Serine/wd/18-04-12/02/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHH_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHH_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHH_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt /Volumes/Serine/wd/18-04-12/02/CHH_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt\n",
      "\n",
      "Summary of parameters for bismark2bedGraph conversion:\n",
      "======================================================\n",
      "bedGraph output:\t\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.bedGraph.gz\n",
      "output directory:\t\t>/Volumes/Serine/wd/18-04-12/02/<\n",
      "remove whitespaces:\t\tno\n",
      "CX context:\t\t\tno (CpG context only, default)\n",
      "No-header selected:\t\tno\n",
      "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
      "Sort buffer size:\t\t2G\n",
      "Coverage threshold:\t\t1\n",
      "=============================================================================\n",
      "Methylation information will now be written into a bedGraph and coverage file\n",
      "=============================================================================\n",
      "\n",
      "Using the following files as Input:\n",
      "Writing bedGraph to file: zr2096_1_s1_R1_val_1_bismark_bt2_pe.bedGraph.gz\n",
      "Also writing out a coverage file including counts methylated and unmethylated residues to file: zr2096_1_s1_R1_val_1_bismark_bt2_pe.bismark.cov.gz\n",
      "\n",
      "Changed directory to /Volumes/Serine/wd/18-04-12/02/\n",
      "The genome of interest was specified to contain gazillions of chromosomes or scaffolds. Merging all input files and sorting everything in memory instead of writing out individual chromosome files...\n",
      "Writing all merged methylation calls to temp file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bedGraph.gz.methylation_calls.merged\n",
      "\n",
      "Finished writing methylation calls from CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt to merged temp file\n",
      "Finished writing methylation calls from CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.txt to merged temp file\n",
      "Sorting input file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bedGraph.gz.methylation_calls.merged by positions (using -S of 2G)\n",
      "sort: stray character in field spec: invalid field specification `3,3V'\n",
      "Died at /Applications/bioinfo/Bismark_v0.19.0/bismark2bedGraph line 486.\n",
      "Finished BedGraph conversion ...\n",
      "\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "/Volumes/Serine/wd/18-04-11/zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam \\\n",
    "-o /Volumes/Serine/wd/18-04-12/02/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
