{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:27:08.707360Z",
     "start_time": "2018-04-16T03:27:08.700501Z"
    }
   },
   "outputs": [],
   "source": [
    "wd=\"~/Desktop/wd/0414\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:05:07.902240Z",
     "start_time": "2018-04-14T13:05:07.756930Z"
    }
   },
   "outputs": [],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:27:10.383243Z",
     "start_time": "2018-04-16T03:27:10.370526Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/Desktop/wd/0414\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:18:12.818474Z",
     "start_time": "2018-04-14T13:12:09.398262Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5633\n",
      "Mapping efficiency:\t56.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2487\n",
      "Sequence pairs did not map uniquely:\t1880\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1459\t((converted) top strand)\n",
      "GA/CT/CT:\t1404\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1399\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1371\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2716\n",
      "Mapping efficiency:\t27.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t6421\n",
      "Sequence pairs did not map uniquely:\t863\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t678\t((converted) top strand)\n",
      "GA/CT/CT:\t727\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t620\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t691\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5054\n",
      "Mapping efficiency:\t50.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3239\n",
      "Sequence pairs did not map uniquely:\t1707\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1289\t((converted) top strand)\n",
      "GA/CT/CT:\t1231\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1276\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1258\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5653\n",
      "Mapping efficiency:\t56.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2384\n",
      "Sequence pairs did not map uniquely:\t1963\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1389\t((converted) top strand)\n",
      "GA/CT/CT:\t1385\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1464\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1415\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5610\n",
      "Mapping efficiency:\t56.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2472\n",
      "Sequence pairs did not map uniquely:\t1918\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1481\t((converted) top strand)\n",
      "GA/CT/CT:\t1339\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1346\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1444\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5413\n",
      "Mapping efficiency:\t54.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2801\n",
      "Sequence pairs did not map uniquely:\t1786\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1361\t((converted) top strand)\n",
      "GA/CT/CT:\t1357\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1299\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1396\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5791\n",
      "Mapping efficiency:\t57.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2268\n",
      "Sequence pairs did not map uniquely:\t1941\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1586\t((converted) top strand)\n",
      "GA/CT/CT:\t1325\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1318\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1562\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5450\n",
      "Mapping efficiency:\t54.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2835\n",
      "Sequence pairs did not map uniquely:\t1715\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1347\t((converted) top strand)\n",
      "GA/CT/CT:\t1332\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1357\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1414\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4999\n",
      "Mapping efficiency:\t50.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3412\n",
      "Sequence pairs did not map uniquely:\t1589\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1229\t((converted) top strand)\n",
      "GA/CT/CT:\t1255\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1221\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1294\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5333\n",
      "Mapping efficiency:\t53.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2830\n",
      "Sequence pairs did not map uniquely:\t1837\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1425\t((converted) top strand)\n",
      "GA/CT/CT:\t1286\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1126\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1496\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/trim14/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-u 10000 \\\n",
    "-p 2 \\\n",
    "--non_directional \\\n",
    "-1 /Users/sr320/Desktop/trim14/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Users/sr320/Desktop/trim14/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark.err\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:18:13.475702Z",
     "start_time": "2018-04-14T13:18:13.309929Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:19:07.227005Z",
     "start_time": "2018-04-14T13:18:43.156863Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_10_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_10_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_1_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_2_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_2_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_3_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_3_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_4_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_4_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_5_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_5_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_6_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_6_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_7_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_7_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_8_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_8_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_9_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_9_s1_R2_val_2.fq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.253479Z",
     "start_time": "2018-04-14T13:19:50.380957Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.410812Z",
     "start_time": "2018-04-14T13:22:51.256976Z"
    }
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.552595Z",
     "start_time": "2018-04-14T13:22:51.418717Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t12154\t12155\t0\r\n",
      "NC_007175.2\t12209\t12210\t0\r\n",
      "NC_007175.2\t12219\t12220\t0\r\n",
      "NC_035780.1\t257956\t257957\t100\r\n",
      "NC_035780.1\t257970\t257971\t100\r\n",
      "NC_035780.1\t257982\t257983\t100\r\n",
      "NC_035780.1\t257994\t257995\t100\r\n",
      "NC_035780.1\t258025\t258026\t100\r\n",
      "NC_035780.1\t322880\t322881\t100\r\n",
      "\r\n",
      "==> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t100634\t100635\t100\r\n",
      "NC_035780.1\t100643\t100644\t100\r\n",
      "NC_035780.1\t100651\t100652\t100\r\n",
      "NC_035780.1\t100664\t100665\t100\r\n",
      "NC_035780.1\t471536\t471537\t100\r\n",
      "NC_035780.1\t471539\t471540\t100\r\n",
      "NC_035780.1\t471550\t471551\t100\r\n",
      "NC_035780.1\t471584\t471585\t100\r\n",
      "NC_035780.1\t471603\t471604\t100\r\n",
      "\r\n",
      "==> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t4394\t4395\t0\r\n",
      "NC_007175.2\t4408\t4409\t0\r\n",
      "NC_007175.2\t4448\t4449\t0\r\n",
      "NC_007175.2\t4478\t4479\t0\r\n",
      "NC_007175.2\t6458\t6459\t0\r\n",
      "NC_007175.2\t6460\t6461\t0\r\n",
      "NC_007175.2\t6466\t6467\t0\r\n",
      "NC_007175.2\t11969\t11970\t0\r\n",
      "NC_007175.2\t11971\t11972\t0\r\n",
      "\r\n",
      "==> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t7089\t7090\t0\r\n",
      "NC_007175.2\t7099\t7100\t0\r\n",
      "NC_007175.2\t7121\t7122\t0\r\n",
      "NC_007175.2\t7147\t7148\t0\r\n",
      "NC_007175.2\t14357\t14358\t0\r\n",
      "NC_035780.1\t257471\t257472\t100\r\n",
      "\r\n",
      "==> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t3345\t3346\t0\r\n",
      "NC_007175.2\t3385\t3386\t0\r\n",
      "NC_007175.2\t3396\t3397\t0\r\n",
      "NC_007175.2\t3859\t3860\t0\r\n",
      "NC_007175.2\t3886\t3887\t0\r\n",
      "NC_007175.2\t7248\t7249\t0\r\n",
      "NC_007175.2\t7253\t7254\t0\r\n",
      "NC_007175.2\t7264\t7265\t0\r\n",
      "NC_007175.2\t7284\t7285\t0\r\n",
      "\r\n",
      "==> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t724\t725\t0\r\n",
      "NC_007175.2\t975\t976\t0\r\n",
      "NC_007175.2\t994\t995\t0\r\n",
      "NC_007175.2\t2145\t2146\t0\r\n",
      "NC_007175.2\t2148\t2149\t0\r\n",
      "\r\n",
      "==> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t250559\t250560\t100\r\n",
      "NC_035780.1\t269759\t269760\t100\r\n",
      "NC_035780.1\t269762\t269763\t100\r\n",
      "NC_035780.1\t269781\t269782\t100\r\n",
      "NC_035780.1\t269789\t269790\t100\r\n",
      "NC_035780.1\t269822\t269823\t100\r\n",
      "NC_035780.1\t270312\t270313\t100\r\n",
      "NC_035780.1\t285489\t285490\t100\r\n",
      "NC_035780.1\t285494\t285495\t100\r\n",
      "\r\n",
      "==> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t664\t665\t0\r\n",
      "NC_007175.2\t683\t684\t0\r\n",
      "NC_007175.2\t708\t709\t0\r\n",
      "NC_007175.2\t724\t725\t0\r\n",
      "NC_007175.2\t9943\t9944\t0\r\n",
      "NC_007175.2\t9947\t9948\t0\r\n",
      "NC_007175.2\t9959\t9960\t0\r\n",
      "NC_007175.2\t9964\t9965\t0\r\n",
      "NC_007175.2\t13202\t13203\t0\r\n",
      "\r\n",
      "==> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2301\t2302\t0\r\n",
      "NC_007175.2\t2304\t2305\t0\r\n",
      "NC_007175.2\t2356\t2357\t0\r\n",
      "NC_007175.2\t2365\t2366\t0\r\n",
      "NC_007175.2\t2472\t2473\t0\r\n",
      "NC_007175.2\t2490\t2491\t0\r\n",
      "NC_007175.2\t2635\t2636\t0\r\n",
      "NC_007175.2\t2653\t2654\t0\r\n",
      "NC_007175.2\t2657\t2658\t0\r\n",
      "\r\n",
      "==> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t3099\t3100\t0\r\n",
      "NC_007175.2\t3102\t3103\t0\r\n",
      "NC_007175.2\t3114\t3115\t0\r\n",
      "NC_007175.2\t3213\t3214\t0\r\n",
      "NC_007175.2\t3222\t3223\t0\r\n",
      "NC_007175.2\t3245\t3246\t0\r\n",
      "NC_007175.2\t10997\t10998\t0\r\n",
      "NC_007175.2\t11102\t11103\t0\r\n",
      "NC_007175.2\t11468\t11469\t0\r\n"
     ]
    }
   ],
   "source": [
    "!head *.deduplicated.bedGraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.959674Z",
     "start_time": "2018-04-14T13:22:51.560989Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 10 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:52.878847Z",
     "start_time": "2018-04-14T13:22:51.967310Z"
    }
   },
   "outputs": [],
   "source": [
    "!open *report.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:52.988699Z",
     "start_time": "2018-04-14T13:22:52.885377Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 10 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:30:59.862210Z",
     "start_time": "2018-04-16T03:30:59.727396Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n"
     ]
    }
   ],
   "source": [
    "ls *deduplicated.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:37:55.461016Z",
     "start_time": "2018-04-16T03:31:10.278888Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Summary of parameters for nucleotide coverage report:\n",
      "=====================================================\n",
      "Output directory:\t\t\t><\n",
      "Parent directory:\t\t\t>/Users/sr320/Desktop/wd/0414/<\n",
      "Genome directory:\t\t\t>/Users/sr320/Dropbox/wd/18-03-15/genome/<\n",
      "Samtools installation:\t\t\t>/Applications/bioinfo/samtools-1.7/samtools<\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Users/sr320/Dropbox/wd/18-03-15/genome/\n",
      "\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Stored sequence information of 11 chromosomes/scaffolds in total\n",
      "\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Could not find genomic nucleotide frequency table in the genome folder, calculating genomic frequencies (this may take several minutes depending on genome size) ...\n",
      "====================================================================================================================================================================\n",
      "Processing chromosome >> NC_035784.1 <<\n",
      "Processing chromosome >> NC_035781.1 <<\n",
      "Processing chromosome >> NC_035783.1 <<\n",
      "Processing chromosome >> NC_035788.1 <<\n",
      "Processing chromosome >> NC_007175.2 <<\n",
      "Processing chromosome >> NC_035786.1 <<\n",
      "Processing chromosome >> NC_035782.1 <<\n",
      "Processing chromosome >> NC_035787.1 <<\n",
      "Processing chromosome >> NC_035780.1 <<\n",
      "Processing chromosome >> NC_035789.1 <<\n",
      "Processing chromosome >> NC_035785.1 <<\n",
      "Writing genomic nucleotide frequencies to the file >/Users/sr320/Dropbox/wd/18-03-15/genome/genomic_nucleotide_frequencies.txt< for future re-use\n",
      "Writing count of (di-)nucleotide: A\t223134775\n",
      "Writing count of (di-)nucleotide: AA\t80795037\n",
      "Writing count of (di-)nucleotide: AC\t37359527\n",
      "Writing count of (di-)nucleotide: AG\t35548672\n",
      "Writing count of (di-)nucleotide: AT\t69431301\n",
      "Writing count of (di-)nucleotide: C\t119229119\n",
      "Writing count of (di-)nucleotide: CA\t45827414\n",
      "Writing count of (di-)nucleotide: CC\t23444678\n",
      "Writing count of (di-)nucleotide: CG\t14458703\n",
      "Writing count of (di-)nucleotide: CT\t35498217\n",
      "Writing count of (di-)nucleotide: G\t119267677\n",
      "Writing count of (di-)nucleotide: GA\t38810489\n",
      "Writing count of (di-)nucleotide: GC\t19646670\n",
      "Writing count of (di-)nucleotide: GG\t23423030\n",
      "Writing count of (di-)nucleotide: GT\t37387368\n",
      "Writing count of (di-)nucleotide: T\t223043757\n",
      "Writing count of (di-)nucleotide: TA\t57701608\n",
      "Writing count of (di-)nucleotide: TC\t38778132\n",
      "Writing count of (di-)nucleotide: TG\t45837169\n",
      "Writing count of (di-)nucleotide: TT\t80726644\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t121078\t27.46\t223134775\t32.59\t0.001\n",
      "C\t91381\t20.72\t119229119\t17.41\t0.001\n",
      "G\t104758\t23.75\t119267677\t17.42\t0.001\n",
      "T\t123788\t28.07\t223043757\t32.58\t0.001\n",
      "AA\t35023\t8.05\t80795037\t11.80\t0.000\n",
      "AC\t23859\t5.48\t37359527\t5.46\t0.001\n",
      "AG\t29337\t6.74\t35548672\t5.19\t0.001\n",
      "AT\t31180\t7.17\t69431301\t10.14\t0.000\n",
      "CA\t29589\t6.80\t45827414\t6.69\t0.001\n",
      "CC\t21478\t4.94\t23444678\t3.42\t0.001\n",
      "CG\t13077\t3.01\t14458703\t2.11\t0.001\n",
      "CT\t25981\t5.97\t35498217\t5.18\t0.001\n",
      "GA\t31192\t7.17\t38810489\t5.67\t0.001\n",
      "GC\t18062\t4.15\t19646670\t2.87\t0.001\n",
      "GG\t26092\t6.00\t23423030\t3.42\t0.001\n",
      "GT\t28026\t6.44\t37387368\t5.46\t0.001\n",
      "TA\t23822\t5.48\t57701608\t8.43\t0.000\n",
      "TC\t26760\t6.15\t38778132\t5.66\t0.001\n",
      "TG\t34741\t7.99\t45837169\t6.69\t0.001\n",
      "TT\t36831\t8.47\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t51573\t28.40\t223134775\t32.59\t0.000\n",
      "C\t36192\t19.93\t119229119\t17.41\t0.000\n",
      "G\t41190\t22.68\t119267677\t17.42\t0.000\n",
      "T\t52632\t28.98\t223043757\t32.58\t0.000\n",
      "AA\t15711\t8.77\t80795037\t11.80\t0.000\n",
      "AC\t9799\t5.47\t37359527\t5.46\t0.000\n",
      "AG\t11759\t6.57\t35548672\t5.19\t0.000\n",
      "AT\t13584\t7.58\t69431301\t10.14\t0.000\n",
      "CA\t12014\t6.71\t45827414\t6.69\t0.000\n",
      "CC\t8170\t4.56\t23444678\t3.42\t0.000\n",
      "CG\t4777\t2.67\t14458703\t2.11\t0.000\n",
      "CT\t10747\t6.00\t35498217\t5.18\t0.000\n",
      "GA\t12630\t7.05\t38810489\t5.67\t0.000\n",
      "GC\t6973\t3.89\t19646670\t2.87\t0.000\n",
      "GG\t9739\t5.44\t23423030\t3.42\t0.000\n",
      "GT\t11329\t6.33\t37387368\t5.46\t0.000\n",
      "TA\t10604\t5.92\t57701608\t8.43\t0.000\n",
      "TC\t10777\t6.02\t38778132\t5.66\t0.000\n",
      "TG\t14270\t7.97\t45837169\t6.69\t0.000\n",
      "TT\t16228\t9.06\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t98812\t27.31\t223134775\t32.59\t0.000\n",
      "C\t72957\t20.16\t119229119\t17.41\t0.001\n",
      "G\t87596\t24.21\t119267677\t17.42\t0.001\n",
      "T\t102479\t28.32\t223043757\t32.58\t0.000\n",
      "AA\t28470\t7.99\t80795037\t11.80\t0.000\n",
      "AC\t19359\t5.43\t37359527\t5.46\t0.001\n",
      "AG\t24148\t6.77\t35548672\t5.19\t0.001\n",
      "AT\t25435\t7.13\t69431301\t10.14\t0.000\n",
      "CA\t23774\t6.67\t45827414\t6.69\t0.001\n",
      "CC\t16884\t4.74\t23444678\t3.42\t0.001\n",
      "CG\t10293\t2.89\t14458703\t2.11\t0.001\n",
      "CT\t20932\t5.87\t35498217\t5.18\t0.001\n",
      "GA\t26228\t7.36\t38810489\t5.67\t0.001\n",
      "GC\t14500\t4.07\t19646670\t2.87\t0.001\n",
      "GG\t22071\t6.19\t23423030\t3.42\t0.001\n",
      "GT\t23534\t6.60\t37387368\t5.46\t0.001\n",
      "TA\t19034\t5.34\t57701608\t8.43\t0.000\n",
      "TC\t21157\t5.93\t38778132\t5.66\t0.001\n",
      "TG\t29668\t8.32\t45837169\t6.69\t0.001\n",
      "TT\t31043\t8.71\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t119301\t27.09\t223134775\t32.59\t0.001\n",
      "C\t92563\t21.02\t119229119\t17.41\t0.001\n",
      "G\t105861\t24.04\t119267677\t17.42\t0.001\n",
      "T\t122661\t27.85\t223043757\t32.58\t0.001\n",
      "AA\t33785\t7.78\t80795037\t11.80\t0.000\n",
      "AC\t24532\t5.65\t37359527\t5.46\t0.001\n",
      "AG\t29145\t6.71\t35548672\t5.19\t0.001\n",
      "AT\t30251\t6.97\t69431301\t10.14\t0.000\n",
      "CA\t29568\t6.81\t45827414\t6.69\t0.001\n",
      "CC\t21931\t5.05\t23444678\t3.42\t0.001\n",
      "CG\t13707\t3.16\t14458703\t2.11\t0.001\n",
      "CT\t26029\t5.99\t35498217\t5.18\t0.001\n",
      "GA\t31406\t7.23\t38810489\t5.67\t0.001\n",
      "GC\t18209\t4.19\t19646670\t2.87\t0.001\n",
      "GG\t26558\t6.11\t23423030\t3.42\t0.001\n",
      "GT\t28267\t6.51\t37387368\t5.46\t0.001\n",
      "TA\t23156\t5.33\t57701608\t8.43\t0.000\n",
      "TC\t26626\t6.13\t38778132\t5.66\t0.001\n",
      "TG\t34775\t8.01\t45837169\t6.69\t0.001\n",
      "TT\t36368\t8.37\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t116684\t27.98\t223134775\t32.59\t0.001\n",
      "C\t83569\t20.04\t119229119\t17.41\t0.001\n",
      "G\t98421\t23.60\t119267677\t17.42\t0.001\n",
      "T\t118422\t28.39\t223043757\t32.58\t0.001\n",
      "AA\t34595\t8.42\t80795037\t11.80\t0.000\n",
      "AC\t22542\t5.48\t37359527\t5.46\t0.001\n",
      "AG\t28003\t6.81\t35548672\t5.19\t0.001\n",
      "AT\t29907\t7.27\t69431301\t10.14\t0.000\n",
      "CA\t27849\t6.77\t45827414\t6.69\t0.001\n",
      "CC\t19482\t4.74\t23444678\t3.42\t0.001\n",
      "CG\t11074\t2.69\t14458703\t2.11\t0.001\n",
      "CT\t23928\t5.82\t35498217\t5.18\t0.001\n",
      "GA\t29938\t7.28\t38810489\t5.67\t0.001\n",
      "GC\t16216\t3.94\t19646670\t2.87\t0.001\n",
      "GG\t24352\t5.92\t23423030\t3.42\t0.001\n",
      "GT\t26494\t6.44\t37387368\t5.46\t0.001\n",
      "TA\t22822\t5.55\t57701608\t8.43\t0.000\n",
      "TC\t24135\t5.87\t38778132\t5.66\t0.001\n",
      "TG\t33435\t8.13\t45837169\t6.69\t0.001\n",
      "TT\t36338\t8.84\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t112593\t28.00\t223134775\t32.59\t0.001\n",
      "C\t81198\t20.19\t119229119\t17.41\t0.001\n",
      "G\t91793\t22.82\t119267677\t17.42\t0.001\n",
      "T\t116583\t28.99\t223043757\t32.58\t0.001\n",
      "AA\t32910\t8.30\t80795037\t11.80\t0.000\n",
      "AC\t21844\t5.51\t37359527\t5.46\t0.001\n",
      "AG\t26534\t6.69\t35548672\t5.19\t0.001\n",
      "AT\t29823\t7.52\t69431301\t10.14\t0.000\n",
      "CA\t27247\t6.87\t45827414\t6.69\t0.001\n",
      "CC\t18675\t4.71\t23444678\t3.42\t0.001\n",
      "CG\t10532\t2.66\t14458703\t2.11\t0.001\n",
      "CT\t23593\t5.95\t35498217\t5.18\t0.001\n",
      "GA\t27761\t7.00\t38810489\t5.67\t0.001\n",
      "GC\t15657\t3.95\t19646670\t2.87\t0.001\n",
      "GG\t21578\t5.44\t23423030\t3.42\t0.001\n",
      "GT\t25565\t6.44\t37387368\t5.46\t0.001\n",
      "TA\t23323\t5.88\t57701608\t8.43\t0.000\n",
      "TC\t23957\t6.04\t38778132\t5.66\t0.001\n",
      "TG\t31757\t8.01\t45837169\t6.69\t0.001\n",
      "TT\t35911\t9.05\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t118321\t27.92\t223134775\t32.59\t0.001\n",
      "C\t84497\t19.94\t119229119\t17.41\t0.001\n",
      "G\t98886\t23.33\t119267677\t17.42\t0.001\n",
      "T\t122128\t28.82\t223043757\t32.58\t0.001\n",
      "AA\t34700\t8.31\t80795037\t11.80\t0.000\n",
      "AC\t22808\t5.46\t37359527\t5.46\t0.001\n",
      "AG\t28049\t6.71\t35548672\t5.19\t0.001\n",
      "AT\t31099\t7.44\t69431301\t10.14\t0.000\n",
      "CA\t28301\t6.77\t45827414\t6.69\t0.001\n",
      "CC\t19025\t4.55\t23444678\t3.42\t0.001\n",
      "CG\t11751\t2.81\t14458703\t2.11\t0.001\n",
      "CT\t24228\t5.80\t35498217\t5.18\t0.001\n",
      "GA\t29721\t7.11\t38810489\t5.67\t0.001\n",
      "GC\t16464\t3.94\t19646670\t2.87\t0.001\n",
      "GG\t23733\t5.68\t23423030\t3.42\t0.001\n",
      "GT\t27514\t6.59\t37387368\t5.46\t0.001\n",
      "TA\t24142\t5.78\t57701608\t8.43\t0.000\n",
      "TC\t25011\t5.99\t38778132\t5.66\t0.001\n",
      "TG\t33764\t8.08\t45837169\t6.69\t0.001\n",
      "TT\t37493\t8.97\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t115161\t28.22\t223134775\t32.59\t0.001\n",
      "C\t81384\t19.95\t119229119\t17.41\t0.001\n",
      "G\t93814\t22.99\t119267677\t17.42\t0.001\n",
      "T\t117661\t28.84\t223043757\t32.58\t0.001\n",
      "AA\t34132\t8.48\t80795037\t11.80\t0.000\n",
      "AC\t22375\t5.56\t37359527\t5.46\t0.001\n",
      "AG\t26945\t6.69\t35548672\t5.19\t0.001\n",
      "AT\t30170\t7.50\t69431301\t10.14\t0.000\n",
      "CA\t27657\t6.87\t45827414\t6.69\t0.001\n",
      "CC\t18536\t4.61\t23444678\t3.42\t0.001\n",
      "CG\t10443\t2.59\t14458703\t2.11\t0.001\n",
      "CT\t23627\t5.87\t35498217\t5.18\t0.001\n",
      "GA\t28649\t7.12\t38810489\t5.67\t0.001\n",
      "GC\t15455\t3.84\t19646670\t2.87\t0.001\n",
      "GG\t22488\t5.59\t23423030\t3.42\t0.001\n",
      "GT\t25971\t6.45\t37387368\t5.46\t0.001\n",
      "TA\t23372\t5.81\t57701608\t8.43\t0.000\n",
      "TC\t23911\t5.94\t38778132\t5.66\t0.001\n",
      "TG\t32587\t8.10\t45837169\t6.69\t0.001\n",
      "TT\t36189\t8.99\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t105115\t28.28\t223134775\t32.59\t0.000\n",
      "C\t72701\t19.56\t119229119\t17.41\t0.001\n",
      "G\t85417\t22.98\t119267677\t17.42\t0.001\n",
      "T\t108429\t29.17\t223043757\t32.58\t0.000\n",
      "AA\t31488\t8.59\t80795037\t11.80\t0.000\n",
      "AC\t19659\t5.36\t37359527\t5.46\t0.001\n",
      "AG\t24452\t6.67\t35548672\t5.19\t0.001\n",
      "AT\t28073\t7.66\t69431301\t10.14\t0.000\n",
      "CA\t24535\t6.69\t45827414\t6.69\t0.001\n",
      "CC\t16663\t4.55\t23444678\t3.42\t0.001\n",
      "CG\t9161\t2.50\t14458703\t2.11\t0.001\n",
      "CT\t21346\t5.82\t35498217\t5.18\t0.001\n",
      "GA\t26133\t7.13\t38810489\t5.67\t0.001\n",
      "GC\t13849\t3.78\t19646670\t2.87\t0.001\n",
      "GG\t20589\t5.62\t23423030\t3.42\t0.001\n",
      "GT\t23740\t6.48\t37387368\t5.46\t0.001\n",
      "TA\t21683\t5.91\t57701608\t8.43\t0.000\n",
      "TC\t21575\t5.88\t38778132\t5.66\t0.001\n",
      "TG\t29970\t8.17\t45837169\t6.69\t0.001\n",
      "TT\t33697\t9.19\t80726644\t11.79\t0.000\n",
      "==================================================================\n",
      "Mono- and di-nucleotide coverage will now be written into a report\n",
      "==================================================================\n",
      "\n",
      "Detected file 'genomic_nucleotide_frequencies.txt' in the genome folder /Users/sr320/Dropbox/wd/18-03-15/genome/ already. Using nucleotide frequencies contained therein ...\n",
      "============================================================================================================================================================================================\n",
      "\n",
      "\n",
      "Calculating read frequencies from file 'zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam'\n",
      "==========================================================================================\n",
      "Determined the file to be paired-end\n",
      "\n",
      "\n",
      "Printing nucleotide stats to >> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.nucleotide_stats.txt <<\n",
      "Final Stage: Calculating averages\n",
      "=================================\n",
      "\n",
      "(di-)nucleotide\tcount sample\tpercent sample\tcount genomic\tpercent genomic\tcoverage\n",
      "A\t108975\t26.91\t223134775\t32.59\t0.000\n",
      "C\t87189\t21.53\t119229119\t17.41\t0.001\n",
      "G\t95790\t23.65\t119267677\t17.42\t0.001\n",
      "T\t113028\t27.91\t223043757\t32.58\t0.001\n",
      "AA\t30091\t7.54\t80795037\t11.80\t0.000\n",
      "AC\t23028\t5.77\t37359527\t5.46\t0.001\n",
      "AG\t26107\t6.54\t35548672\t5.19\t0.001\n",
      "AT\t28199\t7.06\t69431301\t10.14\t0.000\n",
      "CA\t28231\t7.07\t45827414\t6.69\t0.001\n",
      "CC\t20334\t5.09\t23444678\t3.42\t0.001\n",
      "CG\t12687\t3.18\t14458703\t2.11\t0.001\n",
      "CT\t24700\t6.19\t35498217\t5.18\t0.001\n",
      "GA\t27762\t6.95\t38810489\t5.67\t0.001\n",
      "GC\t17284\t4.33\t19646670\t2.87\t0.001\n",
      "GG\t23595\t5.91\t23423030\t3.42\t0.001\n",
      "GT\t25815\t6.46\t37387368\t5.46\t0.001\n",
      "TA\t21593\t5.41\t57701608\t8.43\t0.000\n",
      "TC\t25325\t6.34\t38778132\t5.66\t0.001\n",
      "TG\t31968\t8.01\t45837169\t6.69\t0.001\n",
      "TT\t32628\t8.17\t80726644\t11.79\t0.000\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bam2nuc \\\n",
    "--genome_folder /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "*deduplicated.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:44:06.905686Z",
     "start_time": "2018-04-16T03:44:06.788353Z"
    }
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'y' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                     Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-009520053b00>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0my\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'y' is not defined"
     ]
    }
   ],
   "source": [
    "y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:44:10.262122Z",
     "start_time": "2018-04-16T03:44:10.131477Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n"
     ]
    }
   ],
   "source": [
    "ls *.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tabulate_tsvs.py blah1.cov blah2.cov blah3.cov -k 0 1 -c 4 5 -v \\\n",
    "> compiled_coverage.pre-filt.meth_unmeth.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:48:41.743034Z",
     "start_time": "2018-04-16T03:48:40.168694Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Files used: zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov, zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "Headers are NOT PRESENT.\n",
      "Reading file #10\n",
      "Union of all files produces 112270 rows.\n"
     ]
    }
   ],
   "source": [
    "!python /Users/sr320/Documents/GitHub/nb-2018/scripts/working_with_dna_meth-master/tabulate_tsvs.py \\\n",
    "*.cov \\\n",
    "-k 0 1 -c 4 5 -v \\\n",
    "> compiled_coverage.pre-filt.meth_unmeth.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:49:52.500084Z",
     "start_time": "2018-04-16T03:49:52.276098Z"
    }
   },
   "outputs": [],
   "source": [
    "!gzip compiled_coverage.pre-filt.meth_unmeth.tsv\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:51:03.576441Z",
     "start_time": "2018-04-16T03:51:02.648499Z"
    }
   },
   "outputs": [],
   "source": [
    "!python /Users/sr320/Documents/GitHub/nb-2018/scripts/working_with_dna_meth-master/merge_bismark_cov.py \\\n",
    "*.cov > all.merged.cov\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:58:34.481981Z",
     "start_time": "2018-04-16T03:58:25.355849Z"
    }
   },
   "outputs": [],
   "source": [
    "!python /Users/sr320/Documents/GitHub/nb-2018/scripts/working_with_dna_meth-master/filter_miscalled_Cs.py \\\n",
    "all.merged.cov > all.bona_fide_meth_pos.cov\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:58:34.775344Z",
     "start_time": "2018-04-16T03:58:34.485841Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking positions that are considered bona fide (criteria IV)...\r\n",
      "Traceback (most recent call last):\r\n",
      "  File \"/Users/sr320/Documents/GitHub/nb-2018/scripts/working_with_dna_meth-master/filter_pos.four_criteria.py\", line 46, in <module>\r\n",
      "    scaf_size = int(scaf.split('size')[-1])\r\n",
      "ValueError: invalid literal for int() with base 10: 'NC_007175.2'\r\n"
     ]
    }
   ],
   "source": [
    "!python /Users/sr320/Documents/GitHub/nb-2018/scripts/working_with_dna_meth-master/filter_pos.four_criteria.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
