{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:27:08.707360Z",
     "start_time": "2018-04-16T03:27:08.700501Z"
    }
   },
   "outputs": [],
   "source": [
    "wd=\"~/Desktop/wd/0414\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:05:07.902240Z",
     "start_time": "2018-04-14T13:05:07.756930Z"
    }
   },
   "outputs": [],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:27:10.383243Z",
     "start_time": "2018-04-16T03:27:10.370526Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/Desktop/wd/0414\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:18:12.818474Z",
     "start_time": "2018-04-14T13:12:09.398262Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5633\n",
      "Mapping efficiency:\t56.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2487\n",
      "Sequence pairs did not map uniquely:\t1880\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1459\t((converted) top strand)\n",
      "GA/CT/CT:\t1404\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1399\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1371\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2716\n",
      "Mapping efficiency:\t27.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t6421\n",
      "Sequence pairs did not map uniquely:\t863\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t678\t((converted) top strand)\n",
      "GA/CT/CT:\t727\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t620\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t691\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5054\n",
      "Mapping efficiency:\t50.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3239\n",
      "Sequence pairs did not map uniquely:\t1707\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1289\t((converted) top strand)\n",
      "GA/CT/CT:\t1231\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1276\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1258\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5653\n",
      "Mapping efficiency:\t56.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2384\n",
      "Sequence pairs did not map uniquely:\t1963\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1389\t((converted) top strand)\n",
      "GA/CT/CT:\t1385\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1464\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1415\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5610\n",
      "Mapping efficiency:\t56.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2472\n",
      "Sequence pairs did not map uniquely:\t1918\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1481\t((converted) top strand)\n",
      "GA/CT/CT:\t1339\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1346\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1444\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5413\n",
      "Mapping efficiency:\t54.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2801\n",
      "Sequence pairs did not map uniquely:\t1786\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1361\t((converted) top strand)\n",
      "GA/CT/CT:\t1357\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1299\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1396\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5791\n",
      "Mapping efficiency:\t57.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2268\n",
      "Sequence pairs did not map uniquely:\t1941\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1586\t((converted) top strand)\n",
      "GA/CT/CT:\t1325\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1318\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1562\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5450\n",
      "Mapping efficiency:\t54.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2835\n",
      "Sequence pairs did not map uniquely:\t1715\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1347\t((converted) top strand)\n",
      "GA/CT/CT:\t1332\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1357\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1414\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4999\n",
      "Mapping efficiency:\t50.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3412\n",
      "Sequence pairs did not map uniquely:\t1589\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1229\t((converted) top strand)\n",
      "GA/CT/CT:\t1255\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1221\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1294\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 2 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t5333\n",
      "Mapping efficiency:\t53.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t2830\n",
      "Sequence pairs did not map uniquely:\t1837\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1425\t((converted) top strand)\n",
      "GA/CT/CT:\t1286\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1126\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1496\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/trim14/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-u 10000 \\\n",
    "-p 2 \\\n",
    "--non_directional \\\n",
    "-1 /Users/sr320/Desktop/trim14/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Users/sr320/Desktop/trim14/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark.err\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:18:13.475702Z",
     "start_time": "2018-04-14T13:18:13.309929Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bismark.err\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam\r\n"
     ]
    }
   ],
   "source": [
    "ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:19:07.227005Z",
     "start_time": "2018-04-14T13:18:43.156863Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_10_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_10_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_1_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_2_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_2_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_3_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_3_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_4_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_4_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_5_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_5_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_6_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_6_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_7_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_7_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_8_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_8_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Users/sr320/Dropbox/wd/18-03-15/genome --score_min L,0,-1.2 -u 10000 -p 2 --non_directional -1 /Users/sr320/Desktop/trim14/zr2096_9_s1_R1_val_1.fq.gz -2 /Users/sr320/Desktop/trim14/zr2096_9_s1_R2_val_2.fq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.253479Z",
     "start_time": "2018-04-14T13:19:50.380957Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "\n",
      "/Users/sr320/Desktop/wd/0414/CpG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Users/sr320/Desktop/wd/0414/CpG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.410812Z",
     "start_time": "2018-04-14T13:22:51.256976Z"
    }
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.552595Z",
     "start_time": "2018-04-14T13:22:51.418717Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t12154\t12155\t0\r\n",
      "NC_007175.2\t12209\t12210\t0\r\n",
      "NC_007175.2\t12219\t12220\t0\r\n",
      "NC_035780.1\t257956\t257957\t100\r\n",
      "NC_035780.1\t257970\t257971\t100\r\n",
      "NC_035780.1\t257982\t257983\t100\r\n",
      "NC_035780.1\t257994\t257995\t100\r\n",
      "NC_035780.1\t258025\t258026\t100\r\n",
      "NC_035780.1\t322880\t322881\t100\r\n",
      "\r\n",
      "==> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t100634\t100635\t100\r\n",
      "NC_035780.1\t100643\t100644\t100\r\n",
      "NC_035780.1\t100651\t100652\t100\r\n",
      "NC_035780.1\t100664\t100665\t100\r\n",
      "NC_035780.1\t471536\t471537\t100\r\n",
      "NC_035780.1\t471539\t471540\t100\r\n",
      "NC_035780.1\t471550\t471551\t100\r\n",
      "NC_035780.1\t471584\t471585\t100\r\n",
      "NC_035780.1\t471603\t471604\t100\r\n",
      "\r\n",
      "==> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t4394\t4395\t0\r\n",
      "NC_007175.2\t4408\t4409\t0\r\n",
      "NC_007175.2\t4448\t4449\t0\r\n",
      "NC_007175.2\t4478\t4479\t0\r\n",
      "NC_007175.2\t6458\t6459\t0\r\n",
      "NC_007175.2\t6460\t6461\t0\r\n",
      "NC_007175.2\t6466\t6467\t0\r\n",
      "NC_007175.2\t11969\t11970\t0\r\n",
      "NC_007175.2\t11971\t11972\t0\r\n",
      "\r\n",
      "==> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t7089\t7090\t0\r\n",
      "NC_007175.2\t7099\t7100\t0\r\n",
      "NC_007175.2\t7121\t7122\t0\r\n",
      "NC_007175.2\t7147\t7148\t0\r\n",
      "NC_007175.2\t14357\t14358\t0\r\n",
      "NC_035780.1\t257471\t257472\t100\r\n",
      "\r\n",
      "==> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t3345\t3346\t0\r\n",
      "NC_007175.2\t3385\t3386\t0\r\n",
      "NC_007175.2\t3396\t3397\t0\r\n",
      "NC_007175.2\t3859\t3860\t0\r\n",
      "NC_007175.2\t3886\t3887\t0\r\n",
      "NC_007175.2\t7248\t7249\t0\r\n",
      "NC_007175.2\t7253\t7254\t0\r\n",
      "NC_007175.2\t7264\t7265\t0\r\n",
      "NC_007175.2\t7284\t7285\t0\r\n",
      "\r\n",
      "==> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t724\t725\t0\r\n",
      "NC_007175.2\t975\t976\t0\r\n",
      "NC_007175.2\t994\t995\t0\r\n",
      "NC_007175.2\t2145\t2146\t0\r\n",
      "NC_007175.2\t2148\t2149\t0\r\n",
      "\r\n",
      "==> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t250559\t250560\t100\r\n",
      "NC_035780.1\t269759\t269760\t100\r\n",
      "NC_035780.1\t269762\t269763\t100\r\n",
      "NC_035780.1\t269781\t269782\t100\r\n",
      "NC_035780.1\t269789\t269790\t100\r\n",
      "NC_035780.1\t269822\t269823\t100\r\n",
      "NC_035780.1\t270312\t270313\t100\r\n",
      "NC_035780.1\t285489\t285490\t100\r\n",
      "NC_035780.1\t285494\t285495\t100\r\n",
      "\r\n",
      "==> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t664\t665\t0\r\n",
      "NC_007175.2\t683\t684\t0\r\n",
      "NC_007175.2\t708\t709\t0\r\n",
      "NC_007175.2\t724\t725\t0\r\n",
      "NC_007175.2\t9943\t9944\t0\r\n",
      "NC_007175.2\t9947\t9948\t0\r\n",
      "NC_007175.2\t9959\t9960\t0\r\n",
      "NC_007175.2\t9964\t9965\t0\r\n",
      "NC_007175.2\t13202\t13203\t0\r\n",
      "\r\n",
      "==> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2301\t2302\t0\r\n",
      "NC_007175.2\t2304\t2305\t0\r\n",
      "NC_007175.2\t2356\t2357\t0\r\n",
      "NC_007175.2\t2365\t2366\t0\r\n",
      "NC_007175.2\t2472\t2473\t0\r\n",
      "NC_007175.2\t2490\t2491\t0\r\n",
      "NC_007175.2\t2635\t2636\t0\r\n",
      "NC_007175.2\t2653\t2654\t0\r\n",
      "NC_007175.2\t2657\t2658\t0\r\n",
      "\r\n",
      "==> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t3099\t3100\t0\r\n",
      "NC_007175.2\t3102\t3103\t0\r\n",
      "NC_007175.2\t3114\t3115\t0\r\n",
      "NC_007175.2\t3213\t3214\t0\r\n",
      "NC_007175.2\t3222\t3223\t0\r\n",
      "NC_007175.2\t3245\t3246\t0\r\n",
      "NC_007175.2\t10997\t10998\t0\r\n",
      "NC_007175.2\t11102\t11103\t0\r\n",
      "NC_007175.2\t11468\t11469\t0\r\n"
     ]
    }
   ],
   "source": [
    "!head *.deduplicated.bedGraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:51.959674Z",
     "start_time": "2018-04-14T13:22:51.560989Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 10 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:52.878847Z",
     "start_time": "2018-04-14T13:22:51.967310Z"
    }
   },
   "outputs": [],
   "source": [
    "!open *report.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T13:22:52.988699Z",
     "start_time": "2018-04-14T13:22:52.885377Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 10 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-16T03:30:59.862210Z",
     "start_time": "2018-04-16T03:30:59.727396Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam\r\n"
     ]
    }
   ],
   "source": [
    "ls *deduplicated.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-04-16T03:31:10.271Z"
    }
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bam2nuc \\\n",
    "--genome_folder /Users/sr320/Dropbox/wd/18-03-15/genome \\\n",
    "*deduplicated.bam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
