{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.631098Z",
     "start_time": "2018-04-14T15:54:07.625549Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "wd=\"/Volumes/Serine/wd/18-07-30\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.764069Z",
     "start_time": "2018-04-14T15:54:07.634174Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: /Volumes/Serine/wd/18-07-30: File exists\r\n"
     ]
    }
   ],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.782218Z",
     "start_time": "2018-04-14T15:54:07.771078Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-07-30\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "            FastQC - A high throughput sequence QC analysis tool\r\n",
      "\r\n",
      "SYNOPSIS\r\n",
      "\r\n",
      "\tfastqc seqfile1 seqfile2 .. seqfileN\r\n",
      "\r\n",
      "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \r\n",
      "           [-c contaminant file] seqfile1 .. seqfileN\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "    FastQC reads a set of sequence files and produces from each one a quality\r\n",
      "    control report consisting of a number of different modules, each one of \r\n",
      "    which will help to identify a different potential type of problem in your\r\n",
      "    data.\r\n",
      "    \r\n",
      "    If no files to process are specified on the command line then the program\r\n",
      "    will start as an interactive graphical application.  If files are provided\r\n",
      "    on the command line then the program will run with no user interaction\r\n",
      "    required.  In this mode it is suitable for inclusion into a standardised\r\n",
      "    analysis pipeline.\r\n",
      "    \r\n",
      "    The options for the program as as follows:\r\n",
      "    \r\n",
      "    -h --help       Print this help file and exit\r\n",
      "    \r\n",
      "    -v --version    Print the version of the program and exit\r\n",
      "    \r\n",
      "    -o --outdir     Create all output files in the specified output directory.\r\n",
      "                    Please note that this directory must exist as the program\r\n",
      "                    will not create it.  If this option is not set then the \r\n",
      "                    output file for each sequence file is created in the same\r\n",
      "                    directory as the sequence file which was processed.\r\n",
      "                    \r\n",
      "    --casava        Files come from raw casava output. Files in the same sample\r\n",
      "                    group (differing only by the group number) will be analysed\r\n",
      "                    as a set rather than individually. Sequences with the filter\r\n",
      "                    flag set in the header will be excluded from the analysis.\r\n",
      "                    Files must have the same names given to them by casava\r\n",
      "                    (including being gzipped and ending with .gz) otherwise they\r\n",
      "                    won't be grouped together correctly.\r\n",
      "                    \r\n",
      "    --nano          Files come from naopore sequences and are in fast5 format. In\r\n",
      "                    this mode you can pass in directories to process and the program\r\n",
      "                    will take in all fast5 files within those directories and produce\r\n",
      "                    a single output file from the sequences found in all files.                    \r\n",
      "                    \r\n",
      "    --nofilter      If running with --casava then don't remove read flagged by\r\n",
      "                    casava as poor quality when performing the QC analysis.\r\n",
      "                   \r\n",
      "    --extract       If set then the zipped output file will be uncompressed in\r\n",
      "                    the same directory after it has been created.  By default\r\n",
      "                    this option will be set if fastqc is run in non-interactive\r\n",
      "                    mode.\r\n",
      "                    \r\n",
      "    -j --java       Provides the full path to the java binary you want to use to\r\n",
      "                    launch fastqc. If not supplied then java is assumed to be in\r\n",
      "                    your path.\r\n",
      "                   \r\n",
      "    --noextract     Do not uncompress the output file after creating it.  You\r\n",
      "                    should set this option if you do not wish to uncompress\r\n",
      "                    the output when running in non-interactive mode.\r\n",
      "                    \r\n",
      "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\r\n",
      "                    show data for every base in the read.  WARNING: Using this\r\n",
      "                    option will cause fastqc to crash and burn if you use it on\r\n",
      "                    really long reads, and your plots may end up a ridiculous size.\r\n",
      "                    You have been warned!\r\n",
      "                    \r\n",
      "    -f --format     Bypasses the normal sequence file format detection and\r\n",
      "                    forces the program to use the specified format.  Valid\r\n",
      "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\r\n",
      "                    \r\n",
      "    -t --threads    Specifies the number of files which can be processed\r\n",
      "                    simultaneously.  Each thread will be allocated 250MB of\r\n",
      "                    memory so you shouldn't run more threads than your\r\n",
      "                    available memory will cope with, and not more than\r\n",
      "                    6 threads on a 32 bit machine\r\n",
      "                  \r\n",
      "    -c              Specifies a non-default file which contains the list of\r\n",
      "    --contaminants  contaminants to screen overrepresented sequences against.\r\n",
      "                    The file must contain sets of named contaminants in the\r\n",
      "                    form name[tab]sequence.  Lines prefixed with a hash will\r\n",
      "                    be ignored.\r\n",
      "\r\n",
      "    -a              Specifies a non-default file which contains the list of\r\n",
      "    --adapters      adapter sequences which will be explicity searched against\r\n",
      "                    the library. The file must contain sets of named adapters\r\n",
      "                    in the form name[tab]sequence.  Lines prefixed with a hash\r\n",
      "                    will be ignored.\r\n",
      "                    \r\n",
      "    -l              Specifies a non-default file which contains a set of criteria\r\n",
      "    --limits        which will be used to determine the warn/error limits for the\r\n",
      "                    various modules.  This file can also be used to selectively \r\n",
      "                    remove some modules from the output all together.  The format\r\n",
      "                    needs to mirror the default limits.txt file found in the\r\n",
      "                    Configuration folder.\r\n",
      "                    \r\n",
      "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\r\n",
      "                    module. Specified Kmer length must be between 2 and 10. Default\r\n",
      "                    length is 7 if not specified.\r\n",
      "                    \r\n",
      "   -q --quiet       Supress all progress messages on stdout and only report errors.\r\n",
      "   \r\n",
      "   -d --dir         Selects a directory to be used for temporary files written when\r\n",
      "                    generating report images. Defaults to system temp directory if\r\n",
      "                    not specified.\r\n",
      "                    \r\n",
      "BUGS\r\n",
      "\r\n",
      "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\r\n",
      "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\r\n",
      "                   \r\n",
      "    "
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/FastQC/fastqc -h\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir qc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Analysis complete for 10_32_S32_L001_R1_001.fastq.gz\n",
      "Started analysis of 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Analysis complete for 10_32_S32_L001_R2_001.fastq.gz\n",
      "Started analysis of 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Analysis complete for 11_37_S37_L001_R1_001.fastq.gz\n",
      "Started analysis of 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Analysis complete for 11_37_S37_L001_R2_001.fastq.gz\n",
      "Started analysis of 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Analysis complete for 12_38_S38_L001_R1_001.fastq.gz\n",
      "Started analysis of 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Analysis complete for 12_38_S38_L001_R2_001.fastq.gz\n",
      "Started analysis of 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Analysis complete for 13_33_S33_L001_R1_001.fastq.gz\n",
      "Started analysis of 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Analysis complete for 13_33_S33_L001_R2_001.fastq.gz\n",
      "Started analysis of 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Analysis complete for 14_34_S34_L001_R1_001.fastq.gz\n",
      "Started analysis of 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Analysis complete for 14_34_S34_L001_R2_001.fastq.gz\n",
      "Started analysis of 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Analysis complete for 15_39_S39_L001_R1_001.fastq.gz\n",
      "Started analysis of 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Analysis complete for 15_39_S39_L001_R2_001.fastq.gz\n",
      "Started analysis of 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Analysis complete for 16_40_S40_L001_R1_001.fastq.gz\n",
      "Started analysis of 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Analysis complete for 16_40_S40_L001_R2_001.fastq.gz\n",
      "Started analysis of 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Analysis complete for 17_9_S9_L001_R1_001.fastq.gz\n",
      "Started analysis of 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Analysis complete for 17_9_S9_L001_R2_001.fastq.gz\n",
      "Started analysis of 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Analysis complete for 18_10_S10_L001_R1_001.fastq.gz\n",
      "Started analysis of 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Analysis complete for 18_10_S10_L001_R2_001.fastq.gz\n",
      "Started analysis of 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Analysis complete for 19_7_S7_L001_R1_001.fastq.gz\n",
      "Started analysis of 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Analysis complete for 19_7_S7_L001_R2_001.fastq.gz\n",
      "Started analysis of 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Analysis complete for 1_25_S25_L001_R1_001.fastq.gz\n",
      "Started analysis of 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Analysis complete for 1_25_S25_L001_R2_001.fastq.gz\n",
      "Started analysis of 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Analysis complete for 20_8_S8_L001_R1_001.fastq.gz\n",
      "Started analysis of 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Analysis complete for 20_8_S8_L001_R2_001.fastq.gz\n",
      "Started analysis of 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Analysis complete for 2_26_S26_L001_R1_001.fastq.gz\n",
      "Started analysis of 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Analysis complete for 2_26_S26_L001_R2_001.fastq.gz\n",
      "Started analysis of 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Analysis complete for 3_27_S27_L001_R1_001.fastq.gz\n",
      "Started analysis of 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Analysis complete for 3_27_S27_L001_R2_001.fastq.gz\n",
      "Started analysis of 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Analysis complete for 4_28_S28_L001_R1_001.fastq.gz\n",
      "Started analysis of 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Analysis complete for 4_28_S28_L001_R2_001.fastq.gz\n",
      "Started analysis of 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Analysis complete for 5_35_S35_L001_R1_001.fastq.gz\n",
      "Started analysis of 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Analysis complete for 5_35_S35_L001_R2_001.fastq.gz\n",
      "Started analysis of 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Analysis complete for 6_36_S36_L001_R1_001.fastq.gz\n",
      "Started analysis of 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Analysis complete for 6_36_S36_L001_R2_001.fastq.gz\n",
      "Started analysis of 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Analysis complete for 7_29_S29_L001_R1_001.fastq.gz\n",
      "Started analysis of 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Analysis complete for 7_29_S29_L001_R2_001.fastq.gz\n",
      "Started analysis of 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Analysis complete for 8_30_S30_L001_R1_001.fastq.gz\n",
      "Started analysis of 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Analysis complete for 8_30_S30_L001_R2_001.fastq.gz\n",
      "Started analysis of 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Analysis complete for 9_31_S31_L001_R1_001.fastq.gz\n",
      "Started analysis of 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for 9_31_S31_L001_R2_001.fastq.gz\n",
      "Analysis complete for 9_31_S31_L001_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/FastQC/fastqc /Volumes/Serine/wd/18-03-14/*.fastq.gz \\\n",
    "--outdir qc/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## --score_min L,0,-0.6 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:05:13.507058Z",
     "start_time": "2018-04-14T15:54:07.786678Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4008\n",
      "Mapping efficiency:\t40.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4574\n",
      "Sequence pairs did not map uniquely:\t1418\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t631\t((converted) top strand)\n",
      "GA/CT/CT:\t1352\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1373\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t652\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2795\n",
      "Mapping efficiency:\t27.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t6247\n",
      "Sequence pairs did not map uniquely:\t958\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t474\t((converted) top strand)\n",
      "GA/CT/CT:\t865\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t936\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t520\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2712\n",
      "Mapping efficiency:\t27.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t6327\n",
      "Sequence pairs did not map uniquely:\t961\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t496\t((converted) top strand)\n",
      "GA/CT/CT:\t928\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t855\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t433\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4245\n",
      "Mapping efficiency:\t42.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4312\n",
      "Sequence pairs did not map uniquely:\t1443\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t760\t((converted) top strand)\n",
      "GA/CT/CT:\t1367\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1387\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t731\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4531\n",
      "Mapping efficiency:\t45.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3905\n",
      "Sequence pairs did not map uniquely:\t1564\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t824\t((converted) top strand)\n",
      "GA/CT/CT:\t1421\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1443\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t843\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2971\n",
      "Mapping efficiency:\t29.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5943\n",
      "Sequence pairs did not map uniquely:\t1086\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t480\t((converted) top strand)\n",
      "GA/CT/CT:\t997\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t998\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t496\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3159\n",
      "Mapping efficiency:\t31.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5784\n",
      "Sequence pairs did not map uniquely:\t1057\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t547\t((converted) top strand)\n",
      "GA/CT/CT:\t999\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1020\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t593\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t2046\n",
      "Mapping efficiency:\t20.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7163\n",
      "Sequence pairs did not map uniquely:\t791\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t275\t((converted) top strand)\n",
      "GA/CT/CT:\t704\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t786\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t281\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3658\n",
      "Mapping efficiency:\t36.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5057\n",
      "Sequence pairs did not map uniquely:\t1285\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t649\t((converted) top strand)\n",
      "GA/CT/CT:\t1239\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1167\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t603\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3154\n",
      "Mapping efficiency:\t31.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5728\n",
      "Sequence pairs did not map uniquely:\t1118\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t566\t((converted) top strand)\n",
      "GA/CT/CT:\t1008\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1071\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t509\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1597\n",
      "Mapping efficiency:\t16.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7531\n",
      "Sequence pairs did not map uniquely:\t872\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t282\t((converted) top strand)\n",
      "GA/CT/CT:\t524\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t511\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t280\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4450\n",
      "Mapping efficiency:\t44.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t3958\n",
      "Sequence pairs did not map uniquely:\t1592\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t765\t((converted) top strand)\n",
      "GA/CT/CT:\t1525\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1487\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t673\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1141\n",
      "Mapping efficiency:\t11.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t8385\n",
      "Sequence pairs did not map uniquely:\t474\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t182\t((converted) top strand)\n",
      "GA/CT/CT:\t379\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t394\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t186\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t230\n",
      "Mapping efficiency:\t2.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t9632\n",
      "Sequence pairs did not map uniquely:\t138\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t25\t((converted) top strand)\n",
      "GA/CT/CT:\t88\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t88\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t29\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1109\n",
      "Mapping efficiency:\t11.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t8480\n",
      "Sequence pairs did not map uniquely:\t411\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t152\t((converted) top strand)\n",
      "GA/CT/CT:\t394\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t423\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t140\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1839\n",
      "Mapping efficiency:\t18.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7508\n",
      "Sequence pairs did not map uniquely:\t653\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t283\t((converted) top strand)\n",
      "GA/CT/CT:\t656\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t622\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t278\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t1922\n",
      "Mapping efficiency:\t19.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7404\n",
      "Sequence pairs did not map uniquely:\t674\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t290\t((converted) top strand)\n",
      "GA/CT/CT:\t670\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t671\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t291\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3584\n",
      "Mapping efficiency:\t35.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4433\n",
      "Sequence pairs did not map uniquely:\t1983\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t719\t((converted) top strand)\n",
      "GA/CT/CT:\t1109\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1063\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t693\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t3226\n",
      "Mapping efficiency:\t32.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4906\n",
      "Sequence pairs did not map uniquely:\t1868\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t618\t((converted) top strand)\n",
      "GA/CT/CT:\t971\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t984\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t653\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t4306\n",
      "Mapping efficiency:\t43.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4186\n",
      "Sequence pairs did not map uniquely:\t1508\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t735\t((converted) top strand)\n",
      "GA/CT/CT:\t1361\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t1380\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t830\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-03-14/*_R1_001.fastq.gz \\\n",
    "| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-p 4 \\\n",
    "-u 10000 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-03-14/{}_L001_R1_001.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-03-14/{}_L001_R2_001.fastq.gz \\\n",
    "2> bismark.err\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:08:52.856840Z",
     "start_time": "2018-04-14T16:08:52.710155Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\r",
      "\r\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\r\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.bam\r\n",
      "bismark.err\r\n"
     ]
    }
   ],
   "source": [
    "ls "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:09:39.164984Z",
     "start_time": "2018-04-14T16:08:52.860760Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/10_32_S32_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/10_32_S32_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/11_37_S37_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/11_37_S37_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/12_38_S38_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/12_38_S38_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/13_33_S33_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/13_33_S33_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/14_34_S34_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/14_34_S34_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/15_39_S39_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/15_39_S39_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/16_40_S40_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/16_40_S40_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/17_9_S9_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/17_9_S9_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/18_10_S10_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/18_10_S10_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/19_7_S7_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/19_7_S7_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 1_25_S25_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/1_25_S25_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/1_25_S25_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 2_26_S26_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/2_26_S26_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/2_26_S26_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 3_27_S27_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/3_27_S27_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/3_27_S27_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 4_28_S28_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/4_28_S28_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/4_28_S28_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 5_35_S35_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/5_35_S35_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/5_35_S35_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 6_36_S36_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/6_36_S36_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/6_36_S36_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 7_29_S29_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/7_29_S29_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/7_29_S29_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 8_30_S30_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/8_30_S30_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/8_30_S30_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 9_31_S31_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-0.6 -p 4 -u 10000 --non_directional -1 /Volumes/Serine/wd/18-03-14/9_31_S31_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/9_31_S31_L001_R2_001.fastq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.584923Z",
     "start_time": "2018-04-14T16:09:39.169294Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-07-30/CpG_OT_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOT_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_CTOB_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-07-30/CpG_OB_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 4 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.826880Z",
     "start_time": "2018-04-14T16:15:33.591262Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.984133Z",
     "start_time": "2018-04-14T16:15:33.832408Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t264571\t264572\t0\r\n",
      "NC_035780.1\t264610\t264611\t100\r\n",
      "NC_035780.1\t264628\t264629\t100\r\n",
      "NC_035780.1\t269387\t269388\t100\r\n",
      "NC_035780.1\t269403\t269404\t100\r\n",
      "NC_035780.1\t269427\t269428\t100\r\n",
      "NC_035780.1\t269437\t269438\t100\r\n",
      "NC_035780.1\t269494\t269495\t100\r\n",
      "NC_035780.1\t269514\t269515\t100\r\n",
      "\r\n",
      "==> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t291108\t291109\t0\r\n",
      "NC_035780.1\t291143\t291144\t0\r\n",
      "NC_035780.1\t291198\t291199\t0\r\n",
      "NC_035780.1\t291211\t291212\t0\r\n",
      "NC_035780.1\t291232\t291233\t0\r\n",
      "NC_035780.1\t320516\t320517\t100\r\n",
      "NC_035780.1\t320524\t320525\t100\r\n",
      "NC_035780.1\t334842\t334843\t100\r\n",
      "NC_035780.1\t334890\t334891\t100\r\n",
      "\r\n",
      "==> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t5039\t5040\t0\r\n",
      "NC_035780.1\t28\t29\t0\r\n",
      "NC_035780.1\t54\t55\t0\r\n",
      "NC_035780.1\t75\t76\t0\r\n",
      "NC_035780.1\t103\t104\t0\r\n",
      "NC_035780.1\t116\t117\t0\r\n",
      "NC_035780.1\t134\t135\t0\r\n",
      "NC_035780.1\t159\t160\t0\r\n",
      "NC_035780.1\t260203\t260204\t100\r\n",
      "\r\n",
      "==> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t6036\t6037\t0\r\n",
      "NC_035780.1\t6109\t6110\t0\r\n",
      "NC_035780.1\t6128\t6129\t0\r\n",
      "NC_035780.1\t6130\t6131\t0\r\n",
      "NC_035780.1\t6609\t6610\t0\r\n",
      "NC_035780.1\t6631\t6632\t0\r\n",
      "NC_035780.1\t6691\t6692\t0\r\n",
      "NC_035780.1\t80897\t80898\t100\r\n",
      "NC_035780.1\t81019\t81020\t100\r\n",
      "\r\n",
      "==> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t52268\t52269\t0\r\n",
      "NC_035780.1\t52286\t52287\t0\r\n",
      "NC_035780.1\t52293\t52294\t0\r\n",
      "NC_035780.1\t52329\t52330\t0\r\n",
      "NC_035780.1\t256446\t256447\t100\r\n",
      "NC_035780.1\t256466\t256467\t100\r\n",
      "NC_035780.1\t260477\t260478\t100\r\n",
      "NC_035780.1\t264649\t264650\t100\r\n",
      "NC_035780.1\t264704\t264705\t100\r\n",
      "\r\n",
      "==> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t12638\t12639\t0\r\n",
      "NC_007175.2\t12677\t12678\t0\r\n",
      "NC_007175.2\t12695\t12696\t0\r\n",
      "NC_007175.2\t12698\t12699\t0\r\n",
      "NC_007175.2\t12732\t12733\t0\r\n",
      "NC_007175.2\t12755\t12756\t0\r\n",
      "NC_007175.2\t12773\t12774\t0\r\n",
      "NC_035780.1\t262024\t262025\t100\r\n",
      "NC_035780.1\t262054\t262055\t100\r\n",
      "\r\n",
      "==> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t15485\t15486\t0\r\n",
      "NC_007175.2\t15508\t15509\t0\r\n",
      "NC_007175.2\t15513\t15514\t0\r\n",
      "NC_007175.2\t15574\t15575\t0\r\n",
      "NC_007175.2\t15592\t15593\t0\r\n",
      "NC_007175.2\t15601\t15602\t0\r\n",
      "NC_007175.2\t15611\t15612\t0\r\n",
      "NC_007175.2\t15622\t15623\t0\r\n",
      "NC_035780.1\t264854\t264855\t100\r\n",
      "\r\n",
      "==> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2237\t2238\t0\r\n",
      "NC_007175.2\t2265\t2266\t0\r\n",
      "NC_007175.2\t2282\t2283\t0\r\n",
      "NC_007175.2\t3886\t3887\t0\r\n",
      "NC_007175.2\t3908\t3909\t0\r\n",
      "NC_007175.2\t3933\t3934\t0\r\n",
      "NC_007175.2\t3944\t3945\t0\r\n",
      "NC_007175.2\t3946\t3947\t0\r\n",
      "NC_007175.2\t8591\t8592\t0\r\n",
      "\r\n",
      "==> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t1749\t1750\t0\r\n",
      "NC_007175.2\t1766\t1767\t0\r\n",
      "NC_007175.2\t1825\t1826\t0\r\n",
      "NC_007175.2\t1826\t1827\t0\r\n",
      "NC_007175.2\t1836\t1837\t0\r\n",
      "NC_007175.2\t1837\t1838\t0\r\n",
      "NC_007175.2\t1842\t1843\t0\r\n",
      "NC_007175.2\t1843\t1844\t0\r\n",
      "NC_007175.2\t1846\t1847\t0\r\n",
      "\r\n",
      "==> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2300\t2301\t0\r\n",
      "NC_007175.2\t2303\t2304\t0\r\n",
      "NC_007175.2\t2355\t2356\t0\r\n",
      "NC_007175.2\t2364\t2365\t0\r\n",
      "NC_007175.2\t2433\t2434\t0\r\n",
      "NC_007175.2\t2434\t2435\t0\r\n",
      "NC_007175.2\t2471\t2472\t0\r\n",
      "NC_007175.2\t2472\t2473\t0\r\n",
      "NC_007175.2\t2490\t2491\t0\r\n",
      "\r\n",
      "==> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t366\t367\t0\r\n",
      "NC_007175.2\t976\t977\t0\r\n",
      "NC_007175.2\t995\t996\t0\r\n",
      "NC_007175.2\t1010\t1011\t0\r\n",
      "NC_007175.2\t1016\t1017\t0\r\n",
      "NC_007175.2\t1019\t1020\t0\r\n",
      "\r\n",
      "==> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t254242\t254243\t100\r\n",
      "NC_035780.1\t254295\t254296\t100\r\n",
      "NC_035780.1\t254312\t254313\t100\r\n",
      "NC_035780.1\t254334\t254335\t100\r\n",
      "NC_035780.1\t268595\t268596\t100\r\n",
      "NC_035780.1\t268659\t268660\t100\r\n",
      "NC_035780.1\t270866\t270867\t0\r\n",
      "NC_035780.1\t270943\t270944\t100\r\n",
      "NC_035780.1\t270974\t270975\t100\r\n",
      "\r\n",
      "==> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t331\t332\t0\r\n",
      "NC_007175.2\t366\t367\t0\r\n",
      "NC_007175.2\t5697\t5698\t0\r\n",
      "NC_007175.2\t5724\t5725\t0\r\n",
      "NC_007175.2\t6933\t6934\t0\r\n",
      "\r\n",
      "==> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t8386\t8387\t0\r\n",
      "NC_007175.2\t8407\t8408\t0\r\n",
      "NC_007175.2\t8591\t8592\t0\r\n",
      "NC_007175.2\t8599\t8600\t0\r\n",
      "NC_007175.2\t8624\t8625\t0\r\n",
      "NC_007175.2\t8651\t8652\t0\r\n",
      "NC_035780.1\t39621\t39622\t0\r\n",
      "NC_035780.1\t39640\t39641\t0\r\n",
      "NC_035780.1\t39643\t39644\t0\r\n",
      "\r\n",
      "==> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t994\t995\t0\r\n",
      "NC_007175.2\t1009\t1010\t0\r\n",
      "NC_007175.2\t1015\t1016\t0\r\n",
      "NC_007175.2\t1018\t1019\t0\r\n",
      "NC_007175.2\t1036\t1037\t0\r\n",
      "NC_007175.2\t5040\t5041\t0\r\n",
      "NC_007175.2\t5171\t5172\t0\r\n",
      "NC_007175.2\t10392\t10393\t0\r\n",
      "NC_007175.2\t13583\t13584\t100\r\n",
      "\r\n",
      "==> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t1009\t1010\t0\r\n",
      "NC_007175.2\t1015\t1016\t0\r\n",
      "NC_007175.2\t1018\t1019\t0\r\n",
      "NC_007175.2\t1036\t1037\t0\r\n",
      "NC_007175.2\t1097\t1098\t0\r\n",
      "NC_007175.2\t1106\t1107\t0\r\n",
      "NC_007175.2\t8749\t8750\t0\r\n",
      "NC_007175.2\t10817\t10818\t0\r\n",
      "NC_007175.2\t10826\t10827\t0\r\n",
      "\r\n",
      "==> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t6357\t6358\t0\r\n",
      "NC_007175.2\t6372\t6373\t0\r\n",
      "NC_007175.2\t12566\t12567\t0\r\n",
      "NC_007175.2\t12590\t12591\t0\r\n",
      "NC_007175.2\t17045\t17046\t0\r\n",
      "NC_007175.2\t17086\t17087\t0\r\n",
      "NC_007175.2\t17120\t17121\t0\r\n",
      "NC_007175.2\t17130\t17131\t0\r\n",
      "NC_007175.2\t17164\t17165\t0\r\n",
      "\r\n",
      "==> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2148\t2149\t0\r\n",
      "NC_007175.2\t2180\t2181\t0\r\n",
      "NC_007175.2\t2206\t2207\t0\r\n",
      "NC_007175.2\t2237\t2238\t0\r\n",
      "NC_007175.2\t2265\t2266\t0\r\n",
      "NC_007175.2\t2282\t2283\t0\r\n",
      "NC_007175.2\t2304\t2305\t0\r\n",
      "NC_007175.2\t2356\t2357\t0\r\n",
      "NC_007175.2\t2365\t2366\t0\r\n",
      "\r\n",
      "==> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t2181\t2182\t0\r\n",
      "NC_007175.2\t2207\t2208\t0\r\n",
      "NC_007175.2\t2238\t2239\t0\r\n",
      "NC_007175.2\t2266\t2267\t0\r\n",
      "NC_007175.2\t2283\t2284\t0\r\n",
      "NC_007175.2\t2301\t2302\t0\r\n",
      "NC_007175.2\t2768\t2769\t0\r\n",
      "NC_007175.2\t2815\t2816\t0\r\n",
      "NC_007175.2\t2997\t2998\t0\r\n",
      "\r\n",
      "==> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_035780.1\t166289\t166290\t0\r\n",
      "NC_035780.1\t166291\t166292\t100\r\n",
      "NC_035780.1\t261705\t261706\t100\r\n",
      "NC_035780.1\t261715\t261716\t100\r\n",
      "NC_035780.1\t261726\t261727\t100\r\n",
      "NC_035780.1\t261738\t261739\t100\r\n",
      "NC_035780.1\t261743\t261744\t100\r\n",
      "NC_035780.1\t261977\t261978\t100\r\n",
      "NC_035780.1\t261992\t261993\t100\r\n"
     ]
    }
   ],
   "source": [
    "!head *.deduplicated.bedGraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.611600Z",
     "start_time": "2018-04-14T16:15:33.997767Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 20 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 1_25_S25_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2_26_S26_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 3_27_S27_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 4_28_S28_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 5_35_S35_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 6_36_S36_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 7_29_S29_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 8_30_S30_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 9_31_S31_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.753123Z",
     "start_time": "2018-04-14T16:15:34.615422Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!#open *report.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.842160Z",
     "start_time": "2018-04-14T16:15:34.758529Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 20 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
