{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# Set working directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.631098Z",
     "start_time": "2018-04-14T15:54:07.625549Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "wd=\"/Volumes/Serine/wd/18-09-14\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.764069Z",
     "start_time": "2018-04-14T15:54:07.634174Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:54:07.782218Z",
     "start_time": "2018-04-14T15:54:07.771078Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-09-14\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "# --score_min L,0,-1.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:05:13.507058Z",
     "start_time": "2018-04-14T15:54:07.786678Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t323356\n",
      "Mapping efficiency:\t56.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t149319\n",
      "Sequence pairs did not map uniquely:\t101671\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t54711\t((converted) top strand)\n",
      "GA/CT/CT:\t106412\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t107961\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t54272\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t299349\n",
      "Mapping efficiency:\t43.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t306265\n",
      "Sequence pairs did not map uniquely:\t89948\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t55952\t((converted) top strand)\n",
      "GA/CT/CT:\t93261\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t94781\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t55355\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t367271\n",
      "Mapping efficiency:\t42.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t384879\n",
      "Sequence pairs did not map uniquely:\t113119\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t67043\t((converted) top strand)\n",
      "GA/CT/CT:\t115426\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t117017\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t67785\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t375006\n",
      "Mapping efficiency:\t58.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t153258\n",
      "Sequence pairs did not map uniquely:\t117204\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t66643\t((converted) top strand)\n",
      "GA/CT/CT:\t120427\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t121542\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t66394\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t416742\n",
      "Mapping efficiency:\t60.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t145519\n",
      "Sequence pairs did not map uniquely:\t129365\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t76891\t((converted) top strand)\n",
      "GA/CT/CT:\t130155\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t132244\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t77452\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t304399\n",
      "Mapping efficiency:\t45.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t272185\n",
      "Sequence pairs did not map uniquely:\t94741\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t54475\t((converted) top strand)\n",
      "GA/CT/CT:\t97098\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t97331\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t55494\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t359456\n",
      "Mapping efficiency:\t46.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t303130\n",
      "Sequence pairs did not map uniquely:\t109219\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t64637\t((converted) top strand)\n",
      "GA/CT/CT:\t114100\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t115419\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t65300\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t114320\n",
      "Mapping efficiency:\t28.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t252142\n",
      "Sequence pairs did not map uniquely:\t37399\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t17173\t((converted) top strand)\n",
      "GA/CT/CT:\t39759\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t40124\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t17264\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t273294\n",
      "Mapping efficiency:\t52.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t166917\n",
      "Sequence pairs did not map uniquely:\t85210\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t44211\t((converted) top strand)\n",
      "GA/CT/CT:\t92806\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t92760\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t43517\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t177077\n",
      "Mapping efficiency:\t47.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t137934\n",
      "Sequence pairs did not map uniquely:\t56656\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t28841\t((converted) top strand)\n",
      "GA/CT/CT:\t59689\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t59619\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t28927\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t125064\n",
      "Mapping efficiency:\t25.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t306217\n",
      "Sequence pairs did not map uniquely:\t56856\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t23786\t((converted) top strand)\n",
      "GA/CT/CT:\t39167\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t38583\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t23528\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t188633\n",
      "Mapping efficiency:\t59.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t66890\n",
      "Sequence pairs did not map uniquely:\t59399\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t31206\t((converted) top strand)\n",
      "GA/CT/CT:\t63033\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t63329\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t31065\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t33556\n",
      "Mapping efficiency:\t19.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t124065\n",
      "Sequence pairs did not map uniquely:\t11333\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t5883\t((converted) top strand)\n",
      "GA/CT/CT:\t10912\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t11002\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t5759\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t12674\n",
      "Mapping efficiency:\t5.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t214472\n",
      "Sequence pairs did not map uniquely:\t4506\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2444\t((converted) top strand)\n",
      "GA/CT/CT:\t3819\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3904\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2507\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t54588\n",
      "Mapping efficiency:\t19.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t207896\n",
      "Sequence pairs did not map uniquely:\t16581\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t8317\t((converted) top strand)\n",
      "GA/CT/CT:\t19003\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t18858\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t8410\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t6286\n",
      "Mapping efficiency:\t28.8%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t13668\n",
      "Sequence pairs did not map uniquely:\t1908\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1001\t((converted) top strand)\n",
      "GA/CT/CT:\t2227\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t2103\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t955\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t9509\n",
      "Mapping efficiency:\t30.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t18946\n",
      "Sequence pairs did not map uniquely:\t2897\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t1454\t((converted) top strand)\n",
      "GA/CT/CT:\t3261\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3288\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t1506\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t296375\n",
      "Mapping efficiency:\t48.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t163636\n",
      "Sequence pairs did not map uniquely:\t152983\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t60533\t((converted) top strand)\n",
      "GA/CT/CT:\t88418\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t87476\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t59948\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t319082\n",
      "Mapping efficiency:\t48.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t185187\n",
      "Sequence pairs did not map uniquely:\t160964\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t64700\t((converted) top strand)\n",
      "GA/CT/CT:\t95484\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t95019\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t63879\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t348584\n",
      "Mapping efficiency:\t58.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t137781\n",
      "Sequence pairs did not map uniquely:\t109081\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t63310\t((converted) top strand)\n",
      "GA/CT/CT:\t110969\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t110881\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t63424\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-03-14/*_R1_001.fastq.gz \\\n",
    "| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-03-14/{}_L001_R1_001.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-03-14/{}_L001_R2_001.fastq.gz \\\n",
    "2> bismark.err\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:09:39.164984Z",
     "start_time": "2018-04-14T16:08:52.860760Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/10_32_S32_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/10_32_S32_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/11_37_S37_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/11_37_S37_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/12_38_S38_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/12_38_S38_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/13_33_S33_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/13_33_S33_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/14_34_S34_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/14_34_S34_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/15_39_S39_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/15_39_S39_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/16_40_S40_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/16_40_S40_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/17_9_S9_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/17_9_S9_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/18_10_S10_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/18_10_S10_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/19_7_S7_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/19_7_S7_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 1_25_S25_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/1_25_S25_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/1_25_S25_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 2_26_S26_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/2_26_S26_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/2_26_S26_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 3_27_S27_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/3_27_S27_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/3_27_S27_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 4_28_S28_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/4_28_S28_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/4_28_S28_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 5_35_S35_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/5_35_S35_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/5_35_S35_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 6_36_S36_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/6_36_S36_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/6_36_S36_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 7_29_S29_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/7_29_S29_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/7_29_S29_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 8_30_S30_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/8_30_S30_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/8_30_S30_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 9_31_S31_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-03-14/9_31_S31_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-03-14/9_31_S31_L001_R2_001.fastq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.584923Z",
     "start_time": "2018-04-14T16:09:39.169294Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-14/CpG_OT_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOT_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_CTOB_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-14/CpG_OB_9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 4 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.826880Z",
     "start_time": "2018-04-14T16:15:33.591262Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:33.984133Z",
     "start_time": "2018-04-14T16:15:33.832408Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t473\t474\t0\r\n",
      "NC_007175.2\t665\t666\t0\r\n",
      "NC_007175.2\t684\t685\t0\r\n",
      "NC_007175.2\t709\t710\t0\r\n",
      "NC_007175.2\t725\t726\t0\r\n",
      "NC_007175.2\t975\t976\t0\r\n",
      "NC_007175.2\t1009\t1010\t0\r\n",
      "NC_007175.2\t1015\t1016\t0\r\n",
      "NC_007175.2\t1018\t1019\t0\r\n",
      "\r\n",
      "==> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t664\t665\t0\r\n",
      "NC_007175.2\t683\t684\t0\r\n",
      "NC_007175.2\t708\t709\t0\r\n",
      "NC_007175.2\t724\t725\t0\r\n",
      "NC_007175.2\t1036\t1037\t0\r\n",
      "NC_007175.2\t1097\t1098\t0\r\n",
      "NC_007175.2\t1106\t1107\t0\r\n",
      "NC_007175.2\t1124\t1125\t0\r\n",
      "NC_007175.2\t2128\t2129\t0\r\n",
      "\r\n",
      "==> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t591\t592\t0\r\n",
      "NC_007175.2\t870\t871\t0\r\n",
      "NC_007175.2\t2993\t2994\t0\r\n",
      "NC_007175.2\t2997\t2998\t0\r\n",
      "NC_007175.2\t3021\t3022\t0\r\n",
      "NC_007175.2\t3029\t3030\t0\r\n",
      "NC_007175.2\t3059\t3060\t0\r\n",
      "NC_007175.2\t3062\t3063\t0\r\n",
      "NC_007175.2\t3213\t3214\t0\r\n",
      "\r\n",
      "==> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t331\t332\t0\r\n",
      "NC_007175.2\t995\t996\t0\r\n",
      "NC_007175.2\t1010\t1011\t0\r\n",
      "NC_007175.2\t1016\t1017\t0\r\n",
      "NC_007175.2\t1019\t1020\t0\r\n",
      "NC_007175.2\t1037\t1038\t0\r\n",
      "NC_007175.2\t1227\t1228\t0\r\n",
      "NC_007175.2\t3222\t3223\t0\r\n",
      "\r\n",
      "==> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t331\t332\t0\r\n",
      "NC_007175.2\t366\t367\t0\r\n",
      "NC_007175.2\t472\t473\t0\r\n",
      "\r\n",
      "==> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t591\t592\t0\r\n",
      "NC_007175.2\t975\t976\t0\r\n",
      "\r\n",
      "==> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "\r\n",
      "==> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t193\t194\t0\r\n",
      "NC_007175.2\t246\t247\t0\r\n",
      "NC_007175.2\t257\t258\t0\r\n",
      "NC_007175.2\t264\t265\t0\r\n",
      "\r\n",
      "==> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t49\t50\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t51\t52\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t88\t89\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t147\t148\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "\r\n",
      "==> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t49\t50\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t51\t52\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t88\t89\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t147\t148\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "\r\n",
      "==> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t88\t89\t0\r\n",
      "NC_007175.2\t147\t148\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t193\t194\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t366\t367\t0\r\n",
      "NC_007175.2\t472\t473\t0\r\n",
      "\r\n",
      "==> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t3474\t3475\t0\r\n",
      "NC_007175.2\t3599\t3600\t0\r\n",
      "NC_007175.2\t4884\t4885\t0\r\n",
      "NC_007175.2\t4950\t4951\t0\r\n",
      "NC_007175.2\t4967\t4968\t0\r\n",
      "NC_007175.2\t5039\t5040\t0\r\n",
      "NC_007175.2\t5065\t5066\t0\r\n",
      "NC_007175.2\t5082\t5083\t0\r\n",
      "NC_007175.2\t7713\t7714\t0\r\n",
      "\r\n",
      "==> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t245\t246\t0\r\n",
      "NC_007175.2\t256\t257\t0\r\n",
      "NC_007175.2\t263\t264\t0\r\n",
      "NC_007175.2\t265\t266\t0\r\n",
      "NC_007175.2\t331\t332\t0\r\n",
      "NC_007175.2\t366\t367\t0\r\n",
      "NC_007175.2\t709\t710\t0\r\n",
      "\r\n",
      "==> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t1825\t1826\t0\r\n",
      "NC_007175.2\t1836\t1837\t0\r\n",
      "NC_007175.2\t1842\t1843\t0\r\n",
      "NC_007175.2\t1846\t1847\t0\r\n",
      "NC_007175.2\t1918\t1919\t0\r\n",
      "NC_007175.2\t2206\t2207\t0\r\n",
      "NC_007175.2\t2237\t2238\t0\r\n",
      "NC_007175.2\t2265\t2266\t0\r\n",
      "NC_007175.2\t3714\t3715\t0\r\n",
      "\r\n",
      "==> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t592\t593\t0\r\n",
      "NC_007175.2\t665\t666\t0\r\n",
      "NC_007175.2\t684\t685\t0\r\n",
      "NC_007175.2\t709\t710\t0\r\n",
      "NC_007175.2\t725\t726\t0\r\n",
      "NC_007175.2\t856\t857\t0\r\n",
      "\r\n",
      "==> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t1009\t1010\t0\r\n",
      "NC_007175.2\t1015\t1016\t0\r\n",
      "NC_007175.2\t1018\t1019\t0\r\n",
      "NC_007175.2\t1036\t1037\t0\r\n",
      "NC_007175.2\t1097\t1098\t0\r\n",
      "NC_007175.2\t1106\t1107\t0\r\n",
      "NC_007175.2\t1581\t1582\t0\r\n",
      "NC_007175.2\t1598\t1599\t0\r\n",
      "NC_007175.2\t1750\t1751\t0\r\n",
      "\r\n",
      "==> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t1439\t1440\t0\r\n",
      "NC_007175.2\t1444\t1445\t0\r\n",
      "NC_007175.2\t1456\t1457\t0\r\n",
      "NC_007175.2\t1510\t1511\t0\r\n",
      "NC_007175.2\t1528\t1529\t0\r\n",
      "NC_007175.2\t1580\t1581\t0\r\n",
      "NC_007175.2\t1581\t1582\t0\r\n",
      "NC_007175.2\t1597\t1598\t0\r\n",
      "NC_007175.2\t1598\t1599\t0\r\n",
      "\r\n",
      "==> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t591\t592\t0\r\n",
      "NC_007175.2\t664\t665\t0\r\n",
      "NC_007175.2\t665\t666\t0\r\n",
      "NC_007175.2\t683\t684\t0\r\n",
      "NC_007175.2\t684\t685\t0\r\n",
      "\r\n",
      "==> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t48\t49\t0\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t88\t89\t0\r\n",
      "NC_007175.2\t146\t147\t0\r\n",
      "NC_007175.2\t147\t148\t0\r\n",
      "NC_007175.2\t246\t247\t0\r\n",
      "NC_007175.2\t257\t258\t0\r\n",
      "NC_007175.2\t264\t265\t0\r\n",
      "\r\n",
      "==> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bedGraph <==\r\n",
      "track type=bedGraph\r\n",
      "NC_007175.2\t50\t51\t0\r\n",
      "NC_007175.2\t87\t88\t0\r\n",
      "NC_007175.2\t192\t193\t0\r\n",
      "NC_007175.2\t331\t332\t0\r\n",
      "NC_007175.2\t1018\t1019\t0\r\n",
      "NC_007175.2\t1036\t1037\t0\r\n",
      "NC_007175.2\t1097\t1098\t0\r\n",
      "NC_007175.2\t1106\t1107\t0\r\n",
      "NC_007175.2\t1124\t1125\t0\r\n"
     ]
    }
   ],
   "source": [
    "!head *.deduplicated.bedGraph"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.611600Z",
     "start_time": "2018-04-14T16:15:33.997767Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 20 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 1_25_S25_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 1_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 2_26_S26_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 2_26_S26_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 3_27_S27_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 3_27_S27_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 4_28_S28_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 4_28_S28_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 5_35_S35_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 5_35_S35_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 6_36_S36_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 6_36_S36_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 7_29_S29_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 7_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 8_30_S30_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 8_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 9_31_S31_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 9_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.753123Z",
     "start_time": "2018-04-14T16:15:34.615422Z"
    },
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!#open *report.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T16:15:34.842160Z",
     "start_time": "2018-04-14T16:15:34.758529Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 20 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 1_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 2_26_S26_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 3_27_S27_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 4_28_S28_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 5_35_S35_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 6_36_S36_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 7_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 8_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 9_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "![img]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
