{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-03-14/10_32_S32_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/10_32_S32_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/11_37_S37_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/11_37_S37_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/12_38_S38_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/12_38_S38_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/13_33_S33_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/13_33_S33_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/14_34_S34_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/14_34_S34_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/15_39_S39_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/15_39_S39_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/16_40_S40_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/16_40_S40_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/17_9_S9_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/17_9_S9_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/18_10_S10_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/18_10_S10_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/19_7_S7_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/19_7_S7_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/1_25_S25_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/1_25_S25_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/20_8_S8_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/20_8_S8_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/2_26_S26_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/2_26_S26_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/3_27_S27_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/3_27_S27_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/4_28_S28_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/4_28_S28_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/5_35_S35_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/5_35_S35_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/6_36_S36_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/6_36_S36_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/7_29_S29_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/7_29_S29_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/8_30_S30_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/8_30_S30_L001_R2_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/9_31_S31_L001_R1_001.fastq.gz\r\n",
      "/Volumes/Serine/wd/18-03-14/9_31_S31_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls /Volumes/Serine/wd/18-03-14/*fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cp /Volumes/Serine/wd/18-03-14/*fastq.gz /Volumes/Serine/wd/18-09-15/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cp /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R1_001.fastq.gz ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "cp /Volumes/Serine/wd/18-03-14/20_8_S8_L001_R2_001.fastq.gz ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 1957488\r\n",
      "-rw-r--r--  1 sr320  staff    29M Sep 14 12:55 10_32_S32_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    28M Sep 14 12:55 10_32_S32_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    35M Sep 14 12:55 11_37_S37_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    35M Sep 14 12:55 11_37_S37_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    43M Sep 14 12:55 12_38_S38_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    43M Sep 14 12:55 12_38_S38_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    32M Sep 14 12:55 13_33_S33_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    32M Sep 14 12:55 13_33_S33_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    34M Sep 14 12:55 14_34_S34_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    35M Sep 14 12:55 14_34_S34_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    33M Sep 14 12:55 15_39_S39_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    34M Sep 14 12:55 15_39_S39_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    38M Sep 14 12:55 16_40_S40_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    39M Sep 14 12:55 16_40_S40_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    20M Sep 14 12:55 17_9_S9_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    21M Sep 14 12:55 17_9_S9_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    26M Sep 14 12:55 18_10_S10_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    27M Sep 14 12:55 18_10_S10_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    18M Sep 14 12:55 19_7_S7_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    18M Sep 14 12:55 19_7_S7_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    25M Sep 14 12:55 1_25_S25_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    24M Sep 14 12:55 1_25_S25_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    16M Sep 14 12:55 20_8_S8_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    15M Sep 14 12:55 20_8_S8_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   8.7M Sep 14 12:55 2_26_S26_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   8.6M Sep 14 12:55 2_26_S26_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    12M Sep 14 12:55 3_27_S27_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    12M Sep 14 12:55 3_27_S27_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    14M Sep 14 12:55 4_28_S28_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    14M Sep 14 12:55 4_28_S28_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   1.1M Sep 14 12:55 5_35_S35_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   1.2M Sep 14 12:55 5_35_S35_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   1.6M Sep 14 12:55 6_36_S36_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff   1.7M Sep 14 12:55 6_36_S36_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    30M Sep 14 12:55 7_29_S29_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    29M Sep 14 12:55 7_29_S29_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--@ 1 sr320  staff    32M Sep 14 12:55 8_30_S30_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    31M Sep 14 12:55 8_30_S30_L001_R2_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    30M Sep 14 12:55 9_31_S31_L001_R1_001.fastq.gz\r\n",
      "-rw-r--r--  1 sr320  staff    29M Sep 14 12:55 9_31_S31_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls -lh /Volumes/Serine/wd/18-09-15/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-09-15\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-09-15/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10_32_S32_L001_R1_001.fastq.gz  17_9_S9_L001_R1_001.fastq.gz\r\n",
      "10_32_S32_L001_R2_001.fastq.gz  17_9_S9_L001_R2_001.fastq.gz\r\n",
      "11_37_S37_L001_R1_001.fastq.gz  18_10_S10_L001_R1_001.fastq.gz\r\n",
      "11_37_S37_L001_R2_001.fastq.gz  18_10_S10_L001_R2_001.fastq.gz\r\n",
      "12_38_S38_L001_R1_001.fastq.gz  19_7_S7_L001_R1_001.fastq.gz\r\n",
      "12_38_S38_L001_R2_001.fastq.gz  19_7_S7_L001_R2_001.fastq.gz\r\n",
      "13_33_S33_L001_R1_001.fastq.gz  1_25_S25_L001_R1_001.fastq.gz\r\n",
      "13_33_S33_L001_R2_001.fastq.gz  1_25_S25_L001_R2_001.fastq.gz\r\n",
      "14_34_S34_L001_R1_001.fastq.gz  7_29_S29_L001_R1_001.fastq.gz\r\n",
      "14_34_S34_L001_R2_001.fastq.gz  7_29_S29_L001_R2_001.fastq.gz\r\n",
      "15_39_S39_L001_R1_001.fastq.gz  8_30_S30_L001_R1_001.fastq.gz\r\n",
      "15_39_S39_L001_R2_001.fastq.gz  8_30_S30_L001_R2_001.fastq.gz\r\n",
      "16_40_S40_L001_R1_001.fastq.gz  9_31_S31_L001_R1_001.fastq.gz\r\n",
      "16_40_S40_L001_R2_001.fastq.gz  9_31_S31_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Volumes/Serine/wd/18-09-15'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t25591\n",
      "Mapping efficiency:\t25.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t62909\n",
      "Sequence pairs did not map uniquely:\t11500\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t4891\t((converted) top strand)\n",
      "GA/CT/CT:\t7949\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t7895\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t4856\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t48364\n",
      "Mapping efficiency:\t48.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t26737\n",
      "Sequence pairs did not map uniquely:\t24899\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t9934\t((converted) top strand)\n",
      "GA/CT/CT:\t14333\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t14428\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t9669\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t47749\n",
      "Mapping efficiency:\t47.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t27914\n",
      "Sequence pairs did not map uniquely:\t24337\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t9636\t((converted) top strand)\n",
      "GA/CT/CT:\t14219\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t14282\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t9612\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t58526\n",
      "Mapping efficiency:\t58.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t23071\n",
      "Sequence pairs did not map uniquely:\t18403\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t10627\t((converted) top strand)\n",
      "GA/CT/CT:\t18434\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t18823\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t10642\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t56247\n",
      "Mapping efficiency:\t56.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t26002\n",
      "Sequence pairs did not map uniquely:\t17751\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t9452\t((converted) top strand)\n",
      "GA/CT/CT:\t18472\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t18925\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t9398\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t43201\n",
      "Mapping efficiency:\t43.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t43975\n",
      "Sequence pairs did not map uniquely:\t12824\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t8005\t((converted) top strand)\n",
      "GA/CT/CT:\t13511\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t13617\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t8068\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t42415\n",
      "Mapping efficiency:\t42.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t44634\n",
      "Sequence pairs did not map uniquely:\t12951\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t7748\t((converted) top strand)\n",
      "GA/CT/CT:\t13295\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t13554\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t7818\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t57923\n",
      "Mapping efficiency:\t57.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t23669\n",
      "Sequence pairs did not map uniquely:\t18408\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t10236\t((converted) top strand)\n",
      "GA/CT/CT:\t18568\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t18825\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t10294\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t60204\n",
      "Mapping efficiency:\t60.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t21032\n",
      "Sequence pairs did not map uniquely:\t18764\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t10952\t((converted) top strand)\n",
      "GA/CT/CT:\t18904\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t18998\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t11350\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t45282\n",
      "Mapping efficiency:\t45.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t40797\n",
      "Sequence pairs did not map uniquely:\t13921\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t8223\t((converted) top strand)\n",
      "GA/CT/CT:\t14384\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t14464\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t8210\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t46355\n",
      "Mapping efficiency:\t46.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t39614\n",
      "Sequence pairs did not map uniquely:\t14031\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t8438\t((converted) top strand)\n",
      "GA/CT/CT:\t14633\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t14780\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t8504\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t28202\n",
      "Mapping efficiency:\t28.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t62587\n",
      "Sequence pairs did not map uniquely:\t9211\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t4299\t((converted) top strand)\n",
      "GA/CT/CT:\t9655\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t9923\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t4325\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t51876\n",
      "Mapping efficiency:\t51.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t31835\n",
      "Sequence pairs did not map uniquely:\t16289\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t8436\t((converted) top strand)\n",
      "GA/CT/CT:\t17594\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t17569\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t8277\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t47596\n",
      "Mapping efficiency:\t47.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t37180\n",
      "Sequence pairs did not map uniquely:\t15224\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t7803\t((converted) top strand)\n",
      "GA/CT/CT:\t16034\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t15900\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t7859\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t59730\n",
      "Mapping efficiency:\t59.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t21260\n",
      "Sequence pairs did not map uniquely:\t19010\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t10003\t((converted) top strand)\n",
      "GA/CT/CT:\t19971\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t19880\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t9876\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-09-15/*_R1_001.fastq.gz \\\n",
    "| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "-u 100000 \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-09-15/{}_L001_R1_001.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-09-15/{}_L001_R2_001.fastq.gz \\\n",
    "2> bismark.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 01_25_S25_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/01_25_S25_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/01_25_S25_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 07_29_S29_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/07_29_S29_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/07_29_S29_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 08_30_S30_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/08_30_S30_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/08_30_S30_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 09_31_S31_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/09_31_S31_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/09_31_S31_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/10_32_S32_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/10_32_S32_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/11_37_S37_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/11_37_S37_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/12_38_S38_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/12_38_S38_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/13_33_S33_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/13_33_S33_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/14_34_S34_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/14_34_S34_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/15_39_S39_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/15_39_S39_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/16_40_S40_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/16_40_S40_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/17_9_S9_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/17_9_S9_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/18_10_S10_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/18_10_S10_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/19_7_S7_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/19_7_S7_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 100000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/20_8_S8_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/20_8_S8_L001_R2_001.fastq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-15/CpG_OT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOT_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_CTOB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-15/CpG_OB_20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 4 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 15 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> 01_25_S25_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 01_25_S25_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 01_25_S25_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 07_29_S29_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 07_29_S29_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 07_29_S29_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 08_30_S30_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 08_30_S30_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 08_30_S30_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 09_31_S31_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 09_31_S31_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 09_31_S31_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 15 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: 01_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 07_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 08_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 09_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 15 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: 01_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 07_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 08_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 09_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 10_32_S32_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 11_37_S37_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 12_38_S38_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 13_33_S33_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 15_39_S39_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 16_40_S40_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 17_9_S9_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 19_7_S7_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 20_8_S8_L001_R1_001_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.20.0/bismark2summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![img](img/q-report4.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "looking at coverage files     \n",
    "`<chromosome> <start position> <end position> <methylation percentage> <count methylated> <count unmethylated>\n",
    "`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_007175.2\t1826\t1826\t0\t0\t1\r\n",
      "NC_007175.2\t1837\t1837\t0\t0\t1\r\n",
      "NC_007175.2\t1843\t1843\t0\t0\t1\r\n",
      "NC_007175.2\t1847\t1847\t0\t0\t1\r\n",
      "NC_007175.2\t1855\t1855\t0\t0\t1\r\n",
      "NC_007175.2\t1874\t1874\t0\t0\t1\r\n",
      "NC_007175.2\t1919\t1919\t0\t0\t1\r\n",
      "NC_007175.2\t2149\t2149\t0\t0\t14\r\n",
      "NC_007175.2\t2181\t2181\t0\t0\t14\r\n",
      "NC_007175.2\t2182\t2182\t0\t0\t12\r\n",
      "NC_007175.2\t2207\t2207\t0\t0\t13\r\n",
      "NC_007175.2\t2208\t2208\t0\t0\t13\r\n",
      "NC_007175.2\t2238\t2238\t0\t0\t14\r\n",
      "NC_007175.2\t2239\t2239\t0\t0\t13\r\n",
      "NC_007175.2\t2266\t2266\t0\t0\t10\r\n",
      "NC_007175.2\t2267\t2267\t0\t0\t12\r\n",
      "NC_007175.2\t2283\t2283\t0\t0\t9\r\n",
      "NC_007175.2\t2284\t2284\t0\t0\t12\r\n",
      "NC_007175.2\t2302\t2302\t9.09090909090909\t1\t10\r\n",
      "NC_007175.2\t2305\t2305\t50\t1\t1\r\n"
     ]
    }
   ],
   "source": [
    "!head -20 /Volumes/Serine/wd/18-09-15/07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_007175.2 1826 1\r\n",
      "NC_007175.2 1837 1\r\n",
      "NC_007175.2 1843 1\r\n",
      "NC_007175.2 1847 1\r\n",
      "NC_007175.2 1855 1\r\n",
      "NC_007175.2 1874 1\r\n",
      "NC_007175.2 1919 1\r\n",
      "NC_007175.2 2149 14\r\n",
      "NC_007175.2 2181 14\r\n",
      "NC_007175.2 2182 12\r\n",
      "NC_007175.2 2207 13\r\n",
      "NC_007175.2 2208 13\r\n",
      "NC_007175.2 2238 14\r\n",
      "NC_007175.2 2239 13\r\n",
      "NC_007175.2 2266 10\r\n",
      "NC_007175.2 2267 12\r\n",
      "NC_007175.2 2283 9\r\n",
      "NC_007175.2 2284 12\r\n",
      "NC_007175.2 2302 11\r\n",
      "NC_007175.2 2305 2\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 1549, file 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      " source line number 1\r\n"
     ]
    }
   ],
   "source": [
    "!awk '{print $1, $2, $5+$6}' 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov \\\n",
    "| head -20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_007175.2 2149 14\r\n",
      "NC_007175.2 2181 14\r\n",
      "NC_007175.2 2182 12\r\n",
      "NC_007175.2 2207 13\r\n",
      "NC_007175.2 2208 13\r\n",
      "NC_007175.2 2238 14\r\n",
      "NC_007175.2 2239 13\r\n",
      "NC_007175.2 2267 12\r\n",
      "NC_007175.2 2284 12\r\n",
      "NC_007175.2 2302 11\r\n",
      "NC_007175.2 2998 15\r\n",
      "NC_007175.2 3022 16\r\n",
      "NC_007175.2 3030 17\r\n",
      "NC_007175.2 3060 18\r\n",
      "NC_007175.2 3063 19\r\n",
      "NC_007175.2 14084 16\r\n",
      "NC_007175.2 14314 12\r\n",
      "NC_035780.1 988 12\r\n",
      "NC_035780.1 989 15\r\n",
      "NC_035780.1 1054 15\r\n"
     ]
    }
   ],
   "source": [
    "!awk '{print $1, $2, $5+$6}' 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov \\\n",
    "| awk '{if ($3 > 10) { print }}' \\\n",
    "| head -20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls *pe.deduplicated.bismark.c*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "gunzip: 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov already exists -- skipping\n",
      "gunzip: 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov already exists -- skipping\n",
      "gunzip: 20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov already exists -- skipping\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov.gz \\\n",
    "| xargs basename -s .gz | xargs -I{} gunzip -k {}.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls *pe.deduplicated.bismark.c*\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     346\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov \\\n",
    "| xargs basename -s .cov | xargs -I{} awk '{print $1, $2, $5+$6}' {}.cov \\\n",
    "| awk '{if ($3 > 10) { print }}' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 2407830\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov \\\n",
    "| xargs basename -s .cov | xargs -I{} awk '{print $1, $2, $5+$6}' {}.cov | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      11\n",
      "     164\n",
      "     130\n",
      "       5\n",
      "       4\n",
      "       1\n",
      "       1\n",
      "       6\n",
      "      14\n",
      "       4\n",
      "       2\n",
      "       0\n",
      "       0\n",
      "       0\n",
      "       4\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in *.cov\n",
    "do\n",
    "    awk '{print $1, $2, $5+$6}' ${f} | awk '{if ($3 > 10) { print }}' | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     330\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "    1761\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "    1407\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     306\n",
      "10_32_S32_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     274\n",
      "11_37_S37_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     143\n",
      "12_38_S38_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     203\n",
      "13_33_S33_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     208\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     282\n",
      "15_39_S39_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     243\n",
      "16_40_S40_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     170\n",
      "17_9_S9_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "       1\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "      12\n",
      "19_7_S7_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "      11\n",
      "20_8_S8_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     212\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in *.cov\n",
    "do\n",
    "    echo $f\n",
    "    awk '{print $1, $2, $5+$6}' ${f} | awk '{if ($3 > 3) { print }}' | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "awk '{print $1, $2, $5+$6}' 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov | awk '{if ($3 > 10) { print }}' | wc -l\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!awk '{print $1, $2, $5+$6}' 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov | awk '{if ($3 > 10) { print }}' | wc -l\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25\n",
      "07_29_S29\n",
      "08_30_S30\n",
      "09_31_S31\n",
      "10_32_S32\n",
      "11_37_S37\n",
      "12_38_S38\n",
      "13_33_S33\n",
      "14_34_S34\n",
      "15_39_S39\n",
      "16_40_S40\n",
      "17_9_S9\n",
      "18_10_S10\n",
      "19_7_S7\n",
      "20_8_S8\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov \\\n",
    "| xargs basename -s _L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov | xargs -I{} echo {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
