{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir /Volumes/Serine/wd/18-09-16/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-09-16\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Serine/wd/18-09-16/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Volumes/Serine/wd/18-09-16'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t125064\n",
      "Mapping efficiency:\t25.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t306217\n",
      "Sequence pairs did not map uniquely:\t56856\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t23786\t((converted) top strand)\n",
      "GA/CT/CT:\t39167\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t38583\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t23528\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t241545\n",
      "Mapping efficiency:\t48.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t133632\n",
      "Sequence pairs did not map uniquely:\t124823\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t49397\t((converted) top strand)\n",
      "GA/CT/CT:\t72009\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t71301\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t48838\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t239632\n",
      "Mapping efficiency:\t47.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t139201\n",
      "Sequence pairs did not map uniquely:\t121167\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t48529\t((converted) top strand)\n",
      "GA/CT/CT:\t71719\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t71570\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t47814\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t292737\n",
      "Mapping efficiency:\t58.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t115474\n",
      "Sequence pairs did not map uniquely:\t91789\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t53078\t((converted) top strand)\n",
      "GA/CT/CT:\t93184\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t93215\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t53260\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t281488\n",
      "Mapping efficiency:\t56.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t129949\n",
      "Sequence pairs did not map uniquely:\t88563\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t47575\t((converted) top strand)\n",
      "GA/CT/CT:\t92484\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t94020\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t47409\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t214999\n",
      "Mapping efficiency:\t43.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t220181\n",
      "Sequence pairs did not map uniquely:\t64820\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t40132\t((converted) top strand)\n",
      "GA/CT/CT:\t66939\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t67978\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t39950\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t212392\n",
      "Mapping efficiency:\t42.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t222114\n",
      "Sequence pairs did not map uniquely:\t65494\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t38915\t((converted) top strand)\n",
      "GA/CT/CT:\t66809\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t67683\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t38985\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t290417\n",
      "Mapping efficiency:\t58.1%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t118739\n",
      "Sequence pairs did not map uniquely:\t90844\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t51405\t((converted) top strand)\n",
      "GA/CT/CT:\t93268\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t94214\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t51530\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t301284\n",
      "Mapping efficiency:\t60.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t105256\n",
      "Sequence pairs did not map uniquely:\t93460\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t55379\t((converted) top strand)\n",
      "GA/CT/CT:\t94300\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t95566\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t56039\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t226518\n",
      "Mapping efficiency:\t45.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t202930\n",
      "Sequence pairs did not map uniquely:\t70552\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t40641\t((converted) top strand)\n",
      "GA/CT/CT:\t72177\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t72465\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t41234\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t232738\n",
      "Mapping efficiency:\t46.5%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t196618\n",
      "Sequence pairs did not map uniquely:\t70644\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t41828\t((converted) top strand)\n",
      "GA/CT/CT:\t73930\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t74812\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t42168\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t114320\n",
      "Mapping efficiency:\t28.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t252142\n",
      "Sequence pairs did not map uniquely:\t37399\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t17173\t((converted) top strand)\n",
      "GA/CT/CT:\t39759\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t40124\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t17264\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t260159\n",
      "Mapping efficiency:\t52.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t158721\n",
      "Sequence pairs did not map uniquely:\t81120\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t42118\t((converted) top strand)\n",
      "GA/CT/CT:\t88350\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t88275\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t41416\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t177077\n",
      "Mapping efficiency:\t47.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t137934\n",
      "Sequence pairs did not map uniquely:\t56656\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t28841\t((converted) top strand)\n",
      "GA/CT/CT:\t59689\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t59619\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t28927\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t188633\n",
      "Mapping efficiency:\t59.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t66890\n",
      "Sequence pairs did not map uniquely:\t59399\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t31206\t((converted) top strand)\n",
      "GA/CT/CT:\t63033\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t63329\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t31065\t((converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-09-15/*_R1_001.fastq.gz \\\n",
    "| xargs basename -s _L001_R1_001.fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "-u 500000 \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-09-15/{}_L001_R1_001.fastq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-09-15/{}_L001_R2_001.fastq.gz \\\n",
    "2> bismark.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 01_25_S25_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/01_25_S25_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/01_25_S25_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 07_29_S29_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/07_29_S29_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/07_29_S29_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 08_30_S30_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/08_30_S30_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/08_30_S30_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 09_31_S31_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/09_31_S31_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/09_31_S31_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/14_34_S34_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/14_34_S34_L001_R2_001.fastq.gz\"\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome -u 500000 --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-09-15/18_10_S10_L001_R1_001.fastq.gz -2 /Volumes/Serine/wd/18-09-15/18_10_S10_L001_R2_001.fastq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file 01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file 18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\t18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-09-16/CpG_OT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOT_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_CTOB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-09-16/CpG_OB_18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 4 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 6 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: 01_25_S25_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 07_29_S29_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 08_30_S30_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 09_31_S31_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 14_34_S34_L001_R1_001_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: 18_10_S10_L001_R1_001_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "looking at coverage files     \n",
    "`<chromosome> <start position> <end position> <methylation percentage> <count methylated> <count unmethylated>\n",
    "`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov.gz \\\n",
    "| xargs basename -s .gz | xargs -I{} gunzip -k {}.gz "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov.gz\r\n"
     ]
    }
   ],
   "source": [
    "ls *pe.deduplicated.bismark.c*\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NC_007175.2\t592\t592\t0\t0\t1\r\n",
      "NC_007175.2\t665\t665\t0\t0\t2\r\n",
      "NC_007175.2\t684\t684\t0\t0\t2\r\n",
      "NC_007175.2\t709\t709\t0\t0\t1\r\n",
      "NC_007175.2\t976\t976\t0\t0\t1\r\n",
      "NC_007175.2\t977\t977\t0\t0\t1\r\n",
      "NC_007175.2\t995\t995\t0\t0\t1\r\n",
      "NC_007175.2\t996\t996\t0\t0\t1\r\n",
      "NC_007175.2\t1010\t1010\t0\t0\t2\r\n",
      "NC_007175.2\t1011\t1011\t0\t0\t1\r\n"
     ]
    }
   ],
   "source": [
    "!head 07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     478\n",
      "    2917\n",
      "    2331\n",
      "     472\n",
      "     464\n",
      "      24\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in *.cov\n",
    "do\n",
    "    awk '{print $1, $2, $5+$6}' ${f} | awk '{if ($3 >= 10) { print }}' | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "    4303\n",
      "07_29_S29_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "   44596\n",
      "08_30_S30_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "   35234\n",
      "09_31_S31_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "   12883\n",
      "14_34_S34_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "   12755\n",
      "18_10_S10_L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "     647\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for f in *.cov\n",
    "do\n",
    "    echo $f\n",
    "    awk '{print $1, $2, $5+$6}' ${f} | awk '{if ($3 > 3) { print }}' | wc -l\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_25_S25\n",
      "07_29_S29\n",
      "08_30_S30\n",
      "09_31_S31\n",
      "10_32_S32\n",
      "11_37_S37\n",
      "12_38_S38\n",
      "13_33_S33\n",
      "14_34_S34\n",
      "15_39_S39\n",
      "16_40_S40\n",
      "17_9_S9\n",
      "18_10_S10\n",
      "19_7_S7\n",
      "20_8_S8\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bismark.cov \\\n",
    "| xargs basename -s _L001_R1_001_bismark_bt2_pe.deduplicated.bismark.cov | xargs -I{} echo {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "for f in *.cov\n",
    "do\n",
    "    awk '{print $1, $2-1, $2, $4, $5+$6}' ${f} | awk '{if ($5 >= 3) { print $1, $2-1, $2, $4 }}' \\\n",
    "> ${f}_3x.bedgraph\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
