{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:36:12.411186Z",
     "start_time": "2018-10-02T12:36:12.401163Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/Desktop/oil/01\n"
     ]
    }
   ],
   "source": [
    "cd /Users/sr320/Desktop/oil/01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:39:05.022372Z",
     "start_time": "2018-10-02T12:37:16.566293Z"
    },
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t105\n",
      "Sequences did not map uniquely:\t283\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t27\t((converted) top strand)\n",
      "CT/GA:\t32\t((converted) bottom strand)\n",
      "GA/CT:\t270\t(complementary to (converted) top strand)\n",
      "GA/GA:\t283\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t130\n",
      "Sequences did not map uniquely:\t272\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t38\t((converted) top strand)\n",
      "CT/GA:\t56\t((converted) bottom strand)\n",
      "GA/CT:\t227\t(complementary to (converted) top strand)\n",
      "GA/GA:\t277\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t229\n",
      "Sequences did not map uniquely:\t270\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t53\t((converted) top strand)\n",
      "CT/GA:\t64\t((converted) bottom strand)\n",
      "GA/CT:\t183\t(complementary to (converted) top strand)\n",
      "GA/GA:\t201\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t105\n",
      "Sequences did not map uniquely:\t274\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t31\t((converted) top strand)\n",
      "CT/GA:\t30\t((converted) bottom strand)\n",
      "GA/CT:\t295\t(complementary to (converted) top strand)\n",
      "GA/GA:\t265\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t101\n",
      "Sequences did not map uniquely:\t325\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t46\t((converted) top strand)\n",
      "CT/GA:\t46\t((converted) bottom strand)\n",
      "GA/CT:\t240\t(complementary to (converted) top strand)\n",
      "GA/GA:\t242\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t217\n",
      "Sequences did not map uniquely:\t247\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t17\t((converted) top strand)\n",
      "CT/GA:\t25\t((converted) bottom strand)\n",
      "GA/CT:\t256\t(complementary to (converted) top strand)\n",
      "GA/GA:\t238\t(complementary to (converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-p 4 \\\n",
    "-u 1000 \\\n",
    "--non_directional \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:41:06.418472Z",
     "start_time": "2018-10-02T12:41:06.268203Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "grep: efficiency: No such file or directory\r\n",
      "bismark.err:Mapping efficiency:\t61.2%\r\n",
      "bismark.err:Mapping efficiency:\t59.8%\r\n",
      "bismark.err:Mapping efficiency:\t50.1%\r\n",
      "bismark.err:Mapping efficiency:\t62.1%\r\n",
      "bismark.err:Mapping efficiency:\t57.4%\r\n",
      "bismark.err:Mapping efficiency:\t53.6%\r\n"
     ]
    }
   ],
   "source": [
    "!grep Mapping efficiency bismark.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:43:04.871362Z",
     "start_time": "2018-10-02T12:41:19.038137Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t145\n",
      "Sequences did not map uniquely:\t269\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t22\t((converted) top strand)\n",
      "CT/GA:\t28\t((converted) bottom strand)\n",
      "GA/CT:\t263\t(complementary to (converted) top strand)\n",
      "GA/GA:\t273\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t186\n",
      "Sequences did not map uniquely:\t258\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t35\t((converted) top strand)\n",
      "CT/GA:\t48\t((converted) bottom strand)\n",
      "GA/CT:\t210\t(complementary to (converted) top strand)\n",
      "GA/GA:\t263\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t252\n",
      "Sequences did not map uniquely:\t270\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t46\t((converted) top strand)\n",
      "CT/GA:\t59\t((converted) bottom strand)\n",
      "GA/CT:\t176\t(complementary to (converted) top strand)\n",
      "GA/GA:\t197\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t145\n",
      "Sequences did not map uniquely:\t273\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t24\t((converted) top strand)\n",
      "CT/GA:\t28\t((converted) bottom strand)\n",
      "GA/CT:\t278\t(complementary to (converted) top strand)\n",
      "GA/GA:\t252\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t137\n",
      "Sequences did not map uniquely:\t320\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t41\t((converted) top strand)\n",
      "CT/GA:\t35\t((converted) bottom strand)\n",
      "GA/CT:\t229\t(complementary to (converted) top strand)\n",
      "GA/GA:\t238\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t280\n",
      "Sequences did not map uniquely:\t238\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t14\t((converted) top strand)\n",
      "CT/GA:\t19\t((converted) bottom strand)\n",
      "GA/CT:\t230\t(complementary to (converted) top strand)\n",
      "GA/GA:\t219\t(complementary to (converted) bottom strand)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "--score_min L,0,-0.9 \\\n",
    "-p 4 \\\n",
    "-u 1000 \\\n",
    "--non_directional \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:43:05.016640Z",
     "start_time": "2018-10-02T12:43:04.881454Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "grep: efficiency: No such file or directory\r\n",
      "bismark.err2:Mapping efficiency:\t58.6%\r\n",
      "bismark.err2:Mapping efficiency:\t55.6%\r\n",
      "bismark.err2:Mapping efficiency:\t47.8%\r\n",
      "bismark.err2:Mapping efficiency:\t58.2%\r\n",
      "bismark.err2:Mapping efficiency:\t54.3%\r\n",
      "bismark.err2:Mapping efficiency:\t48.2%\r\n"
     ]
    }
   ],
   "source": [
    "!grep Mapping efficiency bismark.err2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:44:45.749117Z",
     "start_time": "2018-10-02T12:43:05.021644Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t501\n",
      "Sequences did not map uniquely:\t174\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t10\t((converted) top strand)\n",
      "CT/GA:\t15\t((converted) bottom strand)\n",
      "GA/CT:\t145\t(complementary to (converted) top strand)\n",
      "GA/GA:\t155\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t504\n",
      "Sequences did not map uniquely:\t175\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t18\t((converted) top strand)\n",
      "CT/GA:\t25\t((converted) bottom strand)\n",
      "GA/CT:\t127\t(complementary to (converted) top strand)\n",
      "GA/GA:\t151\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t564\n",
      "Sequences did not map uniquely:\t184\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t25\t((converted) top strand)\n",
      "CT/GA:\t22\t((converted) bottom strand)\n",
      "GA/CT:\t104\t(complementary to (converted) top strand)\n",
      "GA/GA:\t101\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t488\n",
      "Sequences did not map uniquely:\t176\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t8\t((converted) top strand)\n",
      "CT/GA:\t14\t((converted) bottom strand)\n",
      "GA/CT:\t165\t(complementary to (converted) top strand)\n",
      "GA/GA:\t149\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t466\n",
      "Sequences did not map uniquely:\t216\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t19\t((converted) top strand)\n",
      "CT/GA:\t19\t((converted) bottom strand)\n",
      "GA/CT:\t144\t(complementary to (converted) top strand)\n",
      "GA/GA:\t136\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t620\n",
      "Sequences did not map uniquely:\t140\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t7\t((converted) top strand)\n",
      "CT/GA:\t6\t((converted) bottom strand)\n",
      "GA/CT:\t118\t(complementary to (converted) top strand)\n",
      "GA/GA:\t109\t(complementary to (converted) bottom strand)\n",
      "\n",
      "bismark.err3:Mapping efficiency:\t32.5%\n",
      "bismark.err3:Mapping efficiency:\t32.1%\n",
      "bismark.err3:Mapping efficiency:\t25.2%\n",
      "bismark.err3:Mapping efficiency:\t33.6%\n",
      "bismark.err3:Mapping efficiency:\t31.8%\n",
      "bismark.err3:Mapping efficiency:\t24.0%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "grep: efficiency: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "-p 4 \\\n",
    "-u 1000 \\\n",
    "--non_directional \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err3\n",
    "grep Mapping efficiency bismark.err3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T12:48:02.007165Z",
     "start_time": "2018-10-02T12:46:49.911288Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t913\n",
      "Sequences did not map uniquely:\t37\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t24\t((converted) top strand)\n",
      "CT/GA:\t26\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t851\n",
      "Sequences did not map uniquely:\t48\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t50\t((converted) top strand)\n",
      "CT/GA:\t51\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t799\n",
      "Sequences did not map uniquely:\t84\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t58\t((converted) top strand)\n",
      "CT/GA:\t59\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t883\n",
      "Sequences did not map uniquely:\t45\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t32\t((converted) top strand)\n",
      "CT/GA:\t40\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t827\n",
      "Sequences did not map uniquely:\t79\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t47\t((converted) top strand)\n",
      "CT/GA:\t47\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t940\n",
      "Sequences did not map uniquely:\t29\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t14\t((converted) top strand)\n",
      "CT/GA:\t17\t((converted) bottom strand)\n",
      "GA/CT:\t0\t(complementary to (converted) top strand)\n",
      "GA/GA:\t0\t(complementary to (converted) bottom strand)\n",
      "\n",
      "Number of alignments to (merely theoretical) complementary strands being rejected in total:\t0\n",
      "\n",
      "bismark.err4:Mapping efficiency:\t5.0%\n",
      "bismark.err4:Mapping efficiency:\t10.1%\n",
      "bismark.err4:Mapping efficiency:\t11.7%\n",
      "bismark.err4:Mapping efficiency:\t7.2%\n",
      "bismark.err4:Mapping efficiency:\t9.4%\n",
      "bismark.err4:Mapping efficiency:\t3.1%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "grep: efficiency: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "-u 1000 \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err4\n",
    "grep Mapping efficiency bismark.err4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T13:41:36.470633Z",
     "start_time": "2018-10-02T13:40:26.777195Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t240\n",
      "Sequences did not map uniquely:\t246\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t19\t((converted) top strand)\n",
      "CT/GA:\t22\t((converted) bottom strand)\n",
      "GA/CT:\t231\t(complementary to (converted) top strand)\n",
      "GA/GA:\t242\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t288\n",
      "Sequences did not map uniquely:\t230\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t30\t((converted) top strand)\n",
      "CT/GA:\t38\t((converted) bottom strand)\n",
      "GA/CT:\t186\t(complementary to (converted) top strand)\n",
      "GA/GA:\t228\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t251\n",
      "Sequences did not map uniquely:\t245\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t17\t((converted) top strand)\n",
      "CT/GA:\t24\t((converted) bottom strand)\n",
      "GA/CT:\t242\t(complementary to (converted) top strand)\n",
      "GA/GA:\t221\t(complementary to (converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Processed 1000 sequences in total\n",
      "\n",
      "Sequences with no alignments under any condition:\t402\n",
      "Sequences did not map uniquely:\t208\n",
      "Sequences which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequences with unique best (first) alignment came from the bowtie output:\n",
      "CT/CT:\t11\t((converted) top strand)\n",
      "CT/GA:\t13\t((converted) bottom strand)\n",
      "GA/CT:\t185\t(complementary to (converted) top strand)\n",
      "GA/GA:\t181\t(complementary to (converted) bottom strand)\n",
      "\n",
      "bismark.err4:Mapping efficiency:\t51.4%\n",
      "bismark.err4:Mapping efficiency:\t48.2%\n",
      "bismark.err4:Mapping efficiency:\t50.4%\n",
      "bismark.err4:Mapping efficiency:\t39.0%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "grep: efficiency: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "-u 1000 \\\n",
    "--non_directional \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err4\n",
    "grep Mapping efficiency bismark.err4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T13:51:41.309796Z",
     "start_time": "2018-10-02T13:51:41.287004Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/sr320/Desktop/oil/02\n"
     ]
    }
   ],
   "source": [
    "cd ../02"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-10-02T15:39:37.494969Z",
     "start_time": "2018-10-02T13:53:16.448192Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process is terminated.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Users/sr320/Desktop/Qiagen/20150414_trimmed*.fastq.gz \\\n",
    "| xargs basename -s fastq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Users/sr320/Desktop/oakl/genome \\\n",
    "--score_min L,0,-0.6 \\\n",
    "--non_directional \\\n",
    "-p 2 \\\n",
    "/Users/sr320/Desktop/Qiagen/{}fastq.gz \\\n",
    "2> bismark.err\n",
    "grep Mapping efficiency bismark.err\n",
    "\n",
    "\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -s \\\n",
    "*.bam \\\n",
    "2> dedup.err\n",
    "\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "--bedGraph --counts --scaffolds \\\n",
    "--multicore 2 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
