{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running Bismark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T15:52:42.223989Z",
     "start_time": "2018-06-13T15:52:42.216597Z"
    }
   },
   "outputs": [],
   "source": [
    "wd=\"/Volumes/Orange_block/wd/18-06-13\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T15:52:43.617281Z",
     "start_time": "2018-06-13T15:52:43.603691Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Orange_block/wd/18-06-13\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-11T23:50:57.914482Z",
     "start_time": "2018-06-11T23:50:57.910536Z"
    }
   },
   "source": [
    "## Genome Preparation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T15:52:45.411288Z",
     "start_time": "2018-06-13T15:52:45.263458Z"
    }
   },
   "outputs": [],
   "source": [
    "!mkdir genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T15:53:19.408985Z",
     "start_time": "2018-06-13T15:53:10.527570Z"
    }
   },
   "outputs": [],
   "source": [
    "!cp /Users/sr320/Desktop/oly/Olurida_v081.fa genome//Olurida_v081.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T17:38:40.094779Z",
     "start_time": "2018-06-13T15:53:29.915597Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t197058296\n",
      "G->A:\t197049137\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Reading reference sizes\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:37\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:38\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:25\n",
      "bmax according to bmaxDivN setting: 269313774\n",
      "Using parameters --bmax 201985331 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 201985331 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:28\n",
      "bmax according to bmaxDivN setting: 269313774\n",
      "Using parameters --bmax 201985331 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 201985331 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:30\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:31\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:09\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:09\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:15\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.07726e+09 (target: 201985330)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 1077255099 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:18\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.07726e+09 (target: 201985330)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 1077255099 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 01:00:36\n",
      "Returning block of 1077255100 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 538568663\n",
      "fchr[G]: 735626959\n",
      "fchr[T]: 735626959\n",
      "fchr[$]: 1077255099\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 374057575 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 269313780 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1077255099\n",
      "    bwtLen: 1077255100\n",
      "    sz: 269313775\n",
      "    bwtSz: 269313775\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67328444\n",
      "    offsSz: 269313776\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 5610704\n",
      "    numLines: 5610704\n",
      "    ebwtTotLen: 359085056\n",
      "    ebwtTotSz: 359085056\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 01:09:11\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:19\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:08\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 269313774\n",
      "Using parameters --bmax 201985331 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 201985331 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Sorting block time: 01:07:52\n",
      "Returning block of 1077255100 for bucket 1\n",
      "  V-Sorting samples time: 00:00:29\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:18\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:15\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.07726e+09 (target: 201985330)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 1077255099 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 341519526\n",
      "fchr[G]: 341519526\n",
      "fchr[T]: 538568663\n",
      "fchr[$]: 1077255099\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 374057575 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 269313780 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1077255099\n",
      "    bwtLen: 1077255100\n",
      "    sz: 269313775\n",
      "    bwtSz: 269313775\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67328444\n",
      "    offsSz: 269313776\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 5610704\n",
      "    numLines: 5610704\n",
      "    ebwtTotLen: 359085056\n",
      "    ebwtTotSz: 359085056\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 01:18:01\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:19\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:09\n",
      "  Time to reverse reference sequence: 00:00:02\n",
      "bmax according to bmaxDivN setting: 269313774\n",
      "Using parameters --bmax 201985331 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 201985331 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:30\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:11\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:14\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.07726e+09 (target: 201985330)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 1077255099 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:21:29\n",
      "Returning block of 1077255100 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 538568663\n",
      "fchr[G]: 735626959\n",
      "fchr[T]: 735626959\n",
      "fchr[$]: 1077255099\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 374057575 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 269313780 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1077255099\n",
      "    bwtLen: 1077255100\n",
      "    sz: 269313775\n",
      "    bwtSz: 269313775\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67328444\n",
      "    offsSz: 269313776\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 5610704\n",
      "    numLines: 5610704\n",
      "    ebwtTotLen: 359085056\n",
      "    ebwtTotSz: 359085056\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:27:46\n",
      "  Sorting block time: 00:20:57\n",
      "Returning block of 1077255100 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 341519526\n",
      "fchr[G]: 341519526\n",
      "fchr[T]: 538568663\n",
      "fchr[$]: 1077255099\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 374057575 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 269313780 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 1077255099\n",
      "    bwtLen: 1077255100\n",
      "    sz: 269313775\n",
      "    bwtSz: 269313775\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 67328444\n",
      "    offsSz: 269313776\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 5610704\n",
      "    numLines: 5610704\n",
      "    ebwtTotLen: 359085056\n",
      "    ebwtTotSz: 359085056\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:26:31\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bisulfite Genome Indexer version v0.19.0 (last modified 07 November 2016)\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Building a SMALL index\n",
      "Building a SMALL index\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_genome_preparation \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T19:16:03.604090Z",
     "start_time": "2018-06-13T19:16:03.479012Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34mCT_conversion\u001b[m\u001b[m \u001b[34mGA_conversion\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls genome/Bisulfite_Genome/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download Seq Read Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T19:22:21.280274Z",
     "start_time": "2018-06-13T19:22:21.149178Z"
    }
   },
   "outputs": [],
   "source": [
    "!mkdir data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2018-06-13T19:33:22.841Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2018-06-13 12:33:22--  http://owl.fish.washington.edu/nightingales/O_lurida/\n",
      "Resolving owl.fish.washington.edu (owl.fish.washington.edu)... 128.95.149.83\n",
      "Connecting to owl.fish.washington.edu (owl.fish.washington.edu)|128.95.149.83|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html.tmp’\n",
      "\n",
      "index.html.tmp          [ <=>                ]  31.90K  --.-KB/s    in 0.002s  \n",
      "\n",
      "2018-06-13 12:33:23 (15.5 MB/s) - ‘index.html.tmp’ saved [32669]\n",
      "\n",
      "Loading robots.txt; please ignore errors.\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/robots.txt\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 404 Not Found\n",
      "2018-06-13 12:33:23 ERROR 404: Not Found.\n",
      "\n",
      "Removing index.html.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/nightingales/O_lurida/?C=N;O=D\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html?C=N;O=D.tmp’\n",
      "\n",
      "index.html?C=N;O=D.     [ <=>                ]  31.90K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2018-06-13 12:33:23 (29.6 MB/s) - ‘index.html?C=N;O=D.tmp’ saved [32669]\n",
      "\n",
      "Removing index.html?C=N;O=D.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/nightingales/O_lurida/?C=M;O=A\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html?C=M;O=A.tmp’\n",
      "\n",
      "index.html?C=M;O=A.     [ <=>                ]  31.90K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2018-06-13 12:33:23 (33.1 MB/s) - ‘index.html?C=M;O=A.tmp’ saved [32669]\n",
      "\n",
      "Removing index.html?C=M;O=A.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/nightingales/O_lurida/?C=S;O=A\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html?C=S;O=A.tmp’\n",
      "\n",
      "index.html?C=S;O=A.     [ <=>                ]  31.90K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2018-06-13 12:33:23 (33.9 MB/s) - ‘index.html?C=S;O=A.tmp’ saved [32669]\n",
      "\n",
      "Removing index.html?C=S;O=A.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/nightingales/O_lurida/?C=D;O=A\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html?C=D;O=A.tmp’\n",
      "\n",
      "index.html?C=D;O=A.     [ <=>                ]  31.90K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2018-06-13 12:33:23 (22.7 MB/s) - ‘index.html?C=D;O=A.tmp’ saved [32669]\n",
      "\n",
      "Removing index.html?C=D;O=A.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:23--  http://owl.fish.washington.edu/nightingales/O_lurida/1_ATCACG_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 526965148 (503M) [application/x-gzip]\n",
      "Saving to: ‘1_ATCACG_L001_R1_001.fastq.gz.1.tmp’\n",
      "\n",
      "1_ATCACG_L001_R1_00 100%[===================>] 502.55M  46.4MB/s    in 10s     \n",
      "\n",
      "2018-06-13 12:33:34 (49.1 MB/s) - ‘1_ATCACG_L001_R1_001.fastq.gz.1.tmp’ saved [526965148/526965148]\n",
      "\n",
      "Removing 1_ATCACG_L001_R1_001.fastq.gz.1.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:34--  http://owl.fish.washington.edu/nightingales/O_lurida/2_CGATGT_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 668680694 (638M) [application/x-gzip]\n",
      "Saving to: ‘2_CGATGT_L001_R1_001.fastq.gz.1.tmp’\n",
      "\n",
      "2_CGATGT_L001_R1_00 100%[===================>] 637.70M  50.1MB/s    in 14s     \n",
      "\n",
      "2018-06-13 12:33:48 (46.9 MB/s) - ‘2_CGATGT_L001_R1_001.fastq.gz.1.tmp’ saved [668680694/668680694]\n",
      "\n",
      "Removing 2_CGATGT_L001_R1_001.fastq.gz.1.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:48--  http://owl.fish.washington.edu/nightingales/O_lurida/2bRAD_Dec2015/\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html.tmp’\n",
      "\n",
      "index.html.tmp          [ <=>                ]  80.95K  --.-KB/s    in 0.006s  \n",
      "\n",
      "2018-06-13 12:33:49 (13.7 MB/s) - ‘index.html.tmp’ saved [82896]\n",
      "\n",
      "Removing index.html.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:49--  http://owl.fish.washington.edu/nightingales/O_lurida/2bRAD_Jan2017/\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: unspecified [text/html]\n",
      "Saving to: ‘index.html.tmp’\n",
      "\n",
      "index.html.tmp          [ <=>                ]  13.97K  --.-KB/s    in 0.001s  \n",
      "\n",
      "2018-06-13 12:33:49 (24.9 MB/s) - ‘index.html.tmp’ saved [14310]\n",
      "\n",
      "Removing index.html.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:33:49--  http://owl.fish.washington.edu/nightingales/O_lurida/3_TTAGGC_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 663855515 (633M) [application/x-gzip]\n",
      "Saving to: ‘3_TTAGGC_L001_R1_001.fastq.gz.1.tmp’\n",
      "\n",
      "3_TTAGGC_L001_R1_00 100%[===================>] 633.10M  54.1MB/s    in 12s     \n",
      "\n",
      "2018-06-13 12:34:01 (54.7 MB/s) - ‘3_TTAGGC_L001_R1_001.fastq.gz.1.tmp’ saved [663855515/663855515]\n",
      "\n",
      "Removing 3_TTAGGC_L001_R1_001.fastq.gz.1.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:34:01--  http://owl.fish.washington.edu/nightingales/O_lurida/4_TGACCA_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 649308301 (619M) [application/x-gzip]\n",
      "Saving to: ‘4_TGACCA_L001_R1_001.fastq.gz.1.tmp’\n",
      "\n",
      "4_TGACCA_L001_R1_00 100%[===================>] 619.23M  22.8MB/s    in 16s     \n",
      "\n",
      "2018-06-13 12:34:17 (38.5 MB/s) - ‘4_TGACCA_L001_R1_001.fastq.gz.1.tmp’ saved [649308301/649308301]\n",
      "\n",
      "Removing 4_TGACCA_L001_R1_001.fastq.gz.1.tmp since it should be rejected.\n",
      "\n",
      "--2018-06-13 12:34:17--  http://owl.fish.washington.edu/nightingales/O_lurida/5_ACAGTG_L001_R1_001.fastq.gz\n",
      "Reusing existing connection to owl.fish.washington.edu:80.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 704024744 (671M) [application/x-gzip]\n",
      "Saving to: ‘5_ACAGTG_L001_R1_001.fastq.gz.1.tmp’\n",
      "\n",
      "001_R1_001.fastq.gz  36%[======>             ] 245.97M  63.2MB/s    eta 7s     "
     ]
    }
   ],
   "source": [
    "!wget --accept \"[1-8]_*.gz\" --reject \"index\" --recursive -nd -np \\\n",
    "http://owl.fish.washington.edu/nightingales/O_lurida/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-06-13T19:31:31.817290Z",
     "start_time": "2018-06-13T19:31:31.676471Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1_ATCACG_L001_R1_001.fastq.gz  6_GCCAAT_L001_R1_001.fastq.gz\r\n",
      "2_CGATGT_L001_R1_001.fastq.gz  7_CAGATC_L001_R1_001.fastq.gz\r\n",
      "3_TTAGGC_L001_R1_001.fastq.gz  8_ACTTGA_L001_R1_001.fastq.gz\r\n",
      "4_TGACCA_L001_R1_001.fastq.gz  \u001b[34mdata\u001b[m\u001b[m/\r\n",
      "5_ACAGTG_L001_R1_001.fastq.gz  \u001b[34mgenome\u001b[m\u001b[m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-04-27/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-0.9 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-27/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-04-27/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark.err"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
