---
title: "Comparing Loci v Window"
output:
md_document:
    variant: markdown_github
---

```{r setup, include = FALSE }
library(methylKit)
```


```
%%bash
find /Volumes/Serine/wd/18-04-27/zr2096_*R1* \
| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \
--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \
--genome /Volumes/Serine/wd/18-03-15/genome \
--score_min L,0,-1.2 \
-p 4 \
--non_directional \
-1 /Volumes/Serine/wd/18-04-27/{}_s1_R1_val_1.fq.gz \
-2 /Volumes/Serine/wd/18-04-27/{}_s1_R2_val_2.fq.gz \
2> bismark.err

%%bash
/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \
--bam -p \
*.bam \
2> dedup.err

%%bash
find *deduplicated.bam \
| xargs basename -s _s1_R1_val_1_bismark_bt2_pe.deduplicated.bam | xargs -I{} /Applications/bioinfo/samtools-1.3.1/samtools \
sort {}_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam -o {}_dedup.sorted.bam
```

Reading in some files: 


```{r}
#file.list=list(  '../../0510/data/zr2096_1_dedup.sorted.bam',
                 '../../0510/data/zr2096_2_dedup.sorted.bam',
                 '../../0510/data/zr2096_3_dedup.sorted.bam',
                 '../../0510/data/zr2096_4_dedup.sorted.bam',
                 '../../0510/data/zr2096_5_dedup.sorted.bam',
                 '../../0510/data/zr2096_6_dedup.sorted.bam',
                 '../../0510/data/zr2096_7_dedup.sorted.bam',
                 '../../0510/data/zr2096_8_dedup.sorted.bam',
                 '../../0510/data/zr2096_9_dedup.sorted.bam',
                 '../../0510/data/zr2096_10_dedup.sorted.bam'
)
```



Bringing in bam files, setting context, AND coverage

```{r, eval=FALSE}
#myobj = processBismarkAln(location = file.list, sample.id = list("1","2","3","4","5","6","7","8","9","10"), assembly = "v3", read.context="CpG", mincov=3, treatment = c(0,0,0,0,0,1,1,1,1,1))
```

```{r, eval=FALSE}
#save(myobj, file = "../analyses/myobj_0921")
```


```{r}
load("../../0921/analyses/myobj_0921")
```



# tiling

```{r}
# Function
tileit <- function(object, win.size=500, step.size=500, cov.bases=3) {
  g.meth <- as(object,"GRanges")
  chrs <- as.character(unique(seqnames(g.meth)))
  grl <- split(g.meth, seqnames(g.meth))
  max.length <- max(end(grl))
  numTiles <- floor((max.length - (win.size - step.size))/step.size) + 1
  starts <- unlist(sapply(numTiles,function(x) 1+0:(x-1)*step.size))
  ranges <- IRanges(start=starts, width=rep(win.size,length(starts)))
  all.wins <- GRanges(seqnames=Rle(chrs, numTiles), ranges=ranges)
  rcounts <- regionCounts(object, all.wins, 0, strand.aware=FALSE)
  rcounts.filtered <- rcounts[rcounts$coverage >= cov.bases, ]
  return(rcounts.filtered)
}

# Code to run function on methylKitList
new.list <- lapply(myobj, tileit)
myobj.tiled <- new("methylRawList", new.list, treatment=myobj@treatment)
```






```{r}
filtered.myobj=filterByCoverage(myobj,lo.count=3,lo.perc=NULL,
                                      hi.count=NULL,hi.perc=99.9)

meth_filter=unite(filtered.myobj, destrand=TRUE)

clusterSamples(meth_filter, dist="correlation", method="ward", plot=TRUE)


PCASamples(meth_filter)
```



```{r}

tiled.filtered.myobj=filterByCoverage(myobj.tiled,lo.count=3,lo.perc=NULL,
                                      hi.count=NULL,hi.perc=99.9)

meth_tile=unite(tiled.filtered.myobj, destrand=TRUE)

clusterSamples(meth_tile, dist="correlation", method="ward", plot=TRUE)


PCASamples(meth_tile)
```




```{r, cache=TRUE}
myDiff.loci=calculateDiffMeth(meth_filter)

```

```{r}


myDiff25p.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01)


# get hyper-methylated
hyper.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01,type="hyper")

# get hypo-methylated
hypo.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01,type="hypo")

```


```{r}


myDiff25p.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01)


# get hyper-methylated
hyper.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01,type="hyper")

# get hypo-methylated
hypo.l=getMethylDiff(myDiff.loci,difference=25,qvalue=0.01,type="hypo")

```




```{r, cache=TRUE}
myDiff.tile=calculateDiffMeth(meth_tile)

```



```{r}


myDiff25p.t=getMethylDiff(myDiff.tile,difference=25,qvalue=0.01)


# get hyper-methylated
hyper.t=getMethylDiff(myDiff.tile,difference=25,qvalue=0.01,type="hyper")

# get hypo-methylated
hypo.t=getMethylDiff(myDiff.tile,difference=25,qvalue=0.01,type="hypo")

```
















```{r}
write.table(hyper.l, file = "../analyses/hyper.l", sep = "\t")

write.table(myDiff25p.l, file = "../analyses/myDiff25p.l", sep = "\t")

write.table(myDiff25p.t, file = "../analyses/myDiff25p.t", sep = "\t")


```


---
# Taking the DMLs to a bed

```{r}
library(readr)
myDiff50p <- read_csv("../analyses/myDiff50p.csv")
 
```


```{r}
head(myDiff50p)
```


```{r}
library(tidyverse)
```



```{r}
dml05251200plus <- filter(myDiff50p, strand == "+") %>% mutate(start = start -1, end = end + 1) %>% select(chr, start, end, strand, meth.diff)
dml05251200minus <- filter(myDiff50p, strand == "-") %>% mutate(start = start -2) %>% select(chr, start, end, strand, meth.diff)

dml05251200.bed <- bind_rows(dml05251200plus, dml05251200minus) %>% arrange(chr, start) %>% 
  mutate_if(is.numeric, as.integer) 
#latter was done to refrain from writing as scientific notation


dml05251200.bed
write_delim(dml05251200.bed, "../analyses/dml05251200.bed",  delim = '\t', col_names = FALSE)
```



