{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Running 10bp dtrim through methlykit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T14:53:06.442644Z",
     "start_time": "2018-04-14T14:53:06.434138Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "wd=\"/Volumes/Serine/wd/18-04-29\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T14:53:07.201398Z",
     "start_time": "2018-04-14T14:53:07.063654Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "mkdir $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T14:53:42.954203Z",
     "start_time": "2018-04-14T14:53:42.942549Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Serine/wd/18-04-29\n"
     ]
    }
   ],
   "source": [
    "cd $wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:29:25.184796Z",
     "start_time": "2018-04-14T14:53:42.959241Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t9282636\n",
      "Mapping efficiency:\t53.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t4524985\n",
      "Sequence pairs did not map uniquely:\t3641262\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t0\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2365261\t((converted) top strand)\n",
      "GA/CT/CT:\t2269368\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t2275104\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2372903\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t8273829\n",
      "Mapping efficiency:\t28.9%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t17321484\n",
      "Sequence pairs did not map uniquely:\t3008033\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t2086448\t((converted) top strand)\n",
      "GA/CT/CT:\t2087306\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t2052271\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t2047803\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t15247793\n",
      "Mapping efficiency:\t50.3%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t8686774\n",
      "Sequence pairs did not map uniquely:\t6391039\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t3\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3850832\t((converted) top strand)\n",
      "GA/CT/CT:\t3774008\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3773813\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3849137\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t15542349\n",
      "Mapping efficiency:\t52.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t7362526\n",
      "Sequence pairs did not map uniquely:\t6643878\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3906974\t((converted) top strand)\n",
      "GA/CT/CT:\t3851211\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3865023\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3919139\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t12995901\n",
      "Mapping efficiency:\t54.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5579694\n",
      "Sequence pairs did not map uniquely:\t5394921\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t2\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3325378\t((converted) top strand)\n",
      "GA/CT/CT:\t3162868\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3171036\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3336617\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t16377060\n",
      "Mapping efficiency:\t52.0%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t8772590\n",
      "Sequence pairs did not map uniquely:\t6353631\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t5\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t4233791\t((converted) top strand)\n",
      "GA/CT/CT:\t3962037\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3951534\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t4229693\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t13016030\n",
      "Mapping efficiency:\t54.4%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t5434045\n",
      "Sequence pairs did not map uniquely:\t5459418\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t1\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3414853\t((converted) top strand)\n",
      "GA/CT/CT:\t3092674\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3093838\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3414664\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t15123232\n",
      "Mapping efficiency:\t51.7%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t8438065\n",
      "Sequence pairs did not map uniquely:\t5712338\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t6\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3844203\t((converted) top strand)\n",
      "GA/CT/CT:\t3715088\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3714494\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3849441\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t14200416\n",
      "Mapping efficiency:\t48.2%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t9943053\n",
      "Sequence pairs did not map uniquely:\t5339749\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t11\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t3629946\t((converted) top strand)\n",
      "GA/CT/CT:\t3487296\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3468647\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t3614516\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "chr NC_035780.1 (65668440 bp)\n",
      "chr NC_035781.1 (61752955 bp)\n",
      "chr NC_035782.1 (77061148 bp)\n",
      "chr NC_035783.1 (59691872 bp)\n",
      "chr NC_035784.1 (98698416 bp)\n",
      "chr NC_035785.1 (51258098 bp)\n",
      "chr NC_035786.1 (57830854 bp)\n",
      "chr NC_035787.1 (75944018 bp)\n",
      "chr NC_035788.1 (104168038 bp)\n",
      "chr NC_035789.1 (32650045 bp)\n",
      "chr NC_007175.2 (17244 bp)\n",
      "\n",
      "Number of paired-end alignments with a unique best hit:\t16108088\n",
      "Mapping efficiency:\t50.6%\n",
      "\n",
      "Sequence pairs with no alignments under any condition:\t9175758\n",
      "Sequence pairs did not map uniquely:\t6563695\n",
      "Sequence pairs which were discarded because genomic sequence could not be extracted:\t5\n",
      "\n",
      "Number of sequence pairs with unique best (first) alignment came from the bowtie output:\n",
      "CT/GA/CT:\t4345578\t((converted) top strand)\n",
      "GA/CT/CT:\t3702815\t(complementary to (converted) top strand)\n",
      "GA/CT/GA:\t3705364\t(complementary to (converted) bottom strand)\n",
      "CT/GA/GA:\t4354326\t((converted) bottom strand)\n",
      "\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n",
      "FastQ format assumed (by default)\n",
      "Each Bowtie 2 instance is going to be run with 4 threads. Please monitor performance closely and tune down if needed!\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find /Volumes/Serine/wd/18-04-27/zr2096_*R1* \\\n",
    "| xargs basename -s _s1_R1_val_1.fq.gz | xargs -I{} /Applications/bioinfo/Bismark_v0.19.0/bismark \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 \\\n",
    "--genome /Volumes/Serine/wd/18-03-15/genome \\\n",
    "--score_min L,0,-1.2 \\\n",
    "-p 4 \\\n",
    "--non_directional \\\n",
    "-1 /Volumes/Serine/wd/18-04-27/{}_s1_R1_val_1.fq.gz \\\n",
    "-2 /Volumes/Serine/wd/18-04-27/{}_s1_R2_val_2.fq.gz \\\n",
    "2> bismark.err\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_10_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_10_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_1_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_1_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_2_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_2_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_3_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_3_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_4_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_4_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_5_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_5_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_6_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_6_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_7_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_7_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_8_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_8_s1_R2_val_2.fq.gz\"\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\tskipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
      "skipping header line:\t@SQ\tSN:NC_035780.1\tLN:65668440\n",
      "skipping header line:\t@SQ\tSN:NC_035781.1\tLN:61752955\n",
      "skipping header line:\t@SQ\tSN:NC_035782.1\tLN:77061148\n",
      "skipping header line:\t@SQ\tSN:NC_035783.1\tLN:59691872\n",
      "skipping header line:\t@SQ\tSN:NC_035784.1\tLN:98698416\n",
      "skipping header line:\t@SQ\tSN:NC_035785.1\tLN:51258098\n",
      "skipping header line:\t@SQ\tSN:NC_035786.1\tLN:57830854\n",
      "skipping header line:\t@SQ\tSN:NC_035787.1\tLN:75944018\n",
      "skipping header line:\t@SQ\tSN:NC_035788.1\tLN:104168038\n",
      "skipping header line:\t@SQ\tSN:NC_035789.1\tLN:32650045\n",
      "skipping header line:\t@SQ\tSN:NC_007175.2\tLN:17244\n",
      "skipping header line:\t@PG\tID:Bismark\tVN:v0.19.0\tCL:\"bismark --path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64 --genome /Volumes/Serine/wd/18-03-15/genome --score_min L,0,-1.2 -p 4 --non_directional -1 /Volumes/Serine/wd/18-04-27/zr2096_9_s1_R1_val_1.fq.gz -2 /Volumes/Serine/wd/18-04-27/zr2096_9_s1_R2_val_2.fq.gz\"\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "/Applications/bioinfo/Bismark_v0.19.0/deduplicate_bismark \\\n",
    "--bam -p \\\n",
    "*.bam \\\n",
    "2> dedup.err\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bam_sort_core] merging from 13 files...\n",
      "[bam_sort_core] merging from 11 files...\n",
      "[bam_sort_core] merging from 21 files...\n",
      "[bam_sort_core] merging from 22 files...\n",
      "[bam_sort_core] merging from 19 files...\n",
      "[bam_sort_core] merging from 23 files...\n",
      "[bam_sort_core] merging from 19 files...\n",
      "[bam_sort_core] merging from 22 files...\n",
      "[bam_sort_core] merging from 20 files...\n",
      "[bam_sort_core] merging from 21 files...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "find *deduplicated.bam \\\n",
    "| xargs basename -s _s1_R1_val_1_bismark_bt2_pe.deduplicated.bam | xargs -I{} /Applications/bioinfo/samtools-1.3.1/samtools \\\n",
    "sort {}_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam -o {}_dedup.sorted.bam\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "library(methylKit)\n",
    "\n",
    "file.list_a=list('zr2096_1_dedup.sorted.bam',\n",
    " 'zr2096_2_dedup.sorted.bam',\n",
    "  'zr2096_3_dedup.sorted.bam',\n",
    "  'zr2096_4_dedup.sorted.bam',\n",
    "  'zr2096_5_dedup.sorted.bam',\n",
    "  'zr2096_6_dedup.sorted.bam',\n",
    "  'zr2096_6_dedup.sorted.bam',\n",
    "  'zr2096_8_dedup.sorted.bam',\n",
    "  'zr2096_9_dedup.sorted.bam',\n",
    "  'zr2096_10_dedup.sorted.bam'\n",
    "  )\n",
    "\n",
    "\n",
    "\n",
    "myobj2 = processBismarkAln(location = file.list_a, sample.id = list(\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\"), assembly = \"v3\", read.context=\"CpG\", mincov=1, treatment = c(0,0,0,0,0,1,1,1,1,1))\n",
    "\n",
    "\n",
    "\n",
    "getMethylationStats(myobj2[[1]],plot=FALSE,both.strands=FALSE)\n",
    "\n",
    "getMethylationStats(myobj2[[1]],plot=TRUE,both.strands=FALSE)\n",
    "\n",
    "getCoverageStats(myobj2[[4]],plot=TRUE,both.strands=FALSE)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "meth=unite(myobj2)\n",
    "\n",
    "\n",
    "\n",
    "getCorrelation(meth,plot=TRUE)\n",
    "\n",
    "\n",
    "clusterSamples(meth, dist=\"correlation\", method=\"ward\", plot=TRUE)\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:36:44.782488Z",
     "start_time": "2018-04-14T15:30:21.010200Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Now testing Bismark result file zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n",
      "\n",
      "Now testing Bismark result file zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.bam for positional sorting (which would be bad...)\tzr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8\n",
      "\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.1.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.2.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.3.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.4.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.5.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.6.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.7.mbias\n",
      "zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt.8.mbias\n",
      "\n",
      "/Volumes/Serine/wd/18-04-29/CpG_OT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOT_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_CTOB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\t/Volumes/Serine/wd/18-04-29/CpG_OB_zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark_methylation_extractor \\\n",
    "-p --bedGraph --counts --scaffolds \\\n",
    "--multicore 8 \\\n",
    "*deduplicated.bam \\\n",
    "2> bme.err"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:36:45.102666Z",
     "start_time": "2018-04-14T15:36:44.789837Z"
    }
   },
   "outputs": [],
   "source": [
    "!gunzip *bedGraph.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:36:45.528915Z",
     "start_time": "2018-04-14T15:36:45.245888Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Found 10 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_10_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_1_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_2_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_3_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_4_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_5_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_6_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_7_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_8_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n",
      "\n",
      "Writing Bismark HTML report to >> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.html <<\n",
      "\n",
      "==============================================================================================================\n",
      "Using the following alignment report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt <\n",
      "Processing alignment report zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following deduplication report:\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt <\n",
      "Processing deduplication report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplication_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following splitting report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt <\n",
      "Processing splitting report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated_splitting_report.txt ...\n",
      "Complete\n",
      "\n",
      "Using the following M-bias report:\t\t> zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt <\n",
      "Processing M-bias report zr2096_9_s1_R1_val_1_bismark_bt2_pe.deduplicated.M-bias.txt ...\n",
      "Complete\n",
      "\n",
      "No nucleotide coverage report file specified, skipping this step\n",
      "==============================================================================================================\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2report"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:36:45.659778Z",
     "start_time": "2018-04-14T15:36:45.534306Z"
    },
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!#open *report.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2018-04-14T15:36:45.761755Z",
     "start_time": "2018-04-14T15:36:45.665360Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No Bismark/Bowtie2 single-end BAM files detected\n",
      "Found Bismark/Bowtie2 paired-end files\n",
      "No Bismark/Bowtie single-end BAM files detected\n",
      "No Bismark/Bowtie paired-end BAM files detected\n",
      "\n",
      "Generating Bismark summary report from 10 Bismark BAM file(s)...\n",
      ">> Reading from Bismark report: zr2096_10_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_1_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_2_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_3_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_4_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_5_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_6_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_7_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_8_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      ">> Reading from Bismark report: zr2096_9_s1_R1_val_1_bismark_bt2_PE_report.txt\n",
      "\n",
      "Wrote Bismark project summary to >> bismark_summary_report.html <<\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "/Applications/bioinfo/Bismark_v0.19.0/bismark2summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
